---
title: "16S preliminary analysis"
author: "Liz Mallott"
date: "12/11/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Loading required libraries

The following libraries are required for this analysis:
tidyverse (v.1.3.0)
vegan (v.2.5-6)
pairwiseAdonis (v.0.3, https://github.com/pmartinezarbizu/pairwiseAdonis)
ggtext (v.0.1.1)
cowplot (v.1.1.0)
glmmTMB (v.1.0.2.1)
lme4 (v.1.1-27.1)
emmeans (v.1.6.3)
car (v.3.0-9)
fdrtool (v.1.2.15)
ggpubr (v.0.4.0)
phyloseq (v.1.34.0)
ANCOMBC (v.3.13, https://bioconductor.org/packages/release/bioc/html/ANCOMBC.html)
pROC (v.1.18.0)
VennDiagram (v.1.6.0)


```{r load_libraries}
library(tidyverse)
library(vegan)
library(pairwiseAdonis)
library(ggtext)
library(cowplot)
library(glmmTMB)
library(lme4)
library(multcomp)
library(car)
library(fdrtool)
library(ggpubr)
library(phyloseq)
library(ANCOMBC)
library(VennDiagram)
```

## Importing data

Import distance matrices created in QIIME2-2021.2. Import feature table. Import metadata.

```{r import_distances}
unweighted_re = read_tsv("unweighted-distance-matrix-re-noother.tsv")
weighted_re = read_tsv("weighted-distance-matrix-re-noother.tsv")
bray_re = read_tsv("bray-distance-matrix-noother.tsv")

unweighted_re_age_sex_delivery_feeding = read_tsv("unweighted-distance-matrix-re-age-sex-delivery-feeding.tsv")
weighted_re_age_sex_delivery_feeding = read_tsv("weighted-distance-matrix-re-age-sex-delivery-feeding.tsv")
bray_re_age_sex_delivery_feeding = read_tsv("bray-curtis-distance-matrix-re-age-sex-delivery-feeding.tsv")

unweighted_re_onesubj = read_tsv("unweighted-distance-matrix-re-onesubj.tsv")
weighted_re_onesubj = read_tsv("weighted-distance-matrix-re-onesubj.tsv")

unweighted_re_age_sex_delivery_feeding_onesubj = read_tsv("unweighted-distance-matrix-re-age-sex-delivery-feeding-onesubj.tsv")
weighted_re_age_sex_delivery_feeding_onesubj = read_tsv("weighted-distance-matrix-re-age-sex-delivery-feeding-onesubj.tsv")

unweighted_re_01week = read_tsv("unweighted-distance-matrix-re-01week.tsv")
weighted_re_01week = read_tsv("weighted-distance-matrix-re-01week.tsv")
bray_re_01week = read_tsv("bray-curtis-distance-matrix-re-01week.tsv")
unweighted_re_16week = read_tsv("unweighted-distance-matrix-re-16week.tsv")
weighted_re_16week = read_tsv("weighted-distance-matrix-re-16week.tsv")
bray_re_16week = read_tsv("bray-curtis-distance-matrix-16week.tsv")
unweighted_re_06week = read_tsv("unweighted-distance-matrix-re-06week.tsv")
weighted_re_06week = read_tsv("weighted-distance-matrix-re-06week.tsv")
unweighted_re_6week3month = read_tsv("unweighted-distance-matrix-re-6week3month.tsv")
weighted_re_6week3month = read_tsv("weighted-distance-matrix-re-6week3month.tsv")
bray_re_6week3month = read_tsv("bray-curtis-distance-matrix-6week3month.tsv")
unweighted_re_312month = read_tsv("unweighted-distance-matrix-re-312month.tsv")
weighted_re_312month = read_tsv("weighted-distance-matrix-re-312month.tsv")
bray_re_312month = read_tsv("bray-curtis-distance-matrix-312month.tsv")
unweighted_re_012month = read_tsv("unweighted-distance-matrix-re-012month.tsv")
weighted_re_012month = read_tsv("weighted-distance-matrix-re-012month.tsv")
unweighted_re_13year = read_tsv("unweighted-distance-matrix-re-13year.tsv")
weighted_re_13year = read_tsv("weighted-distance-matrix-re-13year.tsv")
bray_re_13year = read_tsv("bray-curtis-distance-matrix-13year.tsv")
unweighted_re_36year = read_tsv("unweighted-distance-matrix-re-36year.tsv")
weighted_re_36year = read_tsv("weighted-distance-matrix-re-36year.tsv")
unweighted_re_612year = read_tsv("unweighted-distance-matrix-re-612year.tsv")
weighted_re_612year = read_tsv("weighted-distance-matrix-re-612year.tsv")
unweighted_re_312year = read_tsv("unweighted-distance-matrix-re-312year.tsv")
weighted_re_312year = read_tsv("weighted-distance-matrix-re-312year.tsv")
bray_re_312year = read_tsv("bray-curtis-distance-matrix-312year.tsv")

asvs = read_tsv("feature-table-full.tsv")

metadata = read_tsv("metadata-combined-re.txt", 
                    col_types = "cccccffffffffffffdddfdffffdfdffffffffffddd") 
metadata_re = metadata %>% 
  right_join(unweighted_re, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)

metadata_re_onesubj = metadata %>% 
  right_join(unweighted_re_onesubj, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)

metadata_re_age_sex_delivery_feeding = metadata %>% 
  right_join(unweighted_re_age_sex_delivery_feeding, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)

metadata_re_age_sex_del_fed_onesubj = metadata %>% 
  right_join(unweighted_re_age_sex_delivery_feeding_onesubj, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)

metadata_re_01week = metadata %>% 
  right_join(unweighted_re_01week, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_16week = metadata %>% 
  right_join(unweighted_re_16week, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_06week = metadata %>% 
  right_join(unweighted_re_06week, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_6week3month = metadata %>% 
  right_join(unweighted_re_6week3month, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_312month = metadata %>% 
  right_join(unweighted_re_312month, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_012month = metadata %>% 
  right_join(unweighted_re_012month, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_13year = metadata %>% 
  right_join(unweighted_re_13year, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_36year = metadata %>% 
  right_join(unweighted_re_36year, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_612year = metadata %>% 
  right_join(unweighted_re_612year, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)
metadata_re_312year = metadata %>% 
  right_join(unweighted_re_312year, by = c("SampleID" = "X1")) %>% 
               dplyr::select(1:40)

metadata_re_kim = metadata_re %>% 
  filter(Study == "Kim")
metadata_re_herman = metadata_re %>% 
  filter(Study == "Herman")
metadata_re_chu = metadata_re %>% 
  filter(Study == "Chu")
metadata_re_robinson = metadata_re %>% 
  filter(Study == "Robinson")
metadata_re_planer = metadata_re %>% 
  filter(Study == "Planer")
metadata_re_levin = metadata_re %>% 
  filter(Study == "Levin")
metadata_re_db = metadata_re %>% 
  filter(Study == "Dominguez-Bello")
metadata_re_cioffi = metadata_re %>% 
  filter(Study == "Cioffi")


unweighted_re_dist = as.dist(unweighted_re[,2:2768])
weighted_re_dist = as.dist(weighted_re[,2:2768])
bray_re_dist = as.dist(bray_re[,2:2768])

unweighted_re_age_sex_del_fed_dist = as.dist(unweighted_re_age_sex_delivery_feeding[,2:2291])
weighted_re_age_sex_del_fed_dist = as.dist(weighted_re_age_sex_delivery_feeding[,2:2291])
bray_re_age_sex_del_fed_dist = as.dist(bray_re_age_sex_delivery_feeding[,2:2291])

unweighted_re_onesubj_dist = as.dist(unweighted_re_onesubj[,2:717])
weighted_re_onesubj_dist = as.dist(weighted_re_onesubj[,2:717])

unweighted_re_age_sex_del_fed_onesubj_dist = as.dist(unweighted_re_age_sex_delivery_feeding_onesubj[,2:430])
weighted_re_age_sex_del_fed_onesubj_dist = as.dist(weighted_re_age_sex_delivery_feeding_onesubj[,2:430])

unweighted_re_01week_dist = as.dist(unweighted_re_01week[,2:120])
weighted_re_01week_dist = as.dist(weighted_re_01week[,2:120])
bray_re_01week_dist = as.dist(bray_re_01week[,2:120])
unweighted_re_16week_dist = as.dist(unweighted_re_16week[,2:286])
weighted_re_16week_dist = as.dist(weighted_re_16week[,2:286])
bray_re_16week_dist = as.dist(bray_re_16week[,2:286])
unweighted_re_06week_dist = as.dist(unweighted_re_06week[,2:405])
weighted_re_06week_dist = as.dist(weighted_re_06week[,2:405])
unweighted_re_6week3month_dist = as.dist(unweighted_re_6week3month[,2:245])
weighted_re_6week3month_dist = as.dist(weighted_re_6week3month[,2:245])
bray_re_6week3month_dist = as.dist(bray_re_6week3month[,2:245])
unweighted_re_312month_dist = as.dist(unweighted_re_312month[,2:831])
weighted_re_312month_dist = as.dist(weighted_re_312month[,2:831])
bray_re_312month_dist = as.dist(bray_re_312month[,2:831])
unweighted_re_012month_dist = as.dist(unweighted_re_012month[,2:1479])
weighted_re_012month_dist = as.dist(weighted_re_012month[,2:1479])
unweighted_re_13year_dist = as.dist(unweighted_re_13year[,2:1008])
weighted_re_13year_dist = as.dist(weighted_re_13year[,2:1008])
bray_re_13year_dist = as.dist(bray_re_13year[,2:1008])
unweighted_re_36year_dist = as.dist(unweighted_re_36year[,2:157])
weighted_re_36year_dist = as.dist(weighted_re_36year[,2:157])
unweighted_re_612year_dist = as.dist(unweighted_re_612year[,2:112])
weighted_re_612year_dist = as.dist(weighted_re_612year[,2:112])
unweighted_re_312year_dist = as.dist(unweighted_re_312year[,2:268])
weighted_re_312year_dist = as.dist(weighted_re_312year[,2:268])
bray_re_312year_dist = as.dist(bray_re_312year[,2:268])
```

## PERMANOVAs

Permanovas were performed to assess the effect of race and ethnicity on microbial community composition, controlling for study, sequencing method.

#### Unweighted
```{r permanovas_unweighted}
adonis2(unweighted_re_dist ~ Race + Ethnicity + Study + host_subject_id, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(unweighted_re_dist ~ Race + Ethnicity + host_subject_id, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

adonis2(unweighted_re_dist ~ Race + Ethnicity + Study + family, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(unweighted_re_dist ~ Race + Ethnicity + family, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

anova(betadisper(unweighted_re_dist, group = metadata_re$Race))
anova(betadisper(unweighted_re_dist, group = metadata_re$Ethnicity))
anova(betadisper(unweighted_re_dist, group = metadata_re$Study))
anova(betadisper(unweighted_re_dist, group = metadata_re$host_subject_id))
anova(betadisper(unweighted_re_dist, group = metadata_re$family))

adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)

adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Race))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Ethnicity))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Study))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Host_age_cat))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_sex))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_del_route))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Breastfeeding))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_subject_id))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

adonis2(unweighted_re_onesubj_dist ~ Race + Ethnicity + Study, data = metadata_re_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$Race))
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$Ethnicity))
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$Study))
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(unweighted_re_onesubj_dist ~ Race + Ethnicity, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(unweighted_re_onesubj_dist ~ Race + Ethnicity + family, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

adonis2(unweighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Race))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Ethnicity))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Study))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Host_age_cat))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_sex))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_del_route))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Breastfeeding))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(unweighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(unweighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_01week, Study)
adonis2(unweighted_re_01week_dist ~ Race + Ethnicity, data = metadata_re_01week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_01week_dist, group = metadata_re_01week$Race))
anova(betadisper(unweighted_re_01week_dist, group = metadata_re_01week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_16week, Study)
adonis2(unweighted_re_16week_dist ~ Race + Ethnicity, data = metadata_re_16week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_16week_dist, group = metadata_re_16week$Race))
anova(betadisper(unweighted_re_16week_dist, group = metadata_re_16week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_06week, Study)
adonis2(unweighted_re_06week_dist ~ Race + Ethnicity, data = metadata_re_06week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_06week_dist, group = metadata_re_06week$Race))
anova(betadisper(unweighted_re_06week_dist, group = metadata_re_06week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_6week3month, Study)
adonis2(unweighted_re_6week3month_dist ~ Race + Ethnicity, data = metadata_re_6week3month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_6week3month_dist, group = metadata_re_6week3month$Race))
anova(betadisper(unweighted_re_6week3month_dist, group = metadata_re_6week3month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312month, Study)
adonis2(unweighted_re_312month_dist ~ Race + Ethnicity, data = metadata_re_312month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_312month_dist, group = metadata_re_312month$Race))
anova(betadisper(unweighted_re_312month_dist, group = metadata_re_312month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(unweighted_re_012month_dist ~ Race + Ethnicity + host_subject_id, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$Race))
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$Ethnicity))
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$host_subject_id))
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(unweighted_re_012month_dist ~ Race + Ethnicity + family, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(unweighted_re_012month_dist ~ Race + Ethnicity, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_13year, Study)
adonis2(unweighted_re_13year_dist ~ Race + Ethnicity, data = metadata_re_13year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_13year_dist, group = metadata_re_13year$Race))
anova(betadisper(unweighted_re_13year_dist, group = metadata_re_13year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_36year, Study)
adonis2(unweighted_re_36year_dist ~ Race + Ethnicity, data = metadata_re_36year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_36year_dist, group = metadata_re_36year$Race))
anova(betadisper(unweighted_re_36year_dist, group = metadata_re_36year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_612year, Study)
adonis2(unweighted_re_612year_dist ~ Race + Ethnicity, data = metadata_re_612year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_612year_dist, group = metadata_re_612year$Race))
anova(betadisper(unweighted_re_612year_dist, group = metadata_re_612year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312year, Study)
adonis2(unweighted_re_312year_dist ~ Race + Ethnicity, data = metadata_re_312year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_312year_dist, group = metadata_re_312year$Race))
anova(betadisper(unweighted_re_312year_dist, group = metadata_re_312year$Ethnicity))
```
#### Weighted
```{r permanovas_weighted}
adonis2(weighted_re_dist ~ Race + Ethnicity + Study + host_subject_id, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(weighted_re_dist ~ Race + Ethnicity + host_subject_id, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

adonis2(weighted_re_dist ~ Race + Ethnicity + Study + family, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(weighted_re_dist ~ Race + Ethnicity + family, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

anova(betadisper(weighted_re_dist, group = metadata_re$Race))
anova(betadisper(weighted_re_dist, group = metadata_re$Ethnicity))
anova(betadisper(weighted_re_dist, group = metadata_re$Study))
anova(betadisper(weighted_re_dist, group = metadata_re$host_subject_id))
anova(betadisper(weighted_re_dist, group = metadata_re$family))

adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)

adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Race))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Ethnicity))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Study))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Host_age_cat))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_sex))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_del_route))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Breastfeeding))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_subject_id))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

adonis2(weighted_re_onesubj_dist ~ Race + Ethnicity + Study, data = metadata_re_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$Race))
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$Ethnicity))
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$Study))
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(weighted_re_onesubj_dist ~ Race + Ethnicity, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(weighted_re_onesubj_dist ~ Race + Ethnicity + family, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

adonis2(weighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Race))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Ethnicity))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Study))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Host_age_cat))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_sex))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_del_route))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Breastfeeding))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(weighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(weighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_01week, Study)
adonis2(weighted_re_01week_dist ~ Race + Ethnicity, data = metadata_re_01week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_01week_dist, group = metadata_re_01week$Race))
anova(betadisper(weighted_re_01week_dist, group = metadata_re_01week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_16week, Study)
adonis2(weighted_re_16week_dist ~ Race + Ethnicity, data = metadata_re_16week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_16week_dist, group = metadata_re_16week$Race))
anova(betadisper(weighted_re_16week_dist, group = metadata_re_16week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_06week, Study)
adonis2(weighted_re_06week_dist ~ Race + Ethnicity, data = metadata_re_06week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_06week_dist, group = metadata_re_06week$Race))
anova(betadisper(weighted_re_06week_dist, group = metadata_re_06week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_6week3month, Study)
adonis2(weighted_re_6week3month_dist ~ Race + Ethnicity, data = metadata_re_6week3month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_6week3month_dist, group = metadata_re_6week3month$Race))
anova(betadisper(weighted_re_6week3month_dist, group = metadata_re_6week3month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312month, Study)
adonis2(weighted_re_312month_dist ~ Race + Ethnicity, data = metadata_re_312month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_312month_dist, group = metadata_re_312month$Race))
anova(betadisper(weighted_re_312month_dist, group = metadata_re_312month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(weighted_re_012month_dist ~ Race + Ethnicity + host_subject_id, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$Race))
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$Ethnicity))
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$host_subject_id))
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(weighted_re_012month_dist ~ Race + Ethnicity + family, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(weighted_re_012month_dist ~ Race + Ethnicity, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_13year, Study)
adonis2(weighted_re_13year_dist ~ Race + Ethnicity, data = metadata_re_13year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_13year_dist, group = metadata_re_13year$Race))
anova(betadisper(weighted_re_13year_dist, group = metadata_re_13year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_36year, Study)
adonis2(weighted_re_36year_dist ~ Race + Ethnicity, data = metadata_re_36year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_36year_dist, group = metadata_re_36year$Race))
anova(betadisper(weighted_re_36year_dist, group = metadata_re_36year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_612year, Study)
adonis2(weighted_re_612year_dist ~ Race + Ethnicity, data = metadata_re_612year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_612year_dist, group = metadata_re_612year$Race))
anova(betadisper(weighted_re_612year_dist, group = metadata_re_612year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312year, Study)
adonis2(weighted_re_312year_dist ~ Race + Ethnicity, data = metadata_re_312year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_312year_dist, group = metadata_re_312year$Race))
anova(betadisper(weighted_re_312year_dist, group = metadata_re_312year$Ethnicity))
```

#### Bray-Curtis
```{r permanovas_bray}
perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(bray_re_dist ~ Race + Ethnicity + host_subject_id, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(bray_re_dist ~ Race + Ethnicity + family, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

anova(betadisper(bray_re_dist, group = metadata_re$Race))
anova(betadisper(bray_re_dist, group = metadata_re$Ethnicity))
anova(betadisper(bray_re_dist, group = metadata_re$Study))
anova(betadisper(bray_re_dist, group = metadata_re$host_subject_id))
anova(betadisper(bray_re_dist, group = metadata_re$family))


anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Race))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Ethnicity))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Study))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Host_age_cat))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_sex))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_del_route))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Breastfeeding))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_subject_id))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(bray_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(bray_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_01week, Study)
adonis2(bray_re_01week_dist ~ Race + Ethnicity, data = metadata_re_01week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_01week_dist, group = metadata_re_01week$Race))
anova(betadisper(bray_re_01week_dist, group = metadata_re_01week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_16week, Study)
adonis2(bray_re_16week_dist ~ Race + Ethnicity, data = metadata_re_16week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_16week_dist, group = metadata_re_16week$Race))
anova(betadisper(bray_re_16week_dist, group = metadata_re_16week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_6week3month, Study)
adonis2(bray_re_6week3month_dist ~ Race + Ethnicity, data = metadata_re_6week3month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_6week3month_dist, group = metadata_re_6week3month$Race))
anova(betadisper(bray_re_6week3month_dist, group = metadata_re_6week3month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312month, Study)
adonis2(bray_re_312month_dist ~ Race + Ethnicity, data = metadata_re_312month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_312month_dist, group = metadata_re_312month$Race))
anova(betadisper(bray_re_312month_dist, group = metadata_re_312month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_13year, Study)
adonis2(bray_re_13year_dist ~ Race + Ethnicity, data = metadata_re_13year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_13year_dist, group = metadata_re_13year$Race))
anova(betadisper(bray_re_13year_dist, group = metadata_re_13year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312year, Study)
adonis2(bray_re_312year_dist ~ Race + Ethnicity, data = metadata_re_312year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_312year_dist, group = metadata_re_312year$Race))
anova(betadisper(bray_re_312year_dist, group = metadata_re_312year$Ethnicity))
```

### Pairwise comparisons
Pairwise PERMANOVAs were used to examine differences between ethnicities.

```{r pairwise_permanova_unweighted}
pairwise.adonis(unweighted_re_dist, factors = metadata_re$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_dist, factors = metadata_re$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_dist, factors = metadata_re$Study, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Host_age_cat, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_sex, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_del_route, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$Breastfeeding, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(unweighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(unweighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(unweighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
```

```{r pairwise_permanova_weighted}
pairwise.adonis(weighted_re_dist, factors = metadata_re$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_dist, factors = metadata_re$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_dist, factors = metadata_re$Study, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Host_age_cat, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_sex, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_del_route, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$Breastfeeding, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(weighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(weighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(weighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
```

## NMDS plots

NMDS plots were created for each distance matrix to visualize ethnicity-associated differences.

#### Weighted
```{r nmds_weighted}
mds_otus_weighted<-metaMDS(weighted_re_dist, k=2, trymax=499)
mds_otus_weighted_points<-mds_otus_weighted$points
mds_otus_weighted_points2<-merge(x=mds_otus_weighted_points, 
                                 y = metadata_re, 
                                 by.x = "row.names", by.y = "SampleID")
write_csv(mds_otus_weighted_points2, "weighted_mds_full.csv")
```

```{r nmds_weighted_plots}
mds_otus_weighted_points2 = read_csv("weighted_mds_full.csv",
                                     col_types = "cddccccffffffffffffdddfffffffffffffffffffd")
weight_race <- ggplot(mds_otus_weighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Race, 
                          shape = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(2.5)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.012, R<sup>2</sup> = 0.05%</b><br>
           Ethnicity: p = 0.738, R<sup>2</sup> = 0.01%<br>
           Age: p = 0.002, R<sup>2</sup> = 7.9%<br>
           Infant diet: p = 0.004, R<sup>2</sup> = 0.09%<br>
           <b>Subject: p = 0.002, R<sup>2</sup> = 33.7%", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)


mds_otus_weighted_points2_age = mds_otus_weighted_points2 %>% 
  filter(Host_age_cat != is.na(.)) %>% 
  filter(Host_age_cat != "0-1 week") %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-12 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-12 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-6 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-3 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-12 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-3 years"))
mds_otus_weighted_points2_age$Host_age_cat_new = factor(mds_otus_weighted_points2_age$Host_age_cat_new, 
                                    levels = c("1-6 weeks", "6 weeks-3 months", "3-12 months", "1-3 years", "3-12 years"))
weight_age <- ggplot(mds_otus_weighted_points2_age, 
                      aes(x = MDS1, y = MDS2)) +
  geom_point(aes(shape = Race, 
                 fill = Host_age_cat_new), size=3, color = "black") + 
    scale_shape_manual(values = c(21, 24, 22)) +
  scale_fill_manual(values = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462"),
                     guide=guide_legend(override.aes=list(shape=21))) +
  scale_color_manual(values = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462")) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.5))) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Host_age_cat_new, 
                   color = Host_age_cat_new),
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Age: p = 0.002, R<sup>2</sup> = 7.9%", 
           x = -Inf, y = Inf, size = 5,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_01week = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "0-1 week")

weight_age_01week <- ggplot(mds_otus_weighted_points2_age_01week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.022, R<sup>2</sup> = 2.6%</b><br>
           Ethnicity: p = 0.398, R<sup>2</sup> = 0.9%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_weighted_points2_age_16week = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "1-6 weeks")

weight_age_16week <- ggplot(mds_otus_weighted_points2_age_16week, 
                      aes(x = MDS1, y = MDS2, color =  Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.904, R<sup>2</sup> = 2.5%<br>
           Ethnicity: p = 0.886, R<sup>2</sup> = 0.9%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_weighted_points2_age_6week3month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "6 weeks-3 months")

weight_age_6week3month <- ggplot(mds_otus_weighted_points2_age_6week3month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.5))) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.620, R<sup>2</sup> = 2.0%<br>
           Ethnicity: p = 0.532, R<sup>2</sup> = 0.5%", 
           x = -Inf, y = Inf, size = 5,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_312month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "3-12 months")

weight_age_312month <- ggplot(mds_otus_weighted_points2_age_312month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 3.5%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 0.5%", 
           x = -Inf, y = Inf, size = 5,
           hjust = 0, vjust = 1) + guides(fill = "none", shape = "none",
                                          color = "none", linetype = "none")

mds_otus_weighted_points2_age_13year = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "1-3 years")

weight_age_13year <- ggplot(mds_otus_weighted_points2_age_13year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 2.3%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 1.6%", 
           x = -Inf, y = Inf, size = 5,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_312year = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "3-12 years")

weight_age_312year <- ggplot(mds_otus_weighted_points2_age_312year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.008, R<sup>2</sup> = 1.8%</b><br>
           Ethnicity: p = 0.094, R<sup>2</sup> = 0.7%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_weighted_points2_sex = mds_otus_weighted_points2 %>% 
  filter(host_sex != is.na(.))
weight_sex <- ggplot(mds_otus_weighted_points2_sex, 
                      aes(x = MDS1, y = MDS2, color = host_sex)) +
  geom_point(size=3) + scale_color_manual(values = c("#4daf4a", "#ff7f00")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_sex, 
                   linetype = host_sex), 
               type = "t", level = 0.95)

mds_otus_weighted_points2_del = mds_otus_weighted_points2 %>% 
  filter(host_del_route != is.na(.))
weight_del <- ggplot(mds_otus_weighted_points2_del, 
                      aes(x = MDS1, y = MDS2, color = host_del_route)) +
  geom_point(size=3) + scale_color_manual(values = c("#8da0cb", "#a6d854")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_del_route, 
                   linetype = host_del_route), 
               type = "t", level = 0.95)

mds_otus_weighted_points2_feed = mds_otus_weighted_points2 %>% 
  filter(Breastfeeding != is.na(.))
weight_diet <- ggplot(mds_otus_weighted_points2_feed, 
                      aes(x = MDS1, y = MDS2, color = Breastfeeding)) +
  geom_point(size=3) + scale_color_manual(values = c("#66c2a6", "#fc8d62", "#b3b3b3")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Breastfeeding, 
                   linetype = Breastfeeding), 
               type = "t", level = 0.95)

weight_study <- ggplot(mds_otus_weighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Study)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Dark2') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Study, 
                   linetype = Study), 
               type = "t", level = 0.95)
```

#### Unweighted
```{r nmds_unweighted}
mds_otus_unweighted<-metaMDS(unweighted_re_dist, k=2, trymax=499)
mds_otus_unweighted_points<-mds_otus_unweighted$points
mds_otus_unweighted_points2<-merge(x=mds_otus_unweighted_points, y = metadata_re, 
                                 by.x = "row.names", by.y = "SampleID")
write_csv(mds_otus_unweighted_points2, "unweighted_mds_full.csv")
```

```{r nmds_unweighted_plots}
mds_otus_unweighted_points2 = read_csv("unweighted_mds_full.csv",
                                     col_types = "cddccccffffffffffffdddfffffffffffffffffffd")
unweight_race <- ggplot(mds_otus_unweighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Race, 
                          shape = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(2.5)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.076, R<sup>2</sup> = 0.02%<br>
           <b>Ethnicity: p = 0.024, R<sup>2</sup> = 0.03%<br>
           Age: p = 0.002, R<sup>2</sup> = 5.2%</b><br>
           Infant diet: p = 0.228, R<sup>2</sup> = 0.03%<br>
           <b>Subject: p = 0.002, R<sup>2</sup> = 34.4%", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_unweighted_points2_age = mds_otus_unweighted_one_points2 %>% 
  filter(Host_age_cat != is.na(.)) %>% 
  filter(Host_age_cat != "6-12 years")
mds_otus_unweighted_points2_age$Host_age_cat = factor(mds_otus_unweighted_points2_age$Host_age_cat, 
                                    levels = c("0-1 week", "1-6 weeks", "6 weeks-3 months", "3-12 months", "1-3 years", "3-6 years"))
unweight_age <- ggplot(mds_otus_unweighted_points2_age, 
                      aes(x = MDS1, y = MDS2, color = Host_age_cat)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Set3') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Host_age_cat, 
                   linetype = Host_age_cat), 
               type = "t", level = 0.95)

mds_otus_unweighted_points2_sex = mds_otus_unweighted_one_points2 %>% 
  filter(host_sex != is.na(.))
unweight_sex <- ggplot(mds_otus_unweighted_points2_sex, 
                      aes(x = MDS1, y = MDS2, color = host_sex)) +
  geom_point(size=3) + scale_color_manual(values = c("#4daf4a", "#ff7f00")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_sex, 
                   linetype = host_sex), 
               type = "t", level = 0.95)

mds_otus_unweighted_points2_del = mds_otus_unweighted_one_points2 %>% 
  filter(host_del_route != is.na(.))
unweight_del <- ggplot(mds_otus_unweighted_points2_del, 
                      aes(x = MDS1, y = MDS2, color = host_del_route)) +
  geom_point(size=3) + scale_color_manual(values = c("#8da0cb", "#a6d854")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_del_route, 
                   linetype = host_del_route), 
               type = "t", level = 0.95)

mds_otus_unweighted_points2_feed = mds_otus_unweighted_one_points2 %>% 
  filter(Breastfeeding != is.na(.))
unweight_diet <- ggplot(mds_otus_unweighted_points2_feed, 
                      aes(x = MDS1, y = MDS2, color = Breastfeeding)) +
  geom_point(size=3) + scale_color_manual(values = c("#66c2a5", "#fc8d62", "#b3b3b3")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Breastfeeding, 
                   linetype = Breastfeeding), 
               type = "t", level = 0.95)

unweight_study <- ggplot(mds_otus_unweighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Study)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Dark2') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Study, 
                   linetype = Study), 
               type = "t", level = 0.95)
```
#### Bray
```{r nmds_bray}
mds_otus_bray<-metaMDS(bray_re_dist, k=2, trymax=499)
mds_otus_bray_points<-mds_otus_bray$points
mds_otus_bray_points2<-merge(x=mds_otus_bray_points, 
                                 y = metadata_re, 
                                 by.x = "row.names", by.y = "SampleID")
write_csv(mds_otus_bray_points2, "bray_mds_full.csv")
```

```{r nmds_bray_plots}
mds_otus_bray_points2 = read_csv("bray_mds_full.csv")
bray_race <- ggplot(mds_otus_bray_points2, 
                      aes(x = MDS1, y = MDS2, color = Race, 
                          shape = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(2.5)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.004, R<sup>2</sup> = 0.08%</b><br>
           Ethnicity: p = 0.634, R<sup>2</sup> = 0.03%<br>
           Age: p = 0.002, R<sup>2</sup> = 6.2%<br>
           Infant diet: p = 0.008, R<sup>2</sup> = 0.1%<br>
           <b>Subject: p = 0.002, R<sup>2</sup> = 36.9%", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age = mds_otus_bray_points2 %>% 
  filter(Host_age_cat != is.na(.)) %>% 
  filter(Host_age_cat != "0-1 week") %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-12 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-12 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-6 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-3 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-12 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-3 years"))
mds_otus_bray_points2_age$Host_age_cat_new = factor(mds_otus_bray_points2_age$Host_age_cat_new, 
                                    levels = c("1-6 weeks", "6 weeks-3 months", "3-12 months", "1-3 years", "3-12 years"))
bray_age <- ggplot(mds_otus_bray_points2_age, 
                      aes(x = MDS1, y = MDS2, color = Host_age_cat_new,
                          shape = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#8dd3c7", "#bebada", "#fb8072",
                                "#80b1d3", "#fdb462")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Host_age_cat_new, 
                   linetype = Host_age_cat_new), 
               type = "t", level = 0.95)

mds_otus_bray_points2_age_01week = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "0-1 week")

bray_age_01week <- ggplot(mds_otus_bray_points2_age_01week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.022, R<sup>2</sup> = 2.6%</b><br>
           Ethnicity: p = 0.398, R<sup>2</sup> = 0.9%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_16week = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "1-6 weeks")

bray_age_16week <- ggplot(mds_otus_bray_points2_age_16week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.904, R<sup>2</sup> = 2.5%<br>
           Ethnicity: p = 0.886, R<sup>2</sup> = 0.9%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_6week3month = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "6 weeks-3 months")

bray_age_6week3month <- ggplot(mds_otus_bray_points2_age_6week3month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.620, R<sup>2</sup> = 2.0%<br>
           Ethnicity: p = 0.532, R<sup>2</sup> = 0.5%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_312month = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "3-12 months")

bray_age_312month <- ggplot(mds_otus_bray_points2_age_312month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 3.5%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 0.5%</><br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_13year = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "1-3 years")

bray_age_13year <- ggplot(mds_otus_bray_points2_age_13year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 2.3%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 1.6%</b><br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_312year = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "3-12 years")

bray_age_312year <- ggplot(mds_otus_bray_points2_age_312year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.008, R<sup>2</sup> = 1.8%</b><br>
           Ethnicity: p = 0.094, R<sup>2</sup> = 0.7%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_sex = mds_otus_bray_one_points2 %>% 
  filter(host_sex != is.na(.))
bray_sex <- ggplot(mds_otus_bray_points2_sex, 
                      aes(x = MDS1, y = MDS2, color = host_sex)) +
  geom_point(size=3) + scale_color_manual(values = c("#4daf4a", "#ff7f00")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("bray UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_sex, 
                   linetype = host_sex), 
               type = "t", level = 0.95)

mds_otus_bray_points2_del = mds_otus_bray_one_points2 %>% 
  filter(host_del_route != is.na(.))
bray_del <- ggplot(mds_otus_bray_points2_del, 
                      aes(x = MDS1, y = MDS2, color = host_del_route)) +
  geom_point(size=3) + scale_color_manual(values = c("#8da0cb", "#a6d854")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_del_route, 
                   linetype = host_del_route), 
               type = "t", level = 0.95)

mds_otus_bray_points2_feed = mds_otus_bray_one_points2 %>% 
  filter(Breastfeeding != is.na(.))
bray_diet <- ggplot(mds_otus_bray_points2_feed, 
                      aes(x = MDS1, y = MDS2, color = Breastfeeding)) +
  geom_point(size=3) + scale_color_manual(values = c("#66c2a6", "#fc8d62", "#b3b3b3")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("bray UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Breastfeeding, 
                   linetype = Breastfeeding), 
               type = "t", level = 0.95)

bray_study <- ggplot(mds_otus_bray_points2, 
                      aes(x = MDS1, y = MDS2, color = Study)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Dark2') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Study, 
                   linetype = Study), 
               type = "t", level = 0.95)
```

#### Combo
```{r nmds_race_combo}
tiff(file="nmds_plot_race.tif", res=150, width=18, height=9, units="in")
legend1 = get_legend(weight_race + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_race + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_race + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_race_one_combo}
tiff(file="nmds_plot_race_one.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_race_one + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_race_one + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_race_one + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_diet_combo}
tiff(file="nmds_plot_diet.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_diet + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_diet + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_diet + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_age_combo}
tiff(file="nmds_plot_age.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_age + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_age + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_age + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_sex_combo}
tiff(file="nmds_plot_sex.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_sex + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_sex + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_sex + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_del_combo}
tiff(file="nmds_plot_del.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_del + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_del + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_del + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_study_combo}
tiff(file="nmds_plot_study.tif", res=150, width=18, height=9, units="in")
legend1 = get_legend(weight_study + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_study + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(bray_study + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

##Alpha diversity
###All samples
```{r alpha_div}
faith = read.table("alpha_out/faith_pd.tsv", header=T)
otus = read.table("alpha_out/observed_features.tsv", header = T)
shannon = read.table("alpha_out/shannon.tsv", header = T)

faith2 = faith %>% rownames_to_column("SampleID")
otus2 = otus %>% rownames_to_column("SampleID")
shannon2 = shannon %>% rownames_to_column("SampleID")

alpha_all = inner_join(metadata_re, faith2, by = "SampleID") %>% inner_join(otus2, by = "SampleID") %>% inner_join(shannon2, by = "SampleID") %>% 
  filter(Race != "Other") %>% 
  filter(Ethnicity != "Other")

alpha_one = inner_join(metadata_re_onesubj, faith2, by = "SampleID") %>% inner_join(otus2, by = "SampleID") %>% inner_join(shannon2, by = "SampleID") %>% 
  filter(Race != "Other") %>% 
  filter(Ethnicity != "Other")

f = lmer(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(f)
Anova(f)
summary(glht(f,linfct=mcp(Ethnicity="Tukey")))
summary(glht(f,linfct=mcp(Host_age_cat="Tukey")))

f1 = lmer(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_one, na.action = na.omit)
summary(f1)
Anova(f1)
summary(glht(f1,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(f1,linfct=mcp(Breastfeeding="Tukey")))

o = lmer(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(o)
Anova(o)
summary(glht(o,linfct=mcp(Race="Tukey")))
summary(glht(o,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(o,linfct=mcp(host_sex="Tukey")))

o1 = lmer(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_one, na.action = na.omit)
summary(o1)
Anova(o1)
summary(glht(o1,linfct=mcp(Race="Tukey")))
summary(glht(o1,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(o1,linfct=mcp(Breastfeeding="Tukey")))

s = lmer(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(s)
Anova(s)
summary(glht(s,linfct=mcp(Race="Tukey")))
summary(glht(s,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(s,linfct=mcp(Breastfeeding="Tukey")))

s1 = lmer(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_one, na.action = na.omit)
summary(s1)
Anova(s1)
summary(glht(s1,linfct=mcp(Race="Tukey")))
summary(glht(s1,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(s1,linfct=mcp(Breastfeeding="Tukey")))
```

#### Graphs
Each comparison of interest was graphed, using the full dataset.
```{r alpha_div_race_graphs}
plot1_alpha = ggboxplot(alpha_all, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1_alpha = ggpar(plot1_alpha, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.082", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_alpha = ggboxplot(alpha_all, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_alpha = ggpar(plot2_alpha, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("***", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.002", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_alpha = ggboxplot(alpha_all, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_alpha = ggpar(plot3_alpha, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.008", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_race.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1_alpha, plot2_alpha, plot3_alpha, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_eth_graphs}
plot1_alphae = ggboxplot(alpha_all, x = "Ethnicity", 
                  y = "faith_pd", color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1_alphae = ggpar(plot1_alphae, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("*", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.031", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_alphae = ggboxplot(alpha_all, x = "Ethnicity", 
                  y = "observed_features", color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_alphae = ggpar(plot2_alphae, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.092", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_alphae = ggboxplot(alpha_all, x = "Ethnicity", 
                  y = "shannon_entropy", color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_alphae = ggpar(plot3_alphae, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.283", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_ethnicity.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1_alphae, plot2_alphae, plot3_alphae, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_age_graphs}
alpha_all_age = alpha_all %>% filter(Host_age_cat != is.na(.)) 

alpha_all_age$Host_age_cat = factor(alpha_all_age$Host_age_cat, 
                                    levels = c("0-1 week", "1-6 weeks", "6 weeks-3 months", "3-12 months", "1-3 years", "3-6 years", "6-12 years"))

plot1 = ggboxplot(alpha_all_age, x = "Host_age_cat", 
                  y = "faith_pd", color = "Host_age_cat", 
                  palette = "Set3", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("0-1 week", "1-3 years"),
                                        c("0-1 week", "3-6 years"),
                                        c("0-1 week", "6-12 years"),
                                        c("1-6 weeks", 
                                          "3-12 months"),
                                        c("1-6 weeks", "1-3 years"),
                                        c("1-6 weeks", "3-6 years"),
                                        c("1-6 weeks", "6-12 years"),
                                        c("6 weeks-3 months", 
                                          "3-12 months"),
                                        c("6 weeks-3 months",
                                          "1-3 years"),
                                        c("6 weeks-3 months", 
                                          "3-6 years"),
                                        c("6 weeks-3 months", 
                                          "6-12 years"),
                                        c("3-12 months","1-3 years"),
                                        c("3-12 months","3-6 years"),
                                        c("3-12 months", 
                                          "6-12 years")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0000000001, 
                                            0.01, 0.05, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Age: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_age, x = "Host_age_cat", 
                  y = "observed_features", color = "Host_age_cat", 
                  palette = "Set3", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("0-1 week", "1-6 weeks"),
                                        c("0-1 week", 
                                          "6 weeks-3 months"),
                                        c("0-1 week", "1-3 years"),
                                        c("0-1 week", "3-6 years"),
                                        c("0-1 week", "6-12 years"),
                                        c("1-6 weeks", 
                                          "3-12 months"),
                                        c("1-6 weeks", "1-3 years"),
                                        c("1-6 weeks", "3-6 years"),
                                        c("1-6 weeks", "6-12 years"),
                                        c("6 weeks-3 months", 
                                          "3-12 months"),
                                        c("6 weeks-3 months",
                                          "1-3 years"),
                                        c("6 weeks-3 months", 
                                          "3-6 years"),
                                        c("6 weeks-3 months", 
                                          "6-12 years"),
                                        c("3-12 months","1-3 years"),
                                        c("3-12 months","3-6 years"),
                                        c("3-12 months", 
                                          "6-12 years")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.05, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Age: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_age, x = "Host_age_cat", 
                  y = "shannon_entropy", color = "Host_age_cat", 
                  palette = "Set3", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("0-1 week", "1-6 weeks"),
                                        c("0-1 week", "3-12 months"),
                                        c("0-1 week", "1-3 years"),
                                        c("0-1 week", "3-6 years"),
                                        c("0-1 week", "6-12 years"),
                                        c("1-6 weeks", 
                                          "6 weeks-3 months"),
                                        c("1-6 weeks", 
                                          "3-12 months"),
                                        c("1-6 weeks", "1-3 years"),
                                        c("1-6 weeks", "3-6 years"),
                                        c("1-6 weeks", "6-12 years"),
                                        c("6 weeks-3 months", 
                                          "3-12 months"),
                                        c("6 weeks-3 months",
                                          "1-3 years"),
                                        c("6 weeks-3 months", 
                                          "3-6 years"),
                                        c("6 weeks-3 months",
                                          "6-12 years"),
                                        c("3-12 months","1-3 years"),
                                        c("3-12 months",
                                          "3-6 years")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0000000001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Age: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_age.tif", 
     res=300, width=15, height=6, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_sex_graphs}
alpha_all_sex = alpha_all %>% dplyr::filter(host_sex != is.na(.))

plot1 = ggboxplot(alpha_all_sex, x = "host_sex", 
                  y = "faith_pd", color = "host_sex", 
                  palette = c("#4daf4a", "#ff7f00"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Sex: p = 0.091", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_sex, x = "host_sex", 
                  y = "observed_features", color = "host_sex", 
                  palette = c("#4daf4a", "#ff7f00"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Female", "Male")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0001, 0.001, 
                                            0.05, 0.999, 1),
                                        symbols = c("*", "*", "*", 
                                                    "*", "ns")))  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Sex: p = 0.033", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_sex, x = "host_sex", 
                  y = "shannon_entropy", color = "host_sex", 
                  palette = c("#4daf4a", "#ff7f00"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Sex: p = 0.593", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_sex.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_del_graphs}
alpha_all_del = alpha_all %>% dplyr::filter(host_del_route != is.na(.))

plot1 = ggboxplot(alpha_all_del, x = "host_del_route", 
                  y = "faith_pd", color = "host_del_route", 
                  palette = c("#8da0cb", "#a6d854"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Delivery: p = 0.543", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_del, x = "host_del_route", 
                  y = "observed_features", color = "host_del_route", 
                  palette = c("#8da0cb", "#a6d854"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Delivery: p = 0.559", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_del, x = "host_del_route", 
                  y = "shannon_entropy", color = "host_del_route", 
                  palette = c("#8da0cb", "#a6d854"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Delivery: p = 0.979", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_del.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_feed_graphs}
alpha_all_fed = alpha_all %>% dplyr::filter(Breastfeeding != is.na(.))

plot1 = ggboxplot(alpha_all_fed, x = "Breastfeeding", 
                  y = "faith_pd", color = "Breastfeeding", 
                  palette = c("#66c2a6", "#fc8d62", "#b3b3b3"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Infant diet: p = 0.091", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_fed, x = "Breastfeeding", 
                  y = "observed_features", color = "Breastfeeding", 
                  palette = c("#66c2a6", "#fc8d62", "#b3b3b3"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Infant diet: p = 0.066", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_fed, x = "Breastfeeding", 
                  y = "shannon_entropy", color = "Breastfeeding", 
                  palette = c("#66c2a6", "#fc8d62", "#b3b3b3"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Formula fed", 
                                          "Mixed feeding"), 
                                        c("Formula fed", 
                                          "Exclusively breastfed"), 
                                        c("Mixed feeding", 
                                          "Exclusively breastfed")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0001, 0.001, 
                                            0.05, 0.999, 1),
                                        symbols = c("***", 
                                                    "*", "*", 
                                                    "*", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Infant diet: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_feed.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```
###By age category
```{r alpha_agecat}
alpha_zero_one = alpha_all %>% 
  filter(Host_age_cat == "0-1 week")
alpha_one_six = alpha_all %>% 
  filter(Host_age_cat == "1-6 weeks")
alpha_six_three = alpha_all %>% 
  filter(Host_age_cat == "6 weeks-3 months")
alpha_three_twelve = alpha_all %>% 
  filter(Host_age_cat == "3-12 months")
alpha_one_three = alpha_all %>% 
  filter(Host_age_cat == "1-3 years")
alpha_threey_twelvey = alpha_all %>% 
  filter(Host_age_cat == "3-6 years" | 
           Host_age_cat == "6-12 years")

#0-1 weeks
f01 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(f01)
Anova(f01)

o01 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(o01)
Anova(o01)

s01 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(s01)
Anova(s01)

#1-6 weeks
f16 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(f16)
Anova(f16)
summary(glht(f16,linfct=mcp(Breastfeeding="Tukey")))

o16 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(o16)
Anova(o16)

s16 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(s16)
Anova(s16)
summary(glht(s16,linfct=mcp(Breastfeeding="Tukey")))

#6 weeks - 3 months
f63 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(f63)
Anova(f63)

o63 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(o63)
Anova(o63)

s63 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(s63)
Anova(s63)
summary(glht(s63,linfct=mcp(Breastfeeding="Tukey")))

#3-12 months
f312 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(f312)
Anova(f312)
summary(glht(f312,linfct=mcp(Breastfeeding="Tukey")))

o312 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(o312)
Anova(o312)
summary(glht(o312,linfct=mcp(Race="Tukey")))
summary(glht(o312,linfct=mcp(Breastfeeding="Tukey")))

s312 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(s312)
Anova(s312)
summary(glht(s312,linfct=mcp(Race="Tukey")))
summary(glht(s312,linfct=mcp(Breastfeeding="Tukey")))

#1-3 years
f13 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(f13)
Anova(f13)
summary(glht(f13,linfct=mcp(Race="Tukey")))
summary(glht(f13,linfct=mcp(Ethnicity="Tukey")))


o13 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(o13)
Anova(o13)
summary(glht(o13,linfct=mcp(Race="Tukey")))
summary(glht(o13,linfct=mcp(Ethnicity="Tukey")))

s13 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(s13)
Anova(s13)
summary(glht(s13,linfct=mcp(Race="Tukey")))
summary(glht(s13,linfct=mcp(Ethnicity="Tukey")))

#3-12 years
f312y = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(f312y)
Anova(f312y)

o312y = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(o312y)
Anova(o312y)

s312y = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(s312y)
Anova(s312y)

#0 weeks-3 months
alpha_zero_three = alpha_all %>% 
  filter(Host_age_cat == "0-1 week" | 
           Host_age_cat == "1-6 weeks" | 
           Host_age_cat == "6 weeks-3 months")

o03 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_zero_three, na.action = na.omit)
summary(o03)
Anova(o03)

s03 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_zero_three, na.action = na.omit)
summary(s03)
Anova(s03)
```

####Graphs
```{r alpha_div_agerace_graphs}
alpha_all_age = alpha_all %>% filter(Host_age_cat != is.na(.)) %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-12 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-12 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-6 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-3 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-12 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-3 years"))

alpha_all_age_01week = alpha_all_age %>% 
  filter(Host_age_cat_new == "0-1 week")

alpha_all_age_16week = alpha_all_age %>% 
  filter(Host_age_cat_new == "1-6 weeks")

alpha_all_age_6week3month = alpha_all_age %>% 
  filter(Host_age_cat_new == "6 weeks-3 months")

alpha_all_age_312month = alpha_all_age %>% 
  filter(Host_age_cat_new == "3-12 months")

alpha_all_age_13year = alpha_all_age %>% 
  filter(Host_age_cat_new == "1-3 years")

alpha_all_age_312year = alpha_all_age %>% 
  filter(Host_age_cat_new == "3-12 years")

alpha_16week = ggplot(alpha_all_age_16week,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#8dd3c7",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#8dd3c7",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylab("All Age Categories\nShannon Diversity") + 
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#bebada",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#bebada",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")
  
alpha_312month = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#fb8072",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#fb8072",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_13year = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#80b1d3",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#80b1d3",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_312year = ggplot(alpha_all_age_312year,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#fdb462",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#fdb462",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month_race = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#ffd92f", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#ffd92f", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("6 weeks - 3 months\nShannon Diversity") +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")
  
alpha_312month_race = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#ffd92f", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#ffd92f", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black") +
  ylab("3 - 12 months\nShannon Diversity") +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_13year_race = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#ffd92f", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#ffd92f", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("1 - 3 years\nShannon Diversity") +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_6week3month_eth = ggplot(alpha_all_age_6week3month,
                      aes(x = Ethnicity, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("light gray", "dark gray"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("light gray", "dark gray"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Ethnicity),
                  color = "black", fill = "black") + 
  scale_shape_manual(values = c(23, 25)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")
  
alpha_312month_eth = ggplot(alpha_all_age_312month,
                      aes(x = Ethnicity, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("light gray", "dark gray"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("light gray", "dark gray"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Ethnicity),
                  color = "black", fill = "black") + 
  scale_shape_manual(values = c(23, 25)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") 

alpha_13year_eth = ggplot(alpha_all_age_13year,
                      aes(x = Ethnicity, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("light gray", "dark gray"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("light gray", "dark gray"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Ethnicity),
                  color = "black", fill = "black") + 
  scale_shape_manual(values = c(23, 25)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", "Non Hispanic")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

tiff(file="alpha_combined_age_race.tif", 
     res=300, width=15, height=6, units="in")
alpha_age_race_plot = ggarrange(alpha_16week, alpha_6week3month, alpha_312month,
          alpha_13year, alpha_312year, nrow = 1, ncol = 5, 
          align = "v")
alpha_age_race_plot
dev.off()
```

##Different abundance of taxa
ANCOM-BC (a bias correcting version of analysis of microbiome compositions) was used to identify differentially abundant taxa at the level of phyla, family, and ASV.

###Phyla
```{r phyla_abund}
phyla = read_tsv("feature-table-phyla.tsv")
taxonomy = phyla %>% dplyr::select(Phyla)

phyla_matrix = phyla %>% column_to_rownames("Phyla") %>% as.matrix()
phyla_phylo = otu_table(phyla_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

phyla_p = phyloseq(phyla_phylo, meta_phylo)

#Race
race = ancombc(phyla_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Phyla") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_phyla.csv")

#Ethnicity
ethnicity = ancombc(phyla_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e = ethnicity$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Phyla") %>% 
  inner_join(taxonomy)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_phyla.csv")

#Age
age = ancombc(phyla_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Phyla") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_phyla.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
phyla_p_ab = phyloseq(phyla_phylo, meta_ab)
race_ab = ancombc(phyla_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_phyla_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
phyla_p_aw = phyloseq(phyla_phylo, meta_aw)
race_aw = ancombc(phyla_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_phyla_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
phyla_p_bw = phyloseq(phyla_phylo, meta_bw)
race_bw = ancombc(phyla_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_phyla_bw.csv")

```
####Phyla graphs
```{r phyla_graphs_race}
phyla_ra = phyla %>% column_to_rownames("Phyla") %>% 
  decostand(method = "total", MARGIN = 2)
phyla_meta = phyla_ra %>% 
  rownames_to_column("Phyla") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Phyla", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

bact_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes", color = "Race",
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Bacteroidetes") 
bact_race = ggpar(bact_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "White"), 
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.05, 1),
                                        symbols = c("***", "**"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

prot_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria", color = "Race", 
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Proteobacteria") 
prot_race = ggpar(prot_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "White"), 
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.05, 
                                                      1),
                                        symbols = c("***", "***"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tene_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Tenericutes", color = "Race", 
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Tenericutes") 
tene_race = ggpar(tene_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "Black")),
                     symnum.args = list(cutpoints = c(0, 0.05, 1),
                                        symbols = c("***", "***"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.003", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

verr_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Verrucomicrobia", color = "Race", 
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Verrucomicrobia") 
verr_race = ggpar(verr_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "Black")),
                     symnum.args = list(cutpoints = c(0, 0.05,1),
                                        symbols = c("*", "*"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.002", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="phyla_diffs_race.tif", 
     res=300, width=12, height=9, units="in")
ggarrange(bact, prot, tene, verr, nrow = 2, ncol = 2, 
          common.legend = T, align = "hv", legend = "right",
          labels = c("A", "B", "C", "D"))
dev.off()
```

```{r phyla_graphs_ethnicity}
phyla_ra = phyla %>% column_to_rownames("Phyla") %>% 
  decostand(method = "total", MARGIN = 2)
phyla_meta = phyla_ra %>% 
  rownames_to_column("Phyla") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Phyla", values_from = "rel_abund") %>% 
  inner_join(metadata_re)


prot_eth = ggboxplot(phyla_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Proteobacteria", 
                 color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Proteobacteria") 
prot_eth = ggpar(prot_eth, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", 
                                          "Hispanic")),
                     symnum.args = list(cutpoints = c(0, 
                                                      0.5, 1),
                                        symbols = c("***", "***"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tene_eth = ggboxplot(phyla_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Tenericutes", 
                 color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Tenericutes") 
tene_eth = ggpar(tene_eth, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.5, 1),
                                        symbols = c("*", "*"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.024", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

acti_eth = ggboxplot(phyla_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Actinobacteria", 
                 color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Actinobacteria") 
acti_eth = ggpar(acti_eth, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic",
                                          "Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.5, 1),
                                        symbols = c("**", "**")))  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.004", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="phyla_diffs_ethnicity.tif", 
     res=300, width=15, height=5, units="in")
ggarrange(acti, prot, tene, nrow = 1, ncol = 3, 
          common.legend = T, align = "v", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

###Families
```{r families_abund}
family = read_tsv("feature-table-family.tsv")
taxonomy = family %>% dplyr::select(Family)

family_matrix = family %>% column_to_rownames("Family") %>% 
  as.matrix()
family_phylo = otu_table(family_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

family_p = phyloseq(family_phylo, meta_phylo)

#Race
race = ancombc(family_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Family") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_family.csv")

#Ethnicity
ethnicity = ancombc(family_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_family.csv")

#Age
age = ancombc(family_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Family") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_family.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
family_p_ab = phyloseq(family_phylo, meta_ab)
race_ab = ancombc(family_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_family_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
family_p_aw = phyloseq(family_phylo, meta_aw)
race_aw = ancombc(family_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_family_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
family_p_bw = phyloseq(family_phylo, meta_bw)
race_bw = ancombc(family_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_family_bw.csv")
```

```{r family_graphs_race}
family_ra = family %>% column_to_rownames("Family") %>% 
  decostand(method = "total", MARGIN = 2)
family_meta = family_ra %>% 
  rownames_to_column("Family") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Family", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

prev = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Prevotellaceae") 
prev = ggpar(prev, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

porp = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Porphyromonadaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Porphyromonadaceae") 
porp = ggpar(porp, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 
                                                      0.001, 0.05, 
                                                      0.5, 1),
                                        symbols = c("*", "*", "*", 
                                                    "*", "*")))

lact = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Lactobacillaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Lactobacillaceae") 
lact = ggpar(lact, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

camp = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Epsilonproteobacteria;o__Campylobacterales;f__Campylobacteraceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Campylobacteraceae") 
camp = ggpar(camp, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Black", "Other"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

veil = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Veillonellaceae") 
veil = ggpar(veil, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

pept = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Peptostreptococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Peptostreptococcaceae") 
pept = ggpar(pept, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

tiss = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__[Tissierellaceae]", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Tissierellaceae") 
tiss = ggpar(tiss, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

ente = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Enterobacteriaceae") 
ente = ggpar(ente, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

clos = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Clostridiaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Clostridiaceae") 
clos = ggpar(clos, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Other"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

fuso = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Fusobacteria;c__Fusobacteriia;o__Fusobacteriales;f__Fusobacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Fusobacteriaceae") 
fuso = ggpar(fuso, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

stap = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Bacillales;f__Staphylococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Staphylococcaceae") 
stap = ggpar(stap, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

geme = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Gemellales;f__Gemellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Gemellaceae") 
geme = ggpar(geme, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

mogi = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__[Mogibacteriaceae]", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Mogibacteriaceae") 
mogi = ggpar(mogi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Other"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

mora = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pseudomonadales;f__Moraxellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Moraxellaceae") 
mora = ggpar(mora, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Other"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

cory = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Actinobacteria;c__Actinobacteria;o__Actinomycetales;f__Corynebacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Corynebacteriaceae") 
cory = ggpar(cory, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

entec = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Enterococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Enterococcaceae") 
entec = ggpar(entec, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

past = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Pasteurellaceae") 
past = ggpar(past, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

para = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__[Paraprevotellaceae]", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Paraprevotellaceae") 
para = ggpar(para, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

micr = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Actinobacteria;c__Actinobacteria;o__Actinomycetales;f__Micrococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Micrococcaceae") 
micr = ggpar(micr, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

euba = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Eubacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Eubacteriaceae") 
euba = ggpar(euba, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "White"),
                                        c("Black", "Other"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

tiff(file="family_diffs_race.tif", 
     res=300, width=22, height=15, units="in")
ggarrange(prev, porp, lact, camp, veil, pept, tiss, ente, clos, fuso, stap, 
          geme, mogi, mora, cory, entec, past, para, micr, euba, 
          nrow = 4, ncol = 5, 
          common.legend = T, align = "hv", legend = "right")
dev.off()
```

```{r family_graphs_ethn}
family_ra = family %>% column_to_rownames("Family") %>% 
  decostand(method = "total", MARGIN = 2)
family_meta = family_ra %>% 
  rownames_to_column("Family") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Family", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

lach = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Lachnospiraceae") 
lach = ggpar(lach, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

bact = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Bacteroidaceae") 
bact = ggpar(bact, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic",
                                          "Other"), 
                                        c("Hispanic", "Other")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 
                                                      0.001, 0.05, 
                                                      0.5, 1),
                                        symbols = c("*", "*", "*", 
                                                    "*", "*")))

prev = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Prevotellaceae") 
prev = ggpar(prev, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", "Other"),
                                        c("Hispanic", "Other"),
                                        c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

barn = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__[Barnesiellaceae]", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Barnesiellaceae") 
barn = ggpar(barn, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

ente = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Enterobacteriaceae") 
ente = ggpar(ente, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") 

rike = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Rikenellaceae") 
rike = ggpar(rike, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", "Other"),
                                        c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

stre = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Streptococcaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Streptococcaceae") 
stre = ggpar(stre, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

geme = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Gemellales;f__Gemellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Gemellaceae") 
geme = ggpar(geme, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

chri = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Christensenellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Christensenellaceae") 
chri = ggpar(chri, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") 

pseu = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pseudomonadales;f__Pseudomonadaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Pseudomonadaceae") 
pseu = ggpar(pseu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

verr = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Verrucomicrobia;c__Verrucomicrobiae;o__Verrucomicrobiales;f__Verrucomicrobiaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Verrucomicrobiaceae") 
verr = ggpar(verr, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

turi = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Turicibacterales;f__Turicibacteraceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Turicibacteraceae") 
turi = ggpar(turi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

euba = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Eubacteriaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Eubacteriaceae") 
euba = ggpar(euba, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", "Other"),
                                        c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))


tiff(file="family_diffs_ethn.tif", 
     res=300, width=22, height=12, units="in")
ggarrange(lach, bact, prev, barn, ente, rike, stre, geme, 
          chri, pseu, verr, turi, euba, 
          nrow = 3, ncol = 5, 
          common.legend = T, align = "hv", legend = "right")
dev.off()
```

```{r family_graphs_race_heritable}
tiff(file="family_diffs_race_heritable.tif", 
     res=300, width=12, height=4, units="in")
ggarrange(clos, past, pept, 
          nrow = 1, ncol = 3, 
          common.legend = T, align = "v", legend = "right")
dev.off()
```

```{r family_graphs_ethn_heritable}
tiff(file="family_diffs_ethn_heritable.tif", 
     res=300, width=9, height=4, units="in")
ggarrange(lach, chri, 
          nrow = 1, ncol = 2, 
          common.legend = T, align = "v", legend = "right")
dev.off()
```

###Genera
```{r genus_abund}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus.csv")

#Age
age = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_genus.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw.csv")
```

###Species
```{r species_abund}
species = read_tsv("feature-table-species.tsv")
taxonomy = species %>% dplyr::select(Species)

species_matrix = species %>% column_to_rownames("Species") %>% as.matrix()
species_phylo = otu_table(species_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

species_p = phyloseq(species_phylo, meta_phylo)

#Race
race = ancombc(species_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Species") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_species.csv")

#Ethnicity
ethnicity = ancombc(species_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_species.csv")

#Age
age = ancombc(species_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Species") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_species.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
species_p_ab = phyloseq(species_phylo, meta_ab)
race_ab = ancombc(species_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_species_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
species_p_aw = phyloseq(species_phylo, meta_aw)
race_aw = ancombc(species_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_species_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
species_p_bw = phyloseq(species_phylo, meta_bw)
race_bw = ancombc(species_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_species_bw.csv")
```

###ASVs
```{r ASV_abund}
taxonomy = asvs %>% dplyr::select(ASV, taxonomy)

ASV_matrix = asvs %>% 
  dplyr::select(-taxonomy) %>% 
  column_to_rownames("ASV") %>% 
  as.matrix()
ASV_phylo = otu_table(ASV_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

ASV_p = phyloseq(ASV_phylo, meta_phylo)

#Race
race = ancombc(ASV_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr",
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "taxonomy") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_ASV.csv")

#Ethnicity
ethnicity = ancombc(ASV_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr",
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_ASV.csv")

#Age
age = ancombc(ASV_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr",
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "ASV") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_ASV.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
ASV_p_ab = phyloseq(ASV_phylo, meta_ab)
race_ab = ancombc(ASV_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_ASV_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
ASV_p_aw = phyloseq(ASV_phylo, meta_aw)
race_aw = ancombc(ASV_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_ASV_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
ASV_p_bw = phyloseq(ASV_phylo, meta_bw)
race_bw = ancombc(ASV_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_ASV_bw.csv")
```

###Genera by study
```{r study_genera_kim}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)
meta_phylo_kim = metadata_re_kim %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_kim)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_kim.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_kim %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_kim.csv")

##Asian vs. White
meta_aw = metadata_re_kim %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_kim.csv")

##Black vs. White
meta_bw = metadata_re_kim %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_kim.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_kim.csv")
```

```{r study_genera_herman}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_herman = metadata_re_herman %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_herman)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_herman.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_herman %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_herman.csv")

##Asian vs. White
meta_aw = metadata_re_herman %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_herman.csv")

##Black vs. White
meta_bw = metadata_re_herman %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_herman.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_herman.csv")
```

```{r study_genera_chu}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_chu = metadata_re_chu %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()


genus_p = phyloseq(genus_phylo, meta_phylo_chu)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_chu.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_chu %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_chu.csv")

##Asian vs. White
meta_aw = metadata_re_chu %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_chu.csv")

##Black vs. White
meta_bw = metadata_re_chu %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_chu.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_chu.csv")
```

```{r study_genera_robinson}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_robinson = metadata_re_robinson %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_robinson)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_robinson.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_robinson %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_robinson.csv")

##Asian vs. White
meta_aw = metadata_re_robinson %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_robinson.csv")

##Black vs. White
meta_bw = metadata_re_robinson %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_robinson.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_robinson.csv")
```

```{r study_genera_planer}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_planer = metadata_re_planer %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_planer)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r_df = data.frame(
  Species = row.names(race$res$beta),
  beta = unlist(race$res$beta),
  se = unlist(race$res$se),
  W = unlist(race$res$W),
  p_val = unlist(race$res$p_val),
  q_val = unlist(race$res$q_val),
  diff_abn = unlist(race$res$diff_abn))

fdr_ancom_r <- res_r_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r, "Diff_abund_global_race_genus_planer.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_planer.csv")
```

```{r study_genera_levin}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_levin = metadata_re_levin %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()


genus_p = phyloseq(genus_phylo, meta_phylo_levin)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r_df = data.frame(
  Species = row.names(race$res$beta),
  beta = unlist(race$res$beta),
  se = unlist(race$res$se),
  W = unlist(race$res$W),
  p_val = unlist(race$res$p_val),
  q_val = unlist(race$res$q_val),
  diff_abn = unlist(race$res$diff_abn))

fdr_ancom_r <- res_r_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r, "Diff_abund_global_race_genus_levin.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_levin.csv")
```

```{r study_genera_db}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_db = metadata_re_db %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()


genus_p = phyloseq(genus_phylo, meta_phylo_db)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_db.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_db %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_db.csv")

##Asian vs. White
meta_aw = metadata_re_db %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_db.csv")

##Black vs. White
meta_bw = metadata_re_db %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_db.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_db.csv")
```

```{r study_genera_cioffi}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_cioffi = metadata_re_cioffi %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_cioffi)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r_df = data.frame(
  Species = row.names(race$res$beta),
  beta = unlist(race$res$beta),
  se = unlist(race$res$se),
  W = unlist(race$res$W),
  p_val = unlist(race$res$p_val),
  q_val = unlist(race$res$q_val),
  diff_abn = unlist(race$res$diff_abn))

fdr_ancom_r <- res_r_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r, "Diff_abund_global_race_genus_cioffi.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_cioffi.csv")
```

##Venn diagram
```{r venn}
venn = read_csv("venn_input.csv")
venn_c = venn %>% filter(Children != is.na(.))
venn_a = venn %>% filter(Adults != is.na(.))
c = venn_c$Children
a = venn_a$Adults

venn.diagram(x = list(c, a), 
             category.names = c("Children (57)", "Adults (226)"), 
             filename = "Venn_BW.png", output = T, imagetype = "png", 
             height = 1000, width = 1000, resolution = 300, 
             col = c("#4b265c", "#25a165"), lwd = 1,
             fill = c(alpha("#4b265c", 0.3), alpha("#25a165", 0.3)), 
             fontfamily = "sans", cat.fontfamily = "sans",
             cat.default.pos = "outer", cat.dist = c(0.1, 0.1),
             margin = 0.2)
```

##ROC curves
Random Forest classification was performed in QIIME and the predictions/probabilities were exported. (pROC)
```{r roc_curve}
pred = read_tsv("merged-table-filtered-race-classifier/predictions.tsv",
                col_types = "cf")
prob = read_tsv("merged-table-filtered-race-classifier/probabilities.tsv")

rf = metadata %>% inner_join(pred) %>% 
  inner_join(prob)

response_black = rf %>% dplyr::select(Race, Black) %>% 
  mutate(Race_binary = if_else(Race == "Black", 1, 0))
response_white = rf %>% dplyr::select(Race, White) %>% 
  mutate(Race_binary = if_else(Race == "White", 1, 0))
response_asian = rf %>% dplyr::select(Race, `Asian/Pacific Islander`) %>% 
  mutate(Race_binary = if_else(Race == "Asian/Pacific Islander", 1, 0))

White = roc(response_white$Race_binary ~ response_white$White)
Black = roc(response_black$Race_binary ~ response_black$Black)
Asian = roc(response_asian$Race_binary ~ response_asian$`Asian/Pacific Islander`)
roc_plot = ggroc(list(White, Black, Asian), legacy.axes = TRUE, lwd = 2) + xlab("False Positive Rate") + ylab("True Positive Rate") + scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(2.5)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Random Forest ROC") + annotate(geom = "richtext", 
                                          fill = NA, label.color = NA,
           label = "White, AUC = 0.931<br>
           Black, AUC = 0.965<br>
           Asian/Pacific Islander, AUC = 0.930", 
           x = Inf, y = -Inf, size = 5,
           hjust = 1, vjust = 0)

plot(White, col="#377eb8", lwd = 4, main = "Random Forest ROC", 
     legacy.axes = TRUE, xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Black, col = "#e41a1c", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Asian, col = "#ffd92f", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
legend("bottomright", 
       legend = c("White, AUC = 0.931", "Black, AUC = 0.965", "Asian/Pacific Islander, AUC = 0.930"), 
       col = c("#377eb8", "#e41a1c", "#ffd92f"), lwd = 4)
```

##Graphs for publication
```{r pub_graphs}
tiff(file="fig1.tif", res=300, width=20, height=20, units="in")
row1_left = alpha_age_race_plot
row2_left = plot_grid(alpha_6week3month_race, alpha_6week3month_eth,
                 nrow = 1, ncol = 2, 
          rel_widths = c(1, 1), 
          axis = "tb", align = "hv")
row3_left = plot_grid(alpha_312month_race, alpha_312month_eth,
                 nrow = 1, ncol = 2, 
          rel_widths = c(1, 1), 
          axis = "tb", align = "hv")
row4_left = plot_grid(alpha_13year_race, alpha_13year_eth,
                 nrow = 1, ncol = 2, 
          rel_widths = c(1, 1), 
          axis = "tb", align = "hv")
left = plot_grid(row1_left, row2_left, row3_left, row4_left,
                 nrow = 4, ncol = 1,
                 axis = "l", align = "hv",
                 labels = c("A", "B", "C", "D"))
right = plot_grid(weight_age, weight_age_6week3month,
                  weight_age_312month + theme(legend.position = "none"), 
                  weight_age_13year + theme(legend.position = "none"),
                  nrow = 4, ncol = 1,
                  axis = "lr", align = "hv")
plot = plot_grid(left, right,
                 nrow = 1, ncol = 2, 
          axis = "tb", align = "hv")
plot
dev.off()

setEPS()
postscript(file="fig1.eps", width=20, height=20, paper = "special")
plot
dev.off()

venn.diagram(x = list(c, a), 
             category.names = c("Children (57)", "Adults (226)"), 
             filename = "fig2A.tif", output = T, imagetype = "tiff", 
             height = 5, width = 5, units = "in", resolution = 300, 
             col = c("#4b265c", "#25a165"), lwd = 1,
             fill = c(alpha("#4b265c", 0.3), alpha("#25a165", 0.3)), 
             fontfamily = "sans", cat.fontfamily = "sans",
             cat.default.pos = "outer", cat.dist = c(0.1, 0.1),
             margin = 0.2)

venn.diagram(x = list(c, a), 
             category.names = c("Children (57)", "Adults (226)"), 
             filename = "fig2A.svg", output = T, imagetype = "svg", 
             height = 5, width = 5, units = "in", resolution = 300, 
             col = c("#4b265c", "#25a165"), lwd = 1,
             fill = c(alpha("#4b265c", 0.3), alpha("#25a165", 0.3)), 
             fontfamily = "sans", cat.fontfamily = "sans",
             cat.default.pos = "outer", cat.dist = c(0.1, 0.1),
             margin = 0.2)

tiff(file="fig2B.tif", res=300, width=7, height=5, units="in")
plot(White, col="#377eb8", lwd = 4, main = "Random Forest ROC", 
     legacy.axes = TRUE, xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Black, col = "#e41a1c", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Asian, col = "#ffd92f", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
legend("bottomright", 
       legend = c("White, AUC = 0.931", "Black, AUC = 0.965", "Asian/Pacific Islander, AUC = 0.930"), 
       col = c("#377eb8", "#e41a1c", "#ffd92f"), lwd = 4)
dev.off()

setEPS()
postscript(file="fig2B.eps", width=7, height=5, paper = "special")
plot(White, col="#377eb8", lwd = 4, main = "Random Forest ROC", 
     legacy.axes = TRUE, xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Black, col = "#e41a1c", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Asian, col = "#ffd92f", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
legend("bottomright", 
       legend = c("White, AUC = 0.931", "Black, AUC = 0.965", "Asian/Pacific Islander, AUC = 0.930"), 
       col = c("#377eb8", "#e41a1c", "#ffd92f"), lwd = 4)
dev.off()
```