---
title: "16S analysis"
author: "Liz Mallott"
date: "12/11/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Loading required libraries

The following libraries are required for this analysis:
tidyverse (v.1.3.0)
vegan (v.2.5-6)
pairwiseAdonis (v.0.3, https://github.com/pmartinezarbizu/pairwiseAdonis)
ggtext (v.0.1.1)
cowplot (v.1.1.0)
glmmTMB (v.1.0.2.1)
lme4 (v.1.1-27.1)
emmeans (v.1.6.3)
car (v.3.0-9)
fdrtool (v.1.2.15)
ggpubr (v.0.4.0)
phyloseq (v.1.34.0)
ANCOMBC (v.3.13, https://bioconductor.org/packages/release/bioc/html/ANCOMBC.html)
pROC (v.1.18.0)
VennDiagram (v.1.6.0)
dotwhisker (v.0.7.4)
mikropml
furrr


```{r load_libraries}
library(tidyverse)
library(vegan)
library(pairwiseAdonis)
library(ggtext)
library(cowplot)
library(glmmTMB)
library(lme4)
library(multcomp)
library(car)
library(fdrtool)
library(ggpubr)
library(phyloseq)
library(ANCOMBC)
library(pROC)
library(VennDiagram)
library(dotwhisker)
library(mikropml)
library(furrr)
```

## Importing data

Import distance matrices created in QIIME2-2021.2. Import feature table. Import metadata.

```{r import_distances}
unweighted_re = read_tsv("unweighted-distance-matrix-re-noother.tsv")
weighted_re = read_tsv("weighted-distance-matrix-re-noother.tsv")
bray_re = read_tsv("bray-distance-matrix-noother.tsv")

unweighted_re_age_sex_delivery_feeding = read_tsv("unweighted-distance-matrix-re-age-sex-delivery-feeding.tsv")
weighted_re_age_sex_delivery_feeding = read_tsv("weighted-distance-matrix-re-age-sex-delivery-feeding.tsv")
bray_re_age_sex_delivery_feeding = read_tsv("bray-curtis-distance-matrix-re-age-sex-delivery-feeding.tsv")

unweighted_re_01week = read_tsv("unweighted-distance-matrix-re-01week.tsv")
weighted_re_01week = read_tsv("weighted-distance-matrix-re-01week.tsv")
bray_re_01week = read_tsv("bray-curtis-distance-matrix-re-01week.tsv")
unweighted_re_16week = read_tsv("unweighted-distance-matrix-re-16week.tsv")
weighted_re_16week = read_tsv("weighted-distance-matrix-re-16week.tsv")
bray_re_16week = read_tsv("bray-curtis-distance-matrix-16week.tsv")
unweighted_re_06week = read_tsv("unweighted-distance-matrix-re-06week.tsv")
weighted_re_06week = read_tsv("weighted-distance-matrix-re-06week.tsv")
unweighted_re_03month = read_tsv("unweighted-distance-matrix-re-03month.tsv")
weighted_re_03month = read_tsv("weighted-distance-matrix-re-03month.tsv")
unweighted_re_36month = read_tsv("unweighted-distance-matrix-re-36month.tsv")
weighted_re_36month = read_tsv("weighted-distance-matrix-re-36month.tsv")
unweighted_re_69month = read_tsv("unweighted-distance-matrix-re-69month.tsv")
weighted_re_69month = read_tsv("weighted-distance-matrix-re-69month.tsv")
unweighted_re_912month = read_tsv("unweighted-distance-matrix-re-912month.tsv")
weighted_re_912month = read_tsv("weighted-distance-matrix-re-912month.tsv")
unweighted_re_6week3month = read_tsv("unweighted-distance-matrix-re-6week3month.tsv")
weighted_re_6week3month = read_tsv("weighted-distance-matrix-re-6week3month.tsv")
bray_re_6week3month = read_tsv("bray-curtis-distance-matrix-6week3month.tsv")
unweighted_re_312month = read_tsv("unweighted-distance-matrix-re-312month.tsv")
weighted_re_312month = read_tsv("weighted-distance-matrix-re-312month.tsv")
bray_re_312month = read_tsv("bray-curtis-distance-matrix-312month.tsv")
unweighted_re_012month = read_tsv("unweighted-distance-matrix-re-012month.tsv")
weighted_re_012month = read_tsv("weighted-distance-matrix-re-012month.tsv")
unweighted_re_13year = read_tsv("unweighted-distance-matrix-re-13year.tsv")
weighted_re_13year = read_tsv("weighted-distance-matrix-re-13year.tsv")
bray_re_13year = read_tsv("bray-curtis-distance-matrix-13year.tsv")
unweighted_re_36year = read_tsv("unweighted-distance-matrix-re-36year.tsv")
weighted_re_36year = read_tsv("weighted-distance-matrix-re-36year.tsv")
unweighted_re_612year = read_tsv("unweighted-distance-matrix-re-612year.tsv")
weighted_re_612year = read_tsv("weighted-distance-matrix-re-612year.tsv")
unweighted_re_312year = read_tsv("unweighted-distance-matrix-re-312year.tsv")
weighted_re_312year = read_tsv("weighted-distance-matrix-re-312year.tsv")
bray_re_312year = read_tsv("bray-curtis-distance-matrix-312year.tsv")

unweighted_kim = read_tsv("unweighted-distance-matrix-kim.tsv")
weighted_kim = read_tsv("weighted-distance-matrix-kim.tsv")
unweighted_robinson = read_tsv("unweighted-distance-matrix-robinson.tsv")
weighted_robinson = read_tsv("weighted-distance-matrix-robinson.tsv")
unweighted_chu = read_tsv("unweighted-distance-matrix-chu.tsv")
weighted_chu = read_tsv("weighted-distance-matrix-chu.tsv")
unweighted_herman = read_tsv("unweighted-distance-matrix-herman.tsv")
weighted_herman = read_tsv("weighted-distance-matrix-herman.tsv")
unweighted_planer = read_tsv("unweighted-distance-matrix-planer.tsv")
weighted_planer = read_tsv("weighted-distance-matrix-planer.tsv")
unweighted_levin = read_tsv("unweighted-distance-matrix-levin.tsv")
weighted_levin = read_tsv("weighted-distance-matrix-levin.tsv")
unweighted_db = read_tsv("unweighted-distance-matrix-db.tsv")
weighted_db = read_tsv("weighted-distance-matrix-db.tsv")
unweighted_cioffi = read_tsv("unweighted-distance-matrix-cioffi.tsv")
weighted_cioffi = read_tsv("weighted-distance-matrix-cioffi.tsv")

asvs = read_tsv("feature-table-full.tsv")

metadata = read_tsv("metadata-combined-re.txt") 
metadata_re = metadata %>% 
  right_join(unweighted_re, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
write_tsv(metadata_re, "metadata_re.tsv")
metadata_re = read_tsv("metadata_re.tsv")

metadata_re_onesubj = metadata %>% 
  right_join(unweighted_re_onesubj, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)

metadata_re_age_sex_delivery_feeding = metadata %>% 
  right_join(unweighted_re_age_sex_delivery_feeding, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)

metadata_re_age_sex_del_fed_onesubj = metadata %>% 
  right_join(unweighted_re_age_sex_delivery_feeding_onesubj, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)

metadata_re_01week = metadata %>% 
  right_join(unweighted_re_01week, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_16week = metadata %>% 
  right_join(unweighted_re_16week, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_06week = metadata %>% 
  right_join(unweighted_re_06week, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_6week3month = metadata %>% 
  right_join(unweighted_re_6week3month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_312month = metadata %>% 
  right_join(unweighted_re_312month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_012month = metadata %>% 
  right_join(unweighted_re_012month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_03month = metadata %>% 
  right_join(unweighted_re_03month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_36month = metadata %>% 
  right_join(unweighted_re_36month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_69month = metadata %>% 
  right_join(unweighted_re_69month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_912month = metadata %>% 
  right_join(unweighted_re_912month, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_13year = metadata %>% 
  right_join(unweighted_re_13year, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_36year = metadata %>% 
  right_join(unweighted_re_36year, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_612year = metadata %>% 
  right_join(unweighted_re_612year, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_312year = metadata %>% 
  right_join(unweighted_re_312year, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)

metadata_re_kim = metadata %>% 
  right_join(unweighted_kim, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_herman = metadata %>% 
  right_join(unweighted_herman, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_levin = metadata %>% 
  right_join(unweighted_levin, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_robinson = metadata %>% 
  right_join(unweighted_robinson, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_planer = metadata %>% 
  right_join(unweighted_planer, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_levin = metadata %>% 
  right_join(unweighted_levin, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_db = metadata %>% 
  right_join(unweighted_db, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_cioffi = metadata %>% 
  right_join(unweighted_cioffi, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)
metadata_re_chu = metadata %>% 
  right_join(unweighted_chu, by = c("SampleID" = "...1")) %>% 
               dplyr::select(1:41)


unweighted_re_dist = as.dist(unweighted_re[,2:2768])
weighted_re_dist = as.dist(weighted_re[,2:2768])
bray_re_dist = as.dist(bray_re[,2:2768])

unweighted_re_age_sex_del_fed_dist = as.dist(unweighted_re_age_sex_delivery_feeding[,2:2291])
weighted_re_age_sex_del_fed_dist = as.dist(weighted_re_age_sex_delivery_feeding[,2:2291])
bray_re_age_sex_del_fed_dist = as.dist(bray_re_age_sex_delivery_feeding[,2:2291])

unweighted_re_onesubj_dist = as.dist(unweighted_re_onesubj[,2:717])
weighted_re_onesubj_dist = as.dist(weighted_re_onesubj[,2:717])

unweighted_re_age_sex_del_fed_onesubj_dist = as.dist(unweighted_re_age_sex_delivery_feeding_onesubj[,2:430])
weighted_re_age_sex_del_fed_onesubj_dist = as.dist(weighted_re_age_sex_delivery_feeding_onesubj[,2:430])

unweighted_re_01week_dist = as.dist(unweighted_re_01week[,2:120])
weighted_re_01week_dist = as.dist(weighted_re_01week[,2:120])
bray_re_01week_dist = as.dist(bray_re_01week[,2:120])
unweighted_re_16week_dist = as.dist(unweighted_re_16week[,2:286])
weighted_re_16week_dist = as.dist(weighted_re_16week[,2:286])
bray_re_16week_dist = as.dist(bray_re_16week[,2:286])
unweighted_re_06week_dist = as.dist(unweighted_re_06week[,2:405])
weighted_re_06week_dist = as.dist(weighted_re_06week[,2:405])
unweighted_re_6week3month_dist = as.dist(unweighted_re_6week3month[,2:245])
weighted_re_6week3month_dist = as.dist(weighted_re_6week3month[,2:245])
bray_re_6week3month_dist = as.dist(bray_re_6week3month[,2:245])
unweighted_re_312month_dist = as.dist(unweighted_re_312month[,2:831])
weighted_re_312month_dist = as.dist(weighted_re_312month[,2:831])
bray_re_312month_dist = as.dist(bray_re_312month[,2:831])
unweighted_re_012month_dist = as.dist(unweighted_re_012month[,2:1479])
weighted_re_012month_dist = as.dist(weighted_re_012month[,2:1479])
unweighted_re_03month_dist = as.dist(unweighted_re_03month[,2:649])
weighted_re_03month_dist = as.dist(weighted_re_03month[,2:649])
unweighted_re_36month_dist = as.dist(unweighted_re_36month[,2:227])
weighted_re_36month_dist = as.dist(weighted_re_36month[,2:227])
unweighted_re_69month_dist = as.dist(unweighted_re_69month[,2:354])
weighted_re_69month_dist = as.dist(weighted_re_69month[,2:354])
unweighted_re_912month_dist = as.dist(unweighted_re_912month[,2:245])
weighted_re_912month_dist = as.dist(weighted_re_912month[,2:245])
unweighted_re_13year_dist = as.dist(unweighted_re_13year[,2:1008])
weighted_re_13year_dist = as.dist(weighted_re_13year[,2:1008])
bray_re_13year_dist = as.dist(bray_re_13year[,2:1008])
unweighted_re_36year_dist = as.dist(unweighted_re_36year[,2:157])
weighted_re_36year_dist = as.dist(weighted_re_36year[,2:157])
unweighted_re_612year_dist = as.dist(unweighted_re_612year[,2:112])
weighted_re_612year_dist = as.dist(weighted_re_612year[,2:112])
unweighted_re_312year_dist = as.dist(unweighted_re_312year[,2:268])
weighted_re_312year_dist = as.dist(weighted_re_312year[,2:268])
bray_re_312year_dist = as.dist(bray_re_312year[,2:268])

unweighted_kim_dist = as.dist(unweighted_kim[,2:37])
weighted_kim_dist = as.dist(weighted_kim[,2:37])
unweighted_chu_dist = as.dist(unweighted_chu[,2:33])
weighted_chu_dist = as.dist(weighted_chu[,2:33])
unweighted_cioffi_dist = as.dist(unweighted_cioffi[,2:45])
weighted_cioffi_dist = as.dist(weighted_cioffi[,2:45])
unweighted_db_dist = as.dist(unweighted_db[,2:328])
weighted_db_dist = as.dist(weighted_db[,2:328])
unweighted_herman_dist = as.dist(unweighted_herman[,2:252])
weighted_herman_dist = as.dist(weighted_herman[,2:252])
unweighted_levin_dist = as.dist(unweighted_levin[,2:269])
weighted_levin_dist = as.dist(weighted_levin[,2:269])
unweighted_planer_dist = as.dist(unweighted_planer[,2:1664])
weighted_planer_dist = as.dist(weighted_planer[,2:1664])
unweighted_robinson_dist = as.dist(unweighted_robinson[,2:114])
weighted_robinson_dist = as.dist(weighted_robinson[,2:114])
```

## Analysis summary statistics

Summary statistics across all studies were calculated for both the full dataset and the samples for which all metadata were present. 

```{r summary_stats}
race_stats_all = metadata_re %>% count(Race)
race_stats_filtered = metadata_re_age_sex_delivery_feeding %>% count(Race)

ethnicity_stats_all = metadata_re %>% count(Ethnicity)
ethnicity_stats_filtered = metadata_re_age_sex_delivery_feeding %>% count(Ethnicity)

sex_stats_all = metadata_re %>% count(host_sex)

age_stats_all = metadata_re %>% count(Host_age_cat)

del_stats_all = metadata_re %>% count(host_del_route)

diet_stats_all = metadata_re %>% count(Breastfeeding)

race_sex = metadata_re %>% group_by(Race) %>% count(host_sex) %>% 
  filter(host_sex != "NA") %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("host_sex")
chisq.test(as.data.frame(race_sex))
chisq.test(as.data.frame(race_sex))$observed
chisq.test(as.data.frame(race_sex))$expected
chisq.test(as.data.frame(race_sex))$residuals

race_age = metadata_re %>% group_by(Race) %>% count(Host_age_cat) %>% 
  filter(Host_age_cat != "NA") %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("Host_age_cat")
chisq.test(as.data.frame(race_age))
chisq.test(as.data.frame(race_age))$observed
chisq.test(as.data.frame(race_age))$expected
chisq.test(as.data.frame(race_age))$residuals

race_del = metadata_re %>% group_by(Race) %>% count(host_del_route) %>% 
  filter(host_del_route != "NA") %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("host_del_route")
chisq.test(as.data.frame(race_del))
chisq.test(as.data.frame(race_del))$observed
chisq.test(as.data.frame(race_del))$expected

race_diet = metadata_re %>% group_by(Race) %>% count(Breastfeeding) %>% 
  filter(Breastfeeding != "NA") %>% 
  pivot_wider(names_from = "Race", values_from = "n")  %>% 
  column_to_rownames("Breastfeeding")
chisq.test(as.data.frame(race_diet))
chisq.test(as.data.frame(race_diet))$observed
chisq.test(as.data.frame(race_diet))$expected
chisq.test(as.data.frame(race_diet))$residuals

ethnicity_sex = metadata_re %>% group_by(Ethnicity) %>% count(host_sex) %>% 
  filter(host_sex != "NA") %>% 
  pivot_wider(names_from = "Ethnicity", values_from = "n")  %>% 
  column_to_rownames("host_sex")
chisq.test(as.data.frame(ethnicity_sex))
chisq.test(as.data.frame(ethnicity_sex))$observed
chisq.test(as.data.frame(ethnicity_sex))$expected

ethnicity_age = metadata_re %>% group_by(Ethnicity) %>% count(Host_age_cat) %>% 
  filter(Host_age_cat != "NA") %>% 
  pivot_wider(names_from = "Ethnicity", values_from = "n")  %>% 
  column_to_rownames("Host_age_cat")
chisq.test(as.data.frame(ethnicity_age))
chisq.test(as.data.frame(ethnicity_age))$observed
chisq.test(as.data.frame(ethnicity_age))$expected

ethnicity_del = metadata_re %>% group_by(Ethnicity) %>% count(host_del_route) %>% 
  filter(host_del_route != "NA") %>% 
  pivot_wider(names_from = "Ethnicity", values_from = "n")  %>% 
  column_to_rownames("host_del_route")
chisq.test(as.data.frame(ethnicity_del))
chisq.test(as.data.frame(ethnicity_del))$observed
chisq.test(as.data.frame(ethnicity_del))$expected

ethnicity_diet = metadata_re %>% group_by(Ethnicity) %>% count(Breastfeeding) %>% 
  filter(Breastfeeding != "NA") %>% 
  pivot_wider(names_from = "Ethnicity", values_from = "n")  %>% 
  column_to_rownames("Breastfeeding")
chisq.test(as.data.frame(ethnicity_diet))
chisq.test(as.data.frame(ethnicity_diet))$observed
chisq.test(as.data.frame(ethnicity_diet))$expected
chisq.test(as.data.frame(ethnicity_diet))$residuals
```

## PERMANOVAs

Permanovas were performed to assess the effect of race and ethnicity on microbial community composition, controlling for study, sequencing method.

#### Unweighted
```{r permanovas_unweighted}
adonis2(unweighted_re_dist ~ Race + Ethnicity + Study + host_subject_id, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(unweighted_re_dist ~ Race + Ethnicity + host_subject_id, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

adonis2(unweighted_re_dist ~ Race + Ethnicity + Study + family, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(unweighted_re_dist ~ Race + Ethnicity + family, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

anova(betadisper(unweighted_re_dist, group = metadata_re$Race))
anova(betadisper(unweighted_re_dist, group = metadata_re$Ethnicity))
anova(betadisper(unweighted_re_dist, group = metadata_re$Study))
anova(betadisper(unweighted_re_dist, group = metadata_re$host_subject_id))
anova(betadisper(unweighted_re_dist, group = metadata_re$family))

adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)

adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Race))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Ethnicity))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Study))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Host_age_cat))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_sex))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_del_route))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Breastfeeding))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_subject_id))
anova(betadisper(unweighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(unweighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

adonis2(unweighted_re_onesubj_dist ~ Race + Ethnicity + Study, data = metadata_re_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$Race))
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$Ethnicity))
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$Study))
anova(betadisper(unweighted_re_onesubj_dist, group = metadata_re_onesubj$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(unweighted_re_onesubj_dist ~ Race + Ethnicity, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(unweighted_re_onesubj_dist ~ Race + Ethnicity + family, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

adonis2(unweighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Race))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Ethnicity))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Study))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Host_age_cat))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_sex))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_del_route))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Breastfeeding))
anova(betadisper(unweighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(unweighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(unweighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_01week, Study)
adonis2(unweighted_re_01week_dist ~ Race + Ethnicity, data = metadata_re_01week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_01week_dist, group = metadata_re_01week$Race))
anova(betadisper(unweighted_re_01week_dist, group = metadata_re_01week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_16week, Study)
adonis2(unweighted_re_16week_dist ~ Race + Ethnicity, data = metadata_re_16week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_16week_dist, group = metadata_re_16week$Race))
anova(betadisper(unweighted_re_16week_dist, group = metadata_re_16week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_06week, Study)
adonis2(unweighted_re_06week_dist ~ Race + Ethnicity, data = metadata_re_06week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_06week_dist, group = metadata_re_06week$Race))
anova(betadisper(unweighted_re_06week_dist, group = metadata_re_06week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_6week3month, Study)
adonis2(unweighted_re_6week3month_dist ~ Race + Ethnicity, data = metadata_re_6week3month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_6week3month_dist, group = metadata_re_6week3month$Race))
anova(betadisper(unweighted_re_6week3month_dist, group = metadata_re_6week3month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312month, Study)
adonis2(unweighted_re_312month_dist ~ Race + Ethnicity, data = metadata_re_312month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_312month_dist, group = metadata_re_312month$Race))
anova(betadisper(unweighted_re_312month_dist, group = metadata_re_312month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(unweighted_re_012month_dist ~ Race + Ethnicity + host_subject_id, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$Race))
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$Ethnicity))
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$host_subject_id))
anova(betadisper(unweighted_re_012month_dist, group = metadata_re_012month$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(unweighted_re_012month_dist ~ Race + Ethnicity + family, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(unweighted_re_012month_dist ~ Race + Ethnicity, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_03month, Study)
adonis2(unweighted_re_03month_dist ~ Race + Ethnicity, data = metadata_re_03month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_03month_dist, group = metadata_re_03month$Race))
anova(betadisper(unweighted_re_03month_dist, group = metadata_re_03month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_36month, Study)
adonis2(unweighted_re_36month_dist ~ Race + Ethnicity, data = metadata_re_36month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_36month_dist, group = metadata_re_36month$Race))
anova(betadisper(unweighted_re_36month_dist, group = metadata_re_36month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_69month, Study)
adonis2(unweighted_re_69month_dist ~ Race + Ethnicity, data = metadata_re_69month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_69month_dist, group = metadata_re_69month$Race))
anova(betadisper(unweighted_re_69month_dist, group = metadata_re_69month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_912month, Study)
adonis2(unweighted_re_912month_dist ~ Race + Ethnicity, data = metadata_re_912month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_912month_dist, group = metadata_re_912month$Race))
anova(betadisper(unweighted_re_912month_dist, group = metadata_re_912month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_13year, Study)
adonis2(unweighted_re_13year_dist ~ Race + Ethnicity, data = metadata_re_13year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_13year_dist, group = metadata_re_13year$Race))
anova(betadisper(unweighted_re_13year_dist, group = metadata_re_13year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_36year, Study)
adonis2(unweighted_re_36year_dist ~ Race + Ethnicity, data = metadata_re_36year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_36year_dist, group = metadata_re_36year$Race))
anova(betadisper(unweighted_re_36year_dist, group = metadata_re_36year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_612year, Study)
adonis2(unweighted_re_612year_dist ~ Race + Ethnicity, data = metadata_re_612year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_612year_dist, group = metadata_re_612year$Race))
anova(betadisper(unweighted_re_612year_dist, group = metadata_re_612year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312year, Study)
adonis2(unweighted_re_312year_dist ~ Race + Ethnicity, data = metadata_re_312year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(unweighted_re_312year_dist, group = metadata_re_312year$Race))
anova(betadisper(unweighted_re_312year_dist, group = metadata_re_312year$Ethnicity))

adonis2(unweighted_kim_dist ~ Race + Ethnicity + host_sex, data=metadata_re_kim, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_kim_dist, group = metadata_re_kim$Race))
anova(betadisper(unweighted_kim_dist, group = metadata_re_kim$Ethnicity))
anova(betadisper(unweighted_kim_dist, group = metadata_re_kim$host_sex))

adonis2(unweighted_herman_dist ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding, data=metadata_re_herman, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_herman_dist, group = metadata_re_herman$Race))
anova(betadisper(unweighted_herman_dist, group = metadata_re_herman$Ethnicity))
anova(betadisper(unweighted_herman_dist, group = metadata_re_herman$Host_age_cat))
anova(betadisper(unweighted_herman_dist, group = metadata_re_herman$host_sex))
anova(betadisper(unweighted_herman_dist, group = metadata_re_herman$Breastfeeding))

adonis2(unweighted_chu_dist ~ Race + host_sex + host_del_route + Breastfeeding, data=metadata_re_chu, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_chu_dist, group = metadata_re_chu$Race))
anova(betadisper(unweighted_chu_dist, group = metadata_re_chu$host_sex))
anova(betadisper(unweighted_chu_dist, group = metadata_re_chu$host_del_route))
anova(betadisper(unweighted_chu_dist, group = metadata_re_chu$Breastfeeding))

adonis2(unweighted_robinson_dist ~ Race + Ethnicity + host_sex, data=metadata_re_robinson, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_robinson_dist, group = metadata_re_robinson$Race))
anova(betadisper(unweighted_robinson_dist, group = metadata_re_robinson$Ethnicity))
anova(betadisper(unweighted_robinson_dist, group = metadata_re_robinson$host_sex))

adonis2(unweighted_planer_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_planer, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_planer_dist, group = metadata_re_planer$Race))
anova(betadisper(unweighted_planer_dist, group = metadata_re_planer$Ethnicity))
anova(betadisper(unweighted_planer_dist, group = metadata_re_planer$Host_age_cat))
anova(betadisper(unweighted_planer_dist, group = metadata_re_planer$host_sex))
anova(betadisper(unweighted_planer_dist, group = metadata_re_planer$host_del_route))
anova(betadisper(unweighted_planer_dist, group = metadata_re_planer$Breastfeeding))

adonis2(unweighted_levin_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_levin, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_levin_dist, group = metadata_re_levin$Race))
anova(betadisper(unweighted_levin_dist, group = metadata_re_levin$Ethnicity))
anova(betadisper(unweighted_levin_dist, group = metadata_re_levin$Host_age_cat))
anova(betadisper(unweighted_levin_dist, group = metadata_re_levin$host_sex))
anova(betadisper(unweighted_levin_dist, group = metadata_re_levin$host_del_route))
anova(betadisper(unweighted_levin_dist, group = metadata_re_levin$Breastfeeding))

adonis2(unweighted_db_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_db, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_db_dist, group = metadata_re_db$Race))
anova(betadisper(unweighted_db_dist, group = metadata_re_db$Ethnicity))
anova(betadisper(unweighted_db_dist, group = metadata_re_db$Host_age_cat))
anova(betadisper(unweighted_db_dist, group = metadata_re_db$host_sex))
anova(betadisper(unweighted_db_dist, group = metadata_re_db$host_del_route))
anova(betadisper(unweighted_db_dist, group = metadata_re_db$Breastfeeding))

adonis2(unweighted_cioffi_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_cioffi, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(unweighted_cioffi_dist, group = metadata_re_cioffi$Race))
anova(betadisper(unweighted_cioffi_dist, group = metadata_re_cioffi$Ethnicity))
anova(betadisper(unweighted_cioffi_dist, group = metadata_re_cioffi$Host_age_cat))
anova(betadisper(unweighted_cioffi_dist, group = metadata_re_cioffi$host_sex))
anova(betadisper(unweighted_cioffi_dist, group = metadata_re_cioffi$host_del_route))
anova(betadisper(unweighted_cioffi_dist, group = metadata_re_cioffi$Breastfeeding))
```
#### Weighted
```{r permanovas_weighted}
adonis2(weighted_re_dist ~ Race + Ethnicity + Study + host_subject_id, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(weighted_re_dist ~ Race + Ethnicity + host_subject_id, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

adonis2(weighted_re_dist ~ Race + Ethnicity + Study + family, data=metadata_re, by = "margin", permutations = 499, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(weighted_re_dist ~ Race + Ethnicity + family, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

anova(betadisper(weighted_re_dist, group = metadata_re$Race))
anova(betadisper(weighted_re_dist, group = metadata_re$Ethnicity))
anova(betadisper(weighted_re_dist, group = metadata_re$Study))
anova(betadisper(weighted_re_dist, group = metadata_re$host_subject_id))
anova(betadisper(weighted_re_dist, group = metadata_re$family))

adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)

adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Race))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Ethnicity))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Study))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Host_age_cat))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_sex))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_del_route))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Breastfeeding))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_subject_id))
anova(betadisper(weighted_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(weighted_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

adonis2(weighted_re_onesubj_dist ~ Race + Ethnicity + Study, data = metadata_re_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$Race))
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$Ethnicity))
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$Study))
anova(betadisper(weighted_re_onesubj_dist, group = metadata_re_onesubj$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(weighted_re_onesubj_dist ~ Race + Ethnicity, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_onesubj, Study)
adonis2(weighted_re_onesubj_dist ~ Race + Ethnicity + family, data = metadata_re_onesubj, by = "margin", permutations = perm, parallel = 4)

adonis2(weighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Study + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Race))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Ethnicity))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Study))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Host_age_cat))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_sex))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$host_del_route))
anova(betadisper(weighted_re_age_sex_del_fed_onesubj_dist, group = metadata_re_age_sex_del_fed_onesubj$Breastfeeding))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(weighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_del_fed_onesubj, Study)
adonis2(weighted_re_age_sex_del_fed_onesubj_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_del_fed_onesubj, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_01week, Study)
adonis2(weighted_re_01week_dist ~ Race + Ethnicity, data = metadata_re_01week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_01week_dist, group = metadata_re_01week$Race))
anova(betadisper(weighted_re_01week_dist, group = metadata_re_01week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_16week, Study)
adonis2(weighted_re_16week_dist ~ Race + Ethnicity, data = metadata_re_16week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_16week_dist, group = metadata_re_16week$Race))
anova(betadisper(weighted_re_16week_dist, group = metadata_re_16week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_06week, Study)
adonis2(weighted_re_06week_dist ~ Race + Ethnicity, data = metadata_re_06week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_06week_dist, group = metadata_re_06week$Race))
anova(betadisper(weighted_re_06week_dist, group = metadata_re_06week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_6week3month, Study)
adonis2(weighted_re_6week3month_dist ~ Race + Ethnicity, data = metadata_re_6week3month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_6week3month_dist, group = metadata_re_6week3month$Race))
anova(betadisper(weighted_re_6week3month_dist, group = metadata_re_6week3month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312month, Study)
adonis2(weighted_re_312month_dist ~ Race + Ethnicity, data = metadata_re_312month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_312month_dist, group = metadata_re_312month$Race))
anova(betadisper(weighted_re_312month_dist, group = metadata_re_312month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(weighted_re_012month_dist ~ Race + Ethnicity + host_subject_id, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$Race))
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$Ethnicity))
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$host_subject_id))
anova(betadisper(weighted_re_012month_dist, group = metadata_re_012month$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(weighted_re_012month_dist ~ Race + Ethnicity + family, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_012month, Study)
adonis2(weighted_re_012month_dist ~ Race + Ethnicity, data = metadata_re_012month, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_03month, Study)
adonis2(weighted_re_03month_dist ~ Race + Ethnicity, data = metadata_re_03month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_03month_dist, group = metadata_re_03month$Race))
anova(betadisper(weighted_re_03month_dist, group = metadata_re_03month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_36month, Study)
adonis2(weighted_re_36month_dist ~ Race + Ethnicity, data = metadata_re_36month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_36month_dist, group = metadata_re_36month$Race))
anova(betadisper(weighted_re_36month_dist, group = metadata_re_36month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_69month, Study)
adonis2(weighted_re_69month_dist ~ Race + Ethnicity, data = metadata_re_69month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_69month_dist, group = metadata_re_69month$Race))
anova(betadisper(weighted_re_69month_dist, group = metadata_re_69month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_912month, Study)
adonis2(weighted_re_912month_dist ~ Race + Ethnicity, data = metadata_re_912month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_912month_dist, group = metadata_re_912month$Race))
anova(betadisper(weighted_re_912month_dist, group = metadata_re_912month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_13year, Study)
adonis2(weighted_re_13year_dist ~ Race + Ethnicity, data = metadata_re_13year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_13year_dist, group = metadata_re_13year$Race))
anova(betadisper(weighted_re_13year_dist, group = metadata_re_13year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_36year, Study)
adonis2(weighted_re_36year_dist ~ Race + Ethnicity, data = metadata_re_36year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_36year_dist, group = metadata_re_36year$Race))
anova(betadisper(weighted_re_36year_dist, group = metadata_re_36year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_612year, Study)
adonis2(weighted_re_612year_dist ~ Race + Ethnicity, data = metadata_re_612year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_612year_dist, group = metadata_re_612year$Race))
anova(betadisper(weighted_re_612year_dist, group = metadata_re_612year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312year, Study)
adonis2(weighted_re_312year_dist ~ Race + Ethnicity, data = metadata_re_312year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(weighted_re_312year_dist, group = metadata_re_312year$Race))
anova(betadisper(weighted_re_312year_dist, group = metadata_re_312year$Ethnicity))

adonis2(weighted_kim_dist ~ Race + Ethnicity + host_sex, data=metadata_re_kim, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_kim_dist, group = metadata_re_kim$Race))
anova(betadisper(weighted_kim_dist, group = metadata_re_kim$Ethnicity))
anova(betadisper(weighted_kim_dist, group = metadata_re_kim$host_sex))

adonis2(weighted_herman_dist ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding, data=metadata_re_herman, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_herman_dist, group = metadata_re_herman$Race))
anova(betadisper(weighted_herman_dist, group = metadata_re_herman$Ethnicity))
anova(betadisper(weighted_herman_dist, group = metadata_re_herman$Host_age_cat))
anova(betadisper(weighted_herman_dist, group = metadata_re_herman$host_sex))
anova(betadisper(weighted_herman_dist, group = metadata_re_herman$Breastfeeding))

adonis2(weighted_chu_dist ~ Race + host_sex + host_del_route + Breastfeeding, data=metadata_re_chu, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_chu_dist, group = metadata_re_chu$Race))
anova(betadisper(weighted_chu_dist, group = metadata_re_chu$host_sex))
anova(betadisper(weighted_chu_dist, group = metadata_re_chu$host_del_route))
anova(betadisper(weighted_chu_dist, group = metadata_re_chu$Breastfeeding))

adonis2(weighted_robinson_dist ~ Race + Ethnicity + host_sex, data=metadata_re_robinson, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_robinson_dist, group = metadata_re_robinson$Race))
anova(betadisper(weighted_robinson_dist, group = metadata_re_robinson$Ethnicity))
anova(betadisper(weighted_robinson_dist, group = metadata_re_robinson$host_sex))

adonis2(weighted_planer_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_planer, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_planer_dist, group = metadata_re_planer$Race))
anova(betadisper(weighted_planer_dist, group = metadata_re_planer$Ethnicity))
anova(betadisper(weighted_planer_dist, group = metadata_re_planer$Host_age_cat))
anova(betadisper(weighted_planer_dist, group = metadata_re_planer$host_sex))
anova(betadisper(weighted_planer_dist, group = metadata_re_planer$host_del_route))
anova(betadisper(weighted_planer_dist, group = metadata_re_planer$Breastfeeding))

adonis2(weighted_levin_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_levin, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_levin_dist, group = metadata_re_levin$Race))
anova(betadisper(weighted_levin_dist, group = metadata_re_levin$Ethnicity))
anova(betadisper(weighted_levin_dist, group = metadata_re_levin$Host_age_cat))
anova(betadisper(weighted_levin_dist, group = metadata_re_levin$host_sex))
anova(betadisper(weighted_levin_dist, group = metadata_re_levin$host_del_route))
anova(betadisper(weighted_levin_dist, group = metadata_re_levin$Breastfeeding))

adonis2(weighted_db_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_db, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_db_dist, group = metadata_re_db$Race))
anova(betadisper(weighted_db_dist, group = metadata_re_db$Ethnicity))
anova(betadisper(weighted_db_dist, group = metadata_re_db$Host_age_cat))
anova(betadisper(weighted_db_dist, group = metadata_re_db$host_sex))
anova(betadisper(weighted_db_dist, group = metadata_re_db$host_del_route))
anova(betadisper(weighted_db_dist, group = metadata_re_db$Breastfeeding))

adonis2(weighted_cioffi_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=metadata_re_cioffi, by = "margin", permutations = 499, parallel = 4)
anova(betadisper(weighted_cioffi_dist, group = metadata_re_cioffi$Race))
anova(betadisper(weighted_cioffi_dist, group = metadata_re_cioffi$Ethnicity))
anova(betadisper(weighted_cioffi_dist, group = metadata_re_cioffi$Host_age_cat))
anova(betadisper(weighted_cioffi_dist, group = metadata_re_cioffi$host_sex))
anova(betadisper(weighted_cioffi_dist, group = metadata_re_cioffi$host_del_route))
anova(betadisper(weighted_cioffi_dist, group = metadata_re_cioffi$Breastfeeding))
```

#### Bray-Curtis
```{r permanovas_bray}
perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(bray_re_dist ~ Race + Ethnicity + host_subject_id, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re, Study)
adonis2(bray_re_dist ~ Race + Ethnicity + family, data=metadata_re, by = "margin", permutations = perm, parallel = 4)

anova(betadisper(bray_re_dist, group = metadata_re$Race))
anova(betadisper(bray_re_dist, group = metadata_re$Ethnicity))
anova(betadisper(bray_re_dist, group = metadata_re$Study))
anova(betadisper(bray_re_dist, group = metadata_re$host_subject_id))
anova(betadisper(bray_re_dist, group = metadata_re$family))


anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Race))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Ethnicity))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Study))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Host_age_cat))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_sex))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_del_route))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$Breastfeeding))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$host_subject_id))
anova(betadisper(bray_re_age_sex_del_fed_dist, group = metadata_re_age_sex_delivery_feeding$family))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(bray_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + host_subject_id, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_age_sex_delivery_feeding, Study)
adonis2(bray_re_age_sex_del_fed_dist ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + family, data = metadata_re_age_sex_delivery_feeding, by = "margin", permutations = perm, parallel = 4)

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_01week, Study)
adonis2(bray_re_01week_dist ~ Race + Ethnicity, data = metadata_re_01week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_01week_dist, group = metadata_re_01week$Race))
anova(betadisper(bray_re_01week_dist, group = metadata_re_01week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_16week, Study)
adonis2(bray_re_16week_dist ~ Race + Ethnicity, data = metadata_re_16week, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_16week_dist, group = metadata_re_16week$Race))
anova(betadisper(bray_re_16week_dist, group = metadata_re_16week$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_6week3month, Study)
adonis2(bray_re_6week3month_dist ~ Race + Ethnicity, data = metadata_re_6week3month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_6week3month_dist, group = metadata_re_6week3month$Race))
anova(betadisper(bray_re_6week3month_dist, group = metadata_re_6week3month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312month, Study)
adonis2(bray_re_312month_dist ~ Race + Ethnicity, data = metadata_re_312month, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_312month_dist, group = metadata_re_312month$Race))
anova(betadisper(bray_re_312month_dist, group = metadata_re_312month$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_13year, Study)
adonis2(bray_re_13year_dist ~ Race + Ethnicity, data = metadata_re_13year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_13year_dist, group = metadata_re_13year$Race))
anova(betadisper(bray_re_13year_dist, group = metadata_re_13year$Ethnicity))

perm = how(nperm = 499)
setBlocks(perm) = with(metadata_re_312year, Study)
adonis2(bray_re_312year_dist ~ Race + Ethnicity, data = metadata_re_312year, by = "margin", permutations = perm, parallel = 4)
anova(betadisper(bray_re_312year_dist, group = metadata_re_312year$Race))
anova(betadisper(bray_re_312year_dist, group = metadata_re_312year$Ethnicity))
```

### Pairwise comparisons
Pairwise PERMANOVAs were used to examine differences between ethnicities.

```{r pairwise_permanova_unweighted}
pairwise.adonis(unweighted_re_dist, factors = metadata_re$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_dist, factors = metadata_re$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_dist, factors = metadata_re$Study, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Host_age_cat, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_sex, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_del_route, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$Breastfeeding, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(unweighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(unweighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(unweighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(unweighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
```

```{r pairwise_permanova_weighted}
pairwise.adonis(weighted_re_dist, factors = metadata_re$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_dist, factors = metadata_re$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_dist, factors = metadata_re$Study, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Race, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Ethnicity, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, factors = metadata_re_age_sex_delivery_feeding$Host_age_cat, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_sex, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$host_del_route, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_age_sex_del_fed_dist, 
                factors = metadata_re_age_sex_delivery_feeding$Breastfeeding, 
                perm = 499, p.adjust.m='holm')
pairwise.adonis(weighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(weighted_re_onesubj_dist, 
                factors = metadata_re_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(weighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Race, 
                p.adjust.m='holm', perm = 499)
pairwise.adonis(weighted_re_age_sex_del_fed_onesubj_dist, 
                factors = metadata_re_age_sex_del_fed_onesubj$Ethnicity, 
                p.adjust.m='holm', perm = 499)
```

## NMDS plots

NMDS plots were created for each distance matrix to visualize ethnicity-associated differences.

#### Weighted
```{r nmds_weighted}
mds_otus_weighted<-metaMDS(weighted_re_dist, k=2, trymax=499)
mds_otus_weighted_points<-mds_otus_weighted$points
mds_otus_weighted_points2<-merge(x=mds_otus_weighted_points, 
                                 y = metadata_re, 
                                 by.x = "row.names", by.y = "SampleID")
write_csv(mds_otus_weighted_points2, "weighted_mds_full.csv")
```

```{r nmds_weighted_plots}
mds_otus_weighted_points2 = read_csv("weighted_mds_full.csv")
weight_race <- ggplot(mds_otus_weighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Race, 
                          shape = Ethnicity)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.5))) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.012, R<sup>2</sup> = 0.05%</b><br>
           Ethnicity: p = 0.738, R<sup>2</sup> = 0.01%<br>
           <b>Age: p = 0.002, R<sup>2</sup> = 7.9%<br>
           Infant diet: p = 0.004, R<sup>2</sup> = 0.09%<br>
           Subject: p = 0.002, R<sup>2</sup> = 33.7%", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)


mds_otus_weighted_points2_age = mds_otus_weighted_points2 %>% 
  filter(Host_age_cat != "NA") %>% 
  filter(Host_age_cat != "0-1 week") %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-5.9 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-11.9 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-5.9 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-2.9 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-11.9 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-2.9 years"))
mds_otus_weighted_points2_age2 = mds_otus_weighted_points2 %>% 
  filter(Host_age_cat != "NA") %>%  
  mutate(Host_age_cat_new2 = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-11.9 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-11.9 years",
                                      Host_age_cat == "0-1 week" ~
                                        "0-2.9 months",
                                      Host_age_cat == "1-6 weeks" ~
                                        "0-2.9 months",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "0-2.9 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-11.9 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-2.9 years"))
mds_otus_weighted_points2_age$Host_age_cat_new = factor(mds_otus_weighted_points2_age$Host_age_cat_new, 
                                    levels = c("1-5.9 weeks", "6 weeks-2.9 months", "3-11.9 months", "1-2.9 years", "3-11.9 years"))
mds_otus_weighted_points2_age2$Host_age_cat_new2 = factor(mds_otus_weighted_points2_age2$Host_age_cat_new2, 
                                    levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years", "3-11.9 years"))
mds_otus_weighted_points2_age$Host_age_cat2 = factor(mds_otus_weighted_points2_age$Host_age_cat2, 
                                    levels = c("1-6 weeks", "6 weeks-3 months", "3-6 months",
                                               "6-9 months", "9-12 months",
                                                "1-3 years", "3-12 years"))
mds_otus_weighted_points2_age$Race = factor(mds_otus_weighted_points2_age$Race, 
                                    levels = c("White", "Black", 
                                               "Asian/Pacific Islander"))
mds_otus_weighted_points2_age2$Race = factor(mds_otus_weighted_points2_age2$Race, 
                                    levels = c("White", "Black", 
                                               "Asian/Pacific Islander"))
weight_age <- ggplot(mds_otus_weighted_points2_age, 
                      aes(x = MDS1, y = MDS2)) +
  geom_point(aes(shape = Race, 
                 fill = Host_age_cat_new), size=3, color = "black") + 
    scale_shape_manual(values = c(21, 24, 22)) +
  scale_fill_manual(values = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462"),
                     guide=guide_legend(override.aes=list(shape=21))) +
  scale_color_manual(values = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462")) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Host_age_cat_new, 
                   color = Host_age_cat_new),
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Age: p = 0.002, R<sup>2</sup> = 7.9%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1, color = "blue1")

mds_otus_weighted_points2_age_01week = mds_otus_weighted_points2 %>% 
  filter(Host_age_cat == "0-1 week")
mds_otus_weighted_points2_age_01week$Race = factor(mds_otus_weighted_points2_age_01week$Race, 
                                    levels = c("White", "Black", 
                                               "Asian/Pacific Islander"))

weight_age_01week <- ggplot(mds_otus_weighted_points2_age_01week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("0-0.9 week") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.280, R<sup>2</sup> = 2.0%<br>
           Ethnicity: p = 0.518, R<sup>2</sup> = 0.7%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_16week = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "1-5.9 weeks")

weight_age_16week <- ggplot(mds_otus_weighted_points2_age_16week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("1-5.9 weeks") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.542, R<sup>2</sup> = 1.7%<br>
           Ethnicity: p = 0.956, R<sup>2</sup> = 1.8%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_6week3month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "6 weeks-2.9 months")

weight_age_6week3month <- ggplot(mds_otus_weighted_points2_age_6week3month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("6 weeks-2.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.620, R<sup>2</sup> = 2.0%<br>
           Ethnicity: p = 0.532, R<sup>2</sup> = 0.5%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_03month = mds_otus_weighted_points2_age2 %>% 
  filter(Host_age_cat_new2 == "0-2.9 months")

weight_age_03month <- ggplot(mds_otus_weighted_points2_age_03month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("0-2.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.620, R<sup>2</sup> = 1.4%<br>
           Ethnicity: p = 0.064, R<sup>2</sup> = 1.0%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_012month = mds_otus_weighted_points2_age2 %>% 
  filter(Host_age_cat_new2 == "0-2.9 months" | Host_age_cat_new2 == "3-11.9 months")

weight_age_012month <- ggplot(mds_otus_weighted_points2_age_012month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("0-11.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.136, R<sup>2</sup> = 1.4%<br>
           Ethnicity: p = 0.946, R<sup>2</sup> = 0.2%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_36month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat2 == "3-6 months")

weight_age_36month <- ggplot(mds_otus_weighted_points2_age_36month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("3-5.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.024, R<sup>2</sup> = 3.2%<br>
           Ethnicity: p = 0.044, R<sup>2</sup> = 1.0%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_69month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat2 == "6-9 months")

weight_age_69month <- ggplot(mds_otus_weighted_points2_age_69month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("6-8.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 4.7%<br>
           Ethnicity: p = 0.004, R<sup>2</sup> = 0.9%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_912month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat2 == "9-12 months")

weight_age_912month <- ggplot(mds_otus_weighted_points2_age_912month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("9-11.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.024, R<sup>2</sup> = 1.8%</b><br>
           Ethnicity: p = 0.266, R<sup>2</sup> = 0.5%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_age_312month = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "3-11.9 months")

weight_age_312month <- ggplot(mds_otus_weighted_points2_age_312month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("3-11.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(color = "white"),
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.002, R<sup>2</sup> = 3.5%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 0.5%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1, color = "blue1")

mds_otus_weighted_points2_age_13year = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "1-2.9 years")

weight_age_13year <- ggplot(mds_otus_weighted_points2_age_13year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("1-2.9 years") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.5)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.002, R<sup>2</sup> = 2.3%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 1.6%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1, color = "blue1")

mds_otus_weighted_points2_age_312year = mds_otus_weighted_points2_age %>% 
  filter(Host_age_cat_new == "3-11.9 years")

weight_age_312year <- ggplot(mds_otus_weighted_points2_age_312year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("3-11.9 years") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.236, R<sup>2</sup> = 1.3%<br>
           Ethnicity: p = 0.384, R<sup>2</sup> = 0.5%<br>", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_weighted_points2_sex = mds_otus_weighted_points2 %>% 
  filter(host_sex != is.na(.))
weight_sex <- ggplot(mds_otus_weighted_points2_sex, 
                      aes(x = MDS1, y = MDS2, color = host_sex)) +
  geom_point(size=3) + scale_color_manual(values = c("#4daf4a", "#ff7f00")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_sex, 
                   linetype = host_sex), 
               type = "t", level = 0.95)

mds_otus_weighted_points2_del = mds_otus_weighted_points2 %>% 
  filter(host_del_route != is.na(.))
weight_del <- ggplot(mds_otus_weighted_points2_del, 
                      aes(x = MDS1, y = MDS2, color = host_del_route)) +
  geom_point(size=3) + scale_color_manual(values = c("#8da0cb", "#a6d854")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_del_route, 
                   linetype = host_del_route), 
               type = "t", level = 0.95)

mds_otus_weighted_points2_feed = mds_otus_weighted_points2 %>% 
  filter(Breastfeeding != is.na(.))
weight_diet <- ggplot(mds_otus_weighted_points2_feed, 
                      aes(x = MDS1, y = MDS2, color = Breastfeeding)) +
  geom_point(size=3) + scale_color_manual(values = c("#66c2a6", "#fc8d62", "#b3b3b3")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Breastfeeding, 
                   linetype = Breastfeeding), 
               type = "t", level = 0.95)

weight_study <- ggplot(mds_otus_weighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Study)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Dark2') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Weighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Study, 
                   linetype = Study), 
               type = "t", level = 0.95)
```

#### Unweighted
```{r nmds_unweighted}
mds_otus_unweighted<-metaMDS(unweighted_re_dist, k=2, trymax=499)
mds_otus_unweighted_points<-mds_otus_unweighted$points
mds_otus_unweighted_points2<-merge(x=mds_otus_unweighted_points, y = metadata_re, 
                                 by.x = "row.names", by.y = "SampleID")
write_csv(mds_otus_unweighted_points2, "unweighted_mds_full.csv")
```

```{r nmds_unweighted_plots}
mds_otus_unweighted_points2 = read_csv("unweighted_mds_full.csv",
                                     col_types = "cddccccffffffffffffdddfffffffffffffffffffd")
mds_otus_unweighted_points2$Race = factor(mds_otus_unweighted_points2$Race, 
                                    levels = c("White", "Black", 
                                               "Asian/Pacific Islander"))
unweight_race <- ggplot(mds_otus_unweighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Race, 
                          shape = Ethnicity)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.5))) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.076, R<sup>2</sup> = 0.02%<br>
           <b>Ethnicity: p = 0.024, R<sup>2</sup> = 0.03%<br>
           Age: p = 0.002, R<sup>2</sup> = 5.2%</b><br>
           Infant diet: p = 0.228, R<sup>2</sup> = 0.03%<br>
           <b>Subject: p = 0.002, R<sup>2</sup> = 34.4%", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0) 

mds_otus_unweighted_points2_age = mds_otus_unweighted_points2 %>% 
  filter(Host_age_cat != "NA") %>% 
  filter(Host_age_cat != "0-1 week") %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-11.9 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-11.9 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-5.9 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-2.9 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-11.9 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-2.9 years")) %>% 
  mutate(Host_age_cat_new2 = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-11.9 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-11.9 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "0-2.9 months",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "0-2.9 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-11.9 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-2.9 years")) %>% 
  mutate(Host_age_cat2_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-11.9 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-11.9 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-5.9 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-2.9 months",
                                      Host_age_cat == "3-6 months" ~
                                        "3-5.9 months",
                                      Host_age_cat == "6-9 months" ~
                                        "6-8.9 months",
                                      Host_age_cat == "9-12 months" ~
                                        "9-11.9 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-2.9 years"))
mds_otus_unweighted_points2_age$Host_age_cat_new = factor(mds_otus_unweighted_points2_age$Host_age_cat_new, 
                                    levels = c("1-5.9 weeks", "6 weeks-2.9 months", "3-11.9 months", "1-2.9 years", "3-11.9 years"))
mds_otus_unweighted_points2_age$Host_age_cat_new2 = factor(mds_otus_unweighted_points2_age$Host_age_cat_new2, 
                                    levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years", "3-11.9 years"))
mds_otus_unweighted_points2_age$Host_age_cat2_new = factor(mds_otus_unweighted_points2_age$Host_age_cat2_new, 
                                    levels = c("1-5.9 weeks", "6 weeks-2.9 months", "3-5.9 months",
                                               "6-8.9 months", "9-11.9 months",
                                                "1-2.9 years", "3-11.9 years"))
mds_otus_unweighted_points2_age$Race = factor(mds_otus_unweighted_points2_age$Race, 
                                    levels = c("White", "Black", 
                                               "Asian/Pacific Islander"))
unweight_age <- ggplot(mds_otus_unweighted_points2_age, 
                      aes(x = MDS1, y = MDS2)) +
  geom_point(aes(shape = Race, 
                 fill = Host_age_cat_new), size=3, color = "black") + 
    scale_shape_manual(values = c(21, 24, 22)) +
  scale_fill_manual(values = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462"),
                     guide=guide_legend(override.aes=list(shape=21))) +
  scale_color_manual(values = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462")) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "right") + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Host_age_cat_new, 
                   color = Host_age_cat_new),
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Age: p = 0.002, R<sup>2</sup> = 5.2%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1, color = "blue1")

mds_otus_unweighted_points2_age_01week = mds_otus_unweighted_points2 %>% 
  filter(Host_age_cat == "0-1 week")
mds_otus_unweighted_points2_age_01week$Race = factor(mds_otus_unweighted_points2_age_01week$Race, 
                                    levels = c("White", "Black", 
                                               "Asian/Pacific Islander"))

unweight_age_01week <- ggplot(mds_otus_unweighted_points2_age_01week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("0-0.9 week") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.484, R<sup>2</sup> = 1.6%<br>
           Ethnicity: p = 0.516, R<sup>2</sup> = 0.7%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_16week = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new == "1-5.9 weeks")

unweight_age_16week <- ggplot(mds_otus_unweighted_points2_age_16week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("1-5.9 weeks") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.860, R<sup>2</sup> = 1.9%<br>
           Ethnicity: p = 0.764, R<sup>2</sup> = 4.8%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_6week3month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new == "6 weeks-2.9 months")

unweight_age_6week3month <- ggplot(mds_otus_unweighted_points2_age_6week3month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("6 weeks-2.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "right",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.806, R<sup>2</sup> = 2.0%<br>
           Ethnicity: p = 0.676, R<sup>2</sup> = 0.4%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_03month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new2 == "0-2.9 months")

unweight_age_03month <- ggplot(mds_otus_unweighted_points2_age_03month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("0-2.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.562, R<sup>2</sup> = 3.5%<br>
           Ethnicity: p = 0.522, R<sup>2</sup> = 1.7%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_012month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new2 == "0-2.9 months" | Host_age_cat_new == "3-11.9 months")

unweight_age_012month <- ggplot(mds_otus_unweighted_points2_age_012month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("0-11.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.124, R<sup>2</sup> = 1.6%<br>
           Ethnicity: p = 0.478, R<sup>2</sup> = 0.2%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_36month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat2 == "3-6 months")

unweight_age_36month <- ggplot(mds_otus_unweighted_points2_age_36month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("3-5.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 3.7%<br>
           Ethnicity: p = 0.004, R<sup>2</sup> = 1.3%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_69month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat2 == "6-9 months")

unweight_age_69month <- ggplot(mds_otus_unweighted_points2_age_69month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("6-8.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 7.8%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 0.8%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_912month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat2 == "9-12 months")

unweight_age_912month <- ggplot(mds_otus_unweighted_points2_age_912month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("9-11.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 2.3%<br>
           Ethnicity: p = 0.010, R<sup>2</sup> = 0.8%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_312month = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new == "3-11.9 months")

unweight_age_312month <- ggplot(mds_otus_unweighted_points2_age_312month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("3-11.9 months") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(color = "white"),
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 4.6%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 0.7%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_13year = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new == "1-2.9 years")

unweight_age_13year <- ggplot(mds_otus_unweighted_points2_age_13year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("1-2.9 years") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.5)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 2.5%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 1.7%", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_age_312year = mds_otus_unweighted_points2_age %>% 
  filter(Host_age_cat_new == "3-11.9 years")

unweight_age_312year <- ggplot(mds_otus_unweighted_points2_age_312year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
geom_point(aes(shape = Ethnicity, 
                 fill = Race), size=3, color = "black") + 
    scale_shape_manual(values = c(23, 25)) +
  ggtitle("3-11.9 years") +
  scale_fill_manual(values = c("#377eb8", "#f6d128", "#e41a1c"),
                     guide=guide_legend(override.aes=list(shape=23))) +
  scale_color_manual(values = c("#377eb8", "#f6d128", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(1.5)), 
        axis.title.y=element_text(size=rel(1.5)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom",
        plot.title = element_text(size = rel(2.5), face = "bold")) + 
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   color = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.008, R<sup>2</sup> = 2.0%</b><br>
           Ethnicity: p = 0.356, R<sup>2</sup> = 0.6%<br>", 
           x = -Inf, y = Inf, size = 7,
           hjust = 0, vjust = 1)

mds_otus_unweighted_points2_sex = mds_otus_unweighted_one_points2 %>% 
  filter(host_sex != is.na(.))
unweight_sex <- ggplot(mds_otus_unweighted_points2_sex, 
                      aes(x = MDS1, y = MDS2, color = host_sex)) +
  geom_point(size=3) + scale_color_manual(values = c("#4daf4a", "#ff7f00")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_sex, 
                   linetype = host_sex), 
               type = "t", level = 0.95)

mds_otus_unweighted_points2_del = mds_otus_unweighted_one_points2 %>% 
  filter(host_del_route != is.na(.))
unweight_del <- ggplot(mds_otus_unweighted_points2_del, 
                      aes(x = MDS1, y = MDS2, color = host_del_route)) +
  geom_point(size=3) + scale_color_manual(values = c("#8da0cb", "#a6d854")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_del_route, 
                   linetype = host_del_route), 
               type = "t", level = 0.95)

mds_otus_unweighted_points2_feed = mds_otus_unweighted_one_points2 %>% 
  filter(Breastfeeding != is.na(.))
unweight_diet <- ggplot(mds_otus_unweighted_points2_feed, 
                      aes(x = MDS1, y = MDS2, color = Breastfeeding)) +
  geom_point(size=3) + scale_color_manual(values = c("#66c2a5", "#fc8d62", "#b3b3b3")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Breastfeeding, 
                   linetype = Breastfeeding), 
               type = "t", level = 0.95)

unweight_study <- ggplot(mds_otus_unweighted_points2, 
                      aes(x = MDS1, y = MDS2, color = Study)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Dark2') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Unweighted UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Study, 
                   linetype = Study), 
               type = "t", level = 0.95)
```
#### Bray
```{r nmds_bray}
mds_otus_bray<-metaMDS(bray_re_dist, k=2, trymax=499)
mds_otus_bray_points<-mds_otus_bray$points
mds_otus_bray_points2<-merge(x=mds_otus_bray_points, 
                                 y = metadata_re, 
                                 by.x = "row.names", by.y = "SampleID")
write_csv(mds_otus_bray_points2, "bray_mds_full.csv")
```

```{r nmds_bray_plots}
mds_otus_bray_points2 = read_csv("bray_mds_full.csv")
bray_race <- ggplot(mds_otus_bray_points2, 
                      aes(x = MDS1, y = MDS2, color = Race, 
                          shape = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(2.5)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.004, R<sup>2</sup> = 0.08%</b><br>
           Ethnicity: p = 0.634, R<sup>2</sup> = 0.03%<br>
           Age: p = 0.002, R<sup>2</sup> = 6.2%<br>
           Infant diet: p = 0.008, R<sup>2</sup> = 0.1%<br>
           <b>Subject: p = 0.002, R<sup>2</sup> = 36.9%", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age = mds_otus_bray_points2 %>% 
  filter(Host_age_cat != is.na(.)) %>% 
  filter(Host_age_cat != "0-1 week") %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "3-6 years" ~ 
                                        "3-12 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-12 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-6 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-3 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-12 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-3 years"))
mds_otus_bray_points2_age$Host_age_cat_new = factor(mds_otus_bray_points2_age$Host_age_cat_new, 
                                    levels = c("1-6 weeks", "6 weeks-3 months", "3-12 months", "1-3 years", "3-12 years"))
bray_age <- ggplot(mds_otus_bray_points2_age, 
                      aes(x = MDS1, y = MDS2, color = Host_age_cat_new,
                          shape = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#8dd3c7", "#bebada", "#fb8072",
                                "#80b1d3", "#fdb462")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Host_age_cat_new, 
                   linetype = Host_age_cat_new), 
               type = "t", level = 0.95)

mds_otus_bray_points2_age_01week = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "0-1 week")

bray_age_01week <- ggplot(mds_otus_bray_points2_age_01week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.022, R<sup>2</sup> = 2.6%</b><br>
           Ethnicity: p = 0.398, R<sup>2</sup> = 0.9%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_16week = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "1-6 weeks")

bray_age_16week <- ggplot(mds_otus_bray_points2_age_16week, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.904, R<sup>2</sup> = 2.5%<br>
           Ethnicity: p = 0.886, R<sup>2</sup> = 0.9%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_6week3month = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "6 weeks-3 months")

bray_age_6week3month <- ggplot(mds_otus_bray_points2_age_6week3month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.620, R<sup>2</sup> = 2.0%<br>
           Ethnicity: p = 0.532, R<sup>2</sup> = 0.5%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_312month = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "3-12 months")

bray_age_312month <- ggplot(mds_otus_bray_points2_age_312month, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 3.5%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 0.5%</><br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_13year = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "1-3 years")

bray_age_13year <- ggplot(mds_otus_bray_points2_age_13year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.002, R<sup>2</sup> = 2.3%<br>
           Ethnicity: p = 0.002, R<sup>2</sup> = 1.6%</b><br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_age_312year = mds_otus_bray_one_points2 %>% 
  filter(Host_age_cat == "3-12 years")

bray_age_312year <- ggplot(mds_otus_bray_points2_age_312year, 
                      aes(x = MDS1, y = MDS2, color = Race)) +
  geom_point(size=3) + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Race, 
                   linetype = Race), 
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b>Race: p = 0.008, R<sup>2</sup> = 1.8%</b><br>
           Ethnicity: p = 0.094, R<sup>2</sup> = 0.7%<br>", 
           x = -Inf, y = -Inf, size = 5,
           hjust = 0, vjust = 0)

mds_otus_bray_points2_sex = mds_otus_bray_one_points2 %>% 
  filter(host_sex != is.na(.))
bray_sex <- ggplot(mds_otus_bray_points2_sex, 
                      aes(x = MDS1, y = MDS2, color = host_sex)) +
  geom_point(size=3) + scale_color_manual(values = c("#4daf4a", "#ff7f00")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("bray UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_sex, 
                   linetype = host_sex), 
               type = "t", level = 0.95)

mds_otus_bray_points2_del = mds_otus_bray_one_points2 %>% 
  filter(host_del_route != is.na(.))
bray_del <- ggplot(mds_otus_bray_points2_del, 
                      aes(x = MDS1, y = MDS2, color = host_del_route)) +
  geom_point(size=3) + scale_color_manual(values = c("#8da0cb", "#a6d854")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = host_del_route, 
                   linetype = host_del_route), 
               type = "t", level = 0.95)

mds_otus_bray_points2_feed = mds_otus_bray_one_points2 %>% 
  filter(Breastfeeding != is.na(.))
bray_diet <- ggplot(mds_otus_bray_points2_feed, 
                      aes(x = MDS1, y = MDS2, color = Breastfeeding)) +
  geom_point(size=3) + scale_color_manual(values = c("#66c2a6", "#fc8d62", "#b3b3b3")) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("bray UniFrac") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Breastfeeding, 
                   linetype = Breastfeeding), 
               type = "t", level = 0.95)

bray_study <- ggplot(mds_otus_bray_points2, 
                      aes(x = MDS1, y = MDS2, color = Study)) +
  geom_point(size=3) + scale_color_brewer(palette = 'Dark2') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.5))) + 
  ggtitle("Bray-Curtis") +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Study, 
                   linetype = Study), 
               type = "t", level = 0.95)
```

#### Combo
```{r nmds_race_combo}
tiff(file="nmds_plot_race.tif", res=300, width=20, height=9, units="in")
legend1 = get_legend(weight_race + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_race + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_race + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_race_one_combo}
tiff(file="nmds_plot_race_one.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_race_one + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_race_one + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_race_one + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_diet_combo}
tiff(file="nmds_plot_diet.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_diet + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_diet + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_diet + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_age_combo}
tiff(file="nmds_plot_age.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_age + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_age + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_age + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_sex_combo}
tiff(file="nmds_plot_sex.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_sex + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_sex + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_sex + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_del_combo}
tiff(file="nmds_plot_del.tif", res=300, width=18, height=9, units="in")
legend1 = get_legend(weight_del + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_del + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(unweight_del + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

```{r nmds_study_combo}
tiff(file="nmds_plot_study.tif", res=150, width=18, height=9, units="in")
legend1 = get_legend(weight_study + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(weight_study + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(bray_study + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

##Alpha diversity
###All samples
```{r alpha_div}
faith = read.table("alpha_out/faith_pd.tsv", header=T)
otus = read.table("alpha_out/observed_features.tsv", header = T)
shannon = read.table("alpha_out/shannon.tsv", header = T)
pielou = read.table("alpha_out/pielou_e.tsv", header=T, fill = T)
chao = read.table("alpha_out/chao1.tsv", header=T)

faith2 = faith %>% rownames_to_column("SampleID")
otus2 = otus %>% rownames_to_column("SampleID")
shannon2 = shannon %>% rownames_to_column("SampleID")
pielou2 = pielou %>% rownames_to_column("SampleID")
chao2 = chao %>% rownames_to_column("SampleID")

alpha_all = inner_join(metadata_re, faith2, by = "SampleID") %>% 
  inner_join(otus2, by = "SampleID") %>% 
  inner_join(shannon2, by = "SampleID") %>% 
  inner_join(pielou2, by = "SampleID") %>% 
  inner_join(chao2, by = "SampleID") %>% 
  filter(Race != "Other") %>% 
  filter(Ethnicity != "Other")

f = lmer(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(f)
Anova(f)
summary(glht(f,linfct=mcp(Ethnicity="Tukey")))
summary(glht(f,linfct=mcp(Host_age_cat="Tukey")))

o = lmer(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(o)
Anova(o)
summary(glht(o,linfct=mcp(Race="Tukey")))
summary(glht(o,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(o,linfct=mcp(host_sex="Tukey")))

s = lmer(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(s)
Anova(s)
summary(glht(s,linfct=mcp(Race="Tukey")))
summary(glht(s,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(s,linfct=mcp(Breastfeeding="Tukey")))

p = lmer(pielou_evenness ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(p)
Anova(p)
summary(glht(p,linfct=mcp(Race="Tukey")))
summary(glht(p,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(p,linfct=mcp(Breastfeeding="Tukey")))

c = lmer(chao1 ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_all, na.action = na.omit)
summary(c)
Anova(c)
summary(glht(c,linfct=mcp(Race="Tukey")))
summary(glht(c,linfct=mcp(Host_age_cat="Tukey")))
summary(glht(c,linfct=mcp(host_sex="Tukey")))
summary(glht(c,linfct=mcp(Breastfeeding="Tukey")))
```

#### Graphs
Each comparison of interest was graphed, using the full dataset.
```{r alpha_div_race_graphs}
plot1_alpha = ggboxplot(alpha_all, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1_alpha = ggpar(plot1_alpha, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.082", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_alpha = ggboxplot(alpha_all, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_alpha = ggpar(plot2_alpha, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("***", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.002", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_alpha = ggboxplot(alpha_all, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_alpha = ggpar(plot3_alpha, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.008", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_race.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1_alpha, plot2_alpha, plot3_alpha, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_eth_graphs}
plot1_alphae = ggboxplot(alpha_all, x = "Ethnicity", 
                  y = "faith_pd", color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1_alphae = ggpar(plot1_alphae, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("*", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.031", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_alphae = ggboxplot(alpha_all, x = "Ethnicity", 
                  y = "observed_features", color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_alphae = ggpar(plot2_alphae, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.092", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_alphae = ggboxplot(alpha_all, x = "Ethnicity", 
                  y = "shannon_entropy", color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_alphae = ggpar(plot3_alphae, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "p = 0.283", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_ethnicity.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1_alphae, plot2_alphae, plot3_alphae, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_age_graphs}
alpha_all_age = alpha_all %>% filter(!is.na(Host_age_cat)) %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "0-1 week" ~ "0-0.9 week",
                                      Host_age_cat == "1-6 weeks" ~ "1-5.9 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~ "6 weeks-2.9 months",
                                      Host_age_cat == "3-12 months" ~ "3-11.9 months",
                                      Host_age_cat == "1-3 years" ~ "1-2.9 years",
                                      Host_age_cat == "3-6 years" ~ "3-5.9 years",
                                      Host_age_cat == "6-12 years" ~ "6-11.9 years"))

alpha_all_age$Host_age_cat_new = factor(alpha_all_age$Host_age_cat_new, 
                                    levels = c("0-0.9 week", "1-5.9 weeks", "6 weeks-2.9 months", "3-11.9 months", "1-2.9 years", "3-5.9 years", "6-11.9 years"))

plot1 = ggboxplot(alpha_all_age, x = "Host_age_cat_new", 
                  y = "faith_pd", color = "Host_age_cat_new", 
                  palette = "Set3", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("0-0.9 week", "1-2.9 years"),
                                        c("0-0.9 week", "3-5.9 years"),
                                        c("0-0.9 week", "6-11.9 years"),
                                        c("1-5.9 weeks", 
                                          "3-11.9 months"),
                                        c("1-5.9 weeks", "1-2.9 years"),
                                        c("1-5.9 weeks", "3-5.9 years"),
                                        c("1-5.9 weeks", "6-11.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "3-11.9 months"),
                                        c("6 weeks-2.9 months",
                                          "1-2.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "3-5.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "6-11.9 years"),
                                        c("3-11.9 months","1-2.9 years"),
                                        c("3-11.9 months","3-5.9 years"),
                                        c("3-11.9 months", 
                                          "6-11.9 years")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0000000001, 
                                            0.01, 0.05, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Age: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_age, x = "Host_age_cat_new", 
                  y = "observed_features", color = "Host_age_cat_new", 
                  palette = "Set3", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("0-0.9 week", "1-5.9 weeks"),
                                        c("0-0.9 week", 
                                          "6 weeks-2.9 months"),
                                        c("0-0.9 week", "1-2.9 years"),
                                        c("0-0.9 week", "3-5.9 years"),
                                        c("0-0.9 week", "6-11.9 years"),
                                        c("1-5.9 weeks", 
                                          "3-11.9 months"),
                                        c("1-5.9 weeks", "1-2.9 years"),
                                        c("1-5.9 weeks", "3-5.9 years"),
                                        c("1-5.9 weeks", "6-11.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "3-11.9 months"),
                                        c("6 weeks-2.9 months",
                                          "1-2.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "3-5.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "6-11.9 years"),
                                        c("3-11.9 months","1-2.9 years"),
                                        c("3-11.9 months","3-5.9 years"),
                                        c("3-11.9 months", 
                                          "6-11.9 years")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.05, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Age: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_age, x = "Host_age_cat_new", 
                  y = "shannon_entropy", color = "Host_age_cat_new", 
                  palette = "Set3", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("0-0.9 week", "1-5.9 weeks"),
                                        c("0-0.9 week", "3-11.9 months"),
                                        c("0-0.9 week", "1-2.9 years"),
                                        c("0-0.9 week", "3-5.9 years"),
                                        c("0-0.9 week", "6-11.9 years"),
                                        c("1-5.9 weeks", 
                                          "6 weeks-2.9 months"),
                                        c("1-5.9 weeks", 
                                          "3-11.9 months"),
                                        c("1-5.9 weeks", "1-2.9 years"),
                                        c("1-5.9 weeks", "3-5.9 years"),
                                        c("1-5.9 weeks", "6-11.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "3-11.9 months"),
                                        c("6 weeks-2.9 months",
                                          "1-2.9 years"),
                                        c("6 weeks-2.9 months", 
                                          "3-5.9 years"),
                                        c("6 weeks-2.9 months",
                                          "6-11.9 years"),
                                        c("3-11.9 months","1-2.9 years"),
                                        c("3-11.9 months",
                                          "3-5.9 years")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0000000001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Age: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_age.tif", 
     res=300, width=15, height=6, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_sex_graphs}
alpha_all_sex = alpha_all %>% dplyr::filter(host_sex != is.na(.))

plot1 = ggboxplot(alpha_all_sex, x = "host_sex", 
                  y = "faith_pd", color = "host_sex", 
                  palette = c("#4daf4a", "#ff7f00"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Sex: p = 0.091", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_sex, x = "host_sex", 
                  y = "observed_features", color = "host_sex", 
                  palette = c("#4daf4a", "#ff7f00"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Female", "Male")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0001, 0.001, 
                                            0.05, 0.999, 1),
                                        symbols = c("*", "*", "*", 
                                                    "*", "ns")))  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Sex: p = 0.033", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_sex, x = "host_sex", 
                  y = "shannon_entropy", color = "host_sex", 
                  palette = c("#4daf4a", "#ff7f00"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Sex: p = 0.593", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_sex.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_del_graphs}
alpha_all_del = alpha_all %>% dplyr::filter(host_del_route != is.na(.))

plot1 = ggboxplot(alpha_all_del, x = "host_del_route", 
                  y = "faith_pd", color = "host_del_route", 
                  palette = c("#8da0cb", "#a6d854"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Delivery: p = 0.543", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_del, x = "host_del_route", 
                  y = "observed_features", color = "host_del_route", 
                  palette = c("#8da0cb", "#a6d854"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Delivery: p = 0.559", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_del, x = "host_del_route", 
                  y = "shannon_entropy", color = "host_del_route", 
                  palette = c("#8da0cb", "#a6d854"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Delivery: p = 0.979", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_del.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

```{r alpha_div_feed_graphs}
alpha_all_fed = alpha_all %>% dplyr::filter(Breastfeeding != is.na(.))

plot1 = ggboxplot(alpha_all_fed, x = "Breastfeeding", 
                  y = "faith_pd", color = "Breastfeeding", 
                  palette = c("#66c2a6", "#fc8d62", "#b3b3b3"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity") 
plot1 = ggpar(plot1, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Infant diet: p = 0.091", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2 = ggboxplot(alpha_all_fed, x = "Breastfeeding", 
                  y = "observed_features", color = "Breastfeeding", 
                  palette = c("#66c2a6", "#fc8d62", "#b3b3b3"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2 = ggpar(plot2, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Infant diet: p = 0.066", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot3 = ggboxplot(alpha_all_fed, x = "Breastfeeding", 
                  y = "shannon_entropy", color = "Breastfeeding", 
                  palette = c("#66c2a6", "#fc8d62", "#b3b3b3"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3 = ggpar(plot3, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Formula fed", 
                                          "Mixed feeding"), 
                                        c("Formula fed", 
                                          "Exclusively breastfed"), 
                                        c("Mixed feeding", 
                                          "Exclusively breastfed")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0001, 0.001, 
                                            0.05, 0.999, 1),
                                        symbols = c("***", 
                                                    "*", "*", 
                                                    "*", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Infant diet: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

tiff(file="alpha_combined_feed.tif", 
     res=300, width=15, height=4, units="in")
ggarrange(plot1, plot2, plot3, nrow = 1, ncol = 3, 
          common.legend = T, align = "h", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```
###By age category
```{r alpha_agecat}
alpha_zero_one = alpha_all %>% 
  filter(Host_age_cat == "0-1 week")
alpha_one_six = alpha_all %>% 
  filter(Host_age_cat == "1-6 weeks")
alpha_six_three = alpha_all %>% 
  filter(Host_age_cat == "6 weeks-3 months")
alpha_three_twelve = alpha_all %>% 
  filter(Host_age_cat == "3-12 months")
alpha_one_three = alpha_all %>% 
  filter(Host_age_cat == "1-3 years")
alpha_threey_twelvey = alpha_all %>% 
  filter(Host_age_cat == "3-6 years" | 
           Host_age_cat == "6-12 years")
alpha_onem_threem = alpha_all %>% 
  filter(Host_age_cat == "0-1 week" |
           Host_age_cat == "1-6 weeks" |
           Host_age_cat == "6 weeks-3 months")

#0-1 weeks
f01 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(f01)
Anova(f01)
f01r_df = broom::tidy(summary(glht(f01,linfct=mcp(Race="Tukey"))))
f01e_df = broom::tidy(summary(glht(f01,linfct=mcp(Ethnicity="Tukey"))))

o01 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(o01)
Anova(o01)
o01r_df = broom::tidy(summary(glht(o01,linfct=mcp(Race="Tukey"))))
o01e_df = broom::tidy(summary(glht(o01,linfct=mcp(Ethnicity="Tukey"))))

s01 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(s01)
Anova(s01)
s01r_df = broom::tidy(summary(glht(s01,linfct=mcp(Race="Tukey"))))
s01e_df = broom::tidy(summary(glht(s01,linfct=mcp(Ethnicity="Tukey"))))

p01 = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(p01)
Anova(p01)
p01r_df = broom::tidy(summary(glht(p01,linfct=mcp(Race="Tukey"))))
p01e_df = broom::tidy(summary(glht(p01,linfct=mcp(Ethnicity="Tukey"))))

c01 = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_zero_one, na.action = na.omit)
summary(c01)
Anova(c01)
c01r_df = broom::tidy(summary(glht(c01,linfct=mcp(Race="Tukey"))))
c01e_df = broom::tidy(summary(glht(c01,linfct=mcp(Ethnicity="Tukey"))))

#1-6 weeks
f16 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(f16)
Anova(f16)
summary(glht(f16,linfct=mcp(Breastfeeding="Tukey")))
f16r_df = broom::tidy(summary(glht(f16,linfct=mcp(Race="Tukey"))))
f16e_df = broom::tidy(summary(glht(f16,linfct=mcp(Ethnicity="Tukey"))))

o16 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(o16)
Anova(o16)
o16r_df = broom::tidy(summary(glht(o16,linfct=mcp(Race="Tukey"))))
o16e_df = broom::tidy(summary(glht(o16,linfct=mcp(Ethnicity="Tukey"))))

s16 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(s16)
Anova(s16)
summary(glht(s16,linfct=mcp(Breastfeeding="Tukey")))
s16r_df = broom::tidy(summary(glht(s16,linfct=mcp(Race="Tukey"))))
s16e_df = broom::tidy(summary(glht(s16,linfct=mcp(Ethnicity="Tukey"))))

p16 = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(p16)
Anova(p16)
summary(glht(p16,linfct=mcp(Breastfeeding="Tukey")))
p16r_df = broom::tidy(summary(glht(p16,linfct=mcp(Race="Tukey"))))
p16e_df = broom::tidy(summary(glht(p16,linfct=mcp(Ethnicity="Tukey"))))

c16 = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_one_six, na.action = na.omit)
summary(c16)
Anova(c16)
c16r_df = broom::tidy(summary(glht(c16,linfct=mcp(Race="Tukey"))))
c16e_df = broom::tidy(summary(glht(c16,linfct=mcp(Ethnicity="Tukey"))))

#6 weeks - 3 months
f63 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(f63)
Anova(f63)
f63r_df = broom::tidy(summary(glht(f63,linfct=mcp(Race="Tukey"))))
f63e_df = broom::tidy(summary(glht(f63,linfct=mcp(Ethnicity="Tukey"))))

o63 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(o63)
Anova(o63)
o63r_df = broom::tidy(summary(glht(o63,linfct=mcp(Race="Tukey"))))
o63e_df = broom::tidy(summary(glht(o63,linfct=mcp(Ethnicity="Tukey"))))

s63 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(s63)
Anova(s63)
summary(glht(s63,linfct=mcp(Breastfeeding="Tukey")))
s63r_df = broom::tidy(summary(glht(s63,linfct=mcp(Race="Tukey"))))
s63e_df = broom::tidy(summary(glht(s63,linfct=mcp(Ethnicity="Tukey"))))

p63 = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(p63)
Anova(p63)
summary(glht(p63,linfct=mcp(Breastfeeding="Tukey")))
p63r_df = broom::tidy(summary(glht(p63,linfct=mcp(Race="Tukey"))))
p63e_df = broom::tidy(summary(glht(p63,linfct=mcp(Ethnicity="Tukey"))))

c63 = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_six_three, na.action = na.omit)
summary(c63)
Anova(c63)
c63r_df = broom::tidy(summary(glht(c63,linfct=mcp(Race="Tukey"))))
c63e_df = broom::tidy(summary(glht(c63,linfct=mcp(Ethnicity="Tukey"))))

#0-3 months
f1m3m = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_onem_threem, na.action = na.omit)
summary(f1m3m)
Anova(f1m3m)
summary(glht(f1m3m,linfct=mcp(Breastfeeding="Tukey")))
f1m3mr_df = broom::tidy(summary(glht(f1m3m,linfct=mcp(Race="Tukey"))))
f1m3me_df = broom::tidy(summary(glht(f1m3m,linfct=mcp(Ethnicity="Tukey"))))

o1m3m = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_onem_threem, na.action = na.omit)
summary(o1m3m)
Anova(o1m3m)
o1m3mr_df = broom::tidy(summary(glht(o1m3m,linfct=mcp(Race="Tukey"))))
o1m3me_df = broom::tidy(summary(glht(o1m3m,linfct=mcp(Ethnicity="Tukey"))))

s1m3m = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_onem_threem, na.action = na.omit)
summary(s1m3m)
Anova(s1m3m)
summary(glht(s1m3m,linfct=mcp(Breastfeeding="Tukey")))
s1m3mr_df = broom::tidy(summary(glht(s1m3m,linfct=mcp(Race="Tukey"))))
s1m3me_df = broom::tidy(summary(glht(s1m3m,linfct=mcp(Ethnicity="Tukey"))))

p1m3m = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_onem_threem, na.action = na.omit)
summary(p1m3m)
Anova(p1m3m)
summary(glht(p1m3m,linfct=mcp(host_sex="Tukey")))
summary(glht(p1m3m,linfct=mcp(Breastfeeding="Tukey")))
p1m3mr_df = broom::tidy(summary(glht(p1m3m,linfct=mcp(Race="Tukey"))))
p1m3me_df = broom::tidy(summary(glht(p1m3m,linfct=mcp(Ethnicity="Tukey"))))

c1m3m = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_onem_threem, na.action = na.omit)
summary(c1m3m)
Anova(c1m3m)
c1m3mr_df = broom::tidy(summary(glht(c1m3m,linfct=mcp(Race="Tukey"))))
c1m3me_df = broom::tidy(summary(glht(c1m3m,linfct=mcp(Ethnicity="Tukey"))))

#3-12 months
f312 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(f312)
Anova(f312)
summary(glht(f312,linfct=mcp(Breastfeeding="Tukey")))
summary(glht(f312,linfct=mcp(Race="Tukey")))
f312r_df = broom::tidy(summary(glht(f312,linfct=mcp(Race="Tukey"))))
f312e_df = broom::tidy(summary(glht(f312,linfct=mcp(Ethnicity="Tukey"))))

o312 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(o312)
Anova(o312)
summary(glht(o312,linfct=mcp(Race="Tukey")))
summary(glht(o312,linfct=mcp(Breastfeeding="Tukey")))
o312r_df = broom::tidy(summary(glht(o312,linfct=mcp(Race="Tukey"))))
o312e_df = broom::tidy(summary(glht(o312,linfct=mcp(Ethnicity="Tukey"))))

s312 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(s312)
Anova(s312)
summary(glht(s312,linfct=mcp(Race="Tukey")))
summary(glht(s312,linfct=mcp(Breastfeeding="Tukey")))
s312r_df = broom::tidy(summary(glht(s312,linfct=mcp(Race="Tukey"))))
s312e_df = broom::tidy(summary(glht(s312,linfct=mcp(Ethnicity="Tukey"))))

c312 = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(c312)
Anova(c312)
summary(glht(c312,linfct=mcp(Race="Tukey")))
summary(glht(c312,linfct=mcp(Breastfeeding="Tukey")))
c312r_df = broom::tidy(summary(glht(c312,linfct=mcp(Race="Tukey"))))
c312e_df = broom::tidy(summary(glht(c312,linfct=mcp(Ethnicity="Tukey"))))

p312 = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_three_twelve, na.action = na.omit)
summary(p312)
Anova(p312)
summary(glht(p312,linfct=mcp(Race="Tukey")))
summary(glht(p312,linfct=mcp(Breastfeeding="Tukey")))
p312r_df = broom::tidy(summary(glht(p312,linfct=mcp(Race="Tukey"))))
p312e_df = broom::tidy(summary(glht(p312,linfct=mcp(Ethnicity="Tukey"))))

#1-3 years
f13 = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(f13)
Anova(f13)
f13a = lmer(faith_pd ~ Race + Ethnicity + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(f13a)
Anova(f13a)
summary(glht(f13,linfct=mcp(Race="Tukey")))
summary(glht(f13,linfct=mcp(Ethnicity="Tukey")))
f13r_df = broom::tidy(summary(glht(f13a,linfct=mcp(Race="Tukey"))))
f13e_df = broom::tidy(summary(glht(f13,linfct=mcp(Ethnicity="Tukey"))))

o13 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(o13)
Anova(o13)
o13a = lmer(observed_features ~ Race + Ethnicity + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(o13a)
Anova(o13a)
summary(glht(o13,linfct=mcp(Race="Tukey")))
summary(glht(o13,linfct=mcp(Ethnicity="Tukey")))
o13r_df = broom::tidy(summary(glht(o13a,linfct=mcp(Race="Tukey"))))
o13e_df = broom::tidy(summary(glht(o13,linfct=mcp(Ethnicity="Tukey"))))

s13 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(s13)
Anova(s13)
s13a = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(s13a)
Anova(s13a)
summary(glht(s13,linfct=mcp(Race="Tukey")))
summary(glht(s13,linfct=mcp(Ethnicity="Tukey")))
s13r_df = broom::tidy(summary(glht(s13a,linfct=mcp(Race="Tukey"))))
s13e_df = broom::tidy(summary(glht(s13,linfct=mcp(Ethnicity="Tukey"))))

p13 = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(p13)
Anova(p13)
p13a = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(p13a)
Anova(p13a)
summary(glht(p13,linfct=mcp(Race="Tukey")))
summary(glht(p13,linfct=mcp(Ethnicity="Tukey")))
p13r_df = broom::tidy(summary(glht(p13a,linfct=mcp(Race="Tukey"))))
p13e_df = broom::tidy(summary(glht(p13,linfct=mcp(Ethnicity="Tukey"))))

c13 = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(c13)
Anova(c13)
c13a = lmer(chao1 ~ Race + Ethnicity + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_one_three, na.action = na.omit)
summary(c13a)
Anova(c13a)
summary(glht(c13,linfct=mcp(Race="Tukey")))
summary(glht(c13,linfct=mcp(Ethnicity="Tukey")))
c13r_df = broom::tidy(summary(glht(c13a,linfct=mcp(Race="Tukey"))))
c13e_df = broom::tidy(summary(glht(c13,linfct=mcp(Ethnicity="Tukey"))))

#3-12 years
f312y = lmer(faith_pd ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(f312y)
Anova(f312y)
f312yr_df = broom::tidy(summary(glht(f312y,linfct=mcp(Race="Tukey"))))
f312ye_df = broom::tidy(summary(glht(f312y,linfct=mcp(Ethnicity="Tukey"))))

o312y = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(o312y)
Anova(o312y)
o312yr_df = broom::tidy(summary(glht(o312y,linfct=mcp(Race="Tukey"))))
o312ye_df = broom::tidy(summary(glht(o312y,linfct=mcp(Ethnicity="Tukey"))))

s312y = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(s312y)
Anova(s312y)
s312yr_df = broom::tidy(summary(glht(s312y,linfct=mcp(Race="Tukey"))))
s312ye_df = broom::tidy(summary(glht(s312y,linfct=mcp(Ethnicity="Tukey"))))

p312y = lmer(pielou_evenness ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(p312y)
Anova(p312y)
p312yr_df = broom::tidy(summary(glht(p312y,linfct=mcp(Race="Tukey"))))
p312ye_df = broom::tidy(summary(glht(p312y,linfct=mcp(Ethnicity="Tukey"))))

c312y = lmer(chao1 ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study), data=alpha_threey_twelvey, na.action = na.omit)
summary(c312y)
Anova(c312y)
c312yr_df = broom::tidy(summary(glht(c312y,linfct=mcp(Race="Tukey"))))
c312ye_df = broom::tidy(summary(glht(c312y,linfct=mcp(Ethnicity="Tukey"))))

#0 weeks-3 months
alpha_zero_three = alpha_all %>% 
  filter(Host_age_cat == "0-1 week" | 
           Host_age_cat == "1-6 weeks" | 
           Host_age_cat == "6 weeks-3 months")

o03 = lmer(observed_features ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_zero_three, na.action = na.omit)
summary(o03)
Anova(o03)

s03 = lmer(shannon_entropy ~ Race + Ethnicity + host_sex + host_del_route + Breastfeeding + (1|Study) + (1|host_subject_id), data=alpha_zero_three, na.action = na.omit)
summary(s03)
Anova(s03)
```

####Graphs
##### Boxplots
```{r alpha_div_agerace_graphs}
alpha_all_age = alpha_all %>% dplyr::filter(!is.na(Host_age_cat)) %>% 
  mutate(Host_age_cat_new = case_when(Host_age_cat == "0-1 week" ~ 
                                        "0-0.9 week", 
                                      Host_age_cat == "3-6 years" ~ 
                                        "3-11.9 years", 
                                      Host_age_cat == "6-12 years" ~ 
                                        "3-11.9 years",
                                      Host_age_cat == "1-6 weeks" ~
                                        "1-5.9 weeks",
                                      Host_age_cat == "6 weeks-3 months" ~
                                        "6 weeks-2.9 months",
                                      Host_age_cat == "3-12 months" ~
                                        "3-11.9 months",
                                      Host_age_cat == "1-3 years" ~
                                        "1-2.9 years"))

alpha_all_age$Host_age_cat_new = factor(alpha_all_age$Host_age_cat_new, 
                                    levels = c("0-0.9 week", "1-5.9 weeks", "6 weeks-2.9 months", "3-11.9 months", "1-2.9 years", "3-11.9 years"))

alpha_all_age_filtered = alpha_all_age %>% filter(Host_age_cat != "0-1 week")  

alpha_all_age_01week = alpha_all_age %>% 
  filter(Host_age_cat_new == "0-0.9 week")

alpha_all_age_16week = alpha_all_age %>% 
  filter(Host_age_cat_new == "1-5.9 weeks")

alpha_all_age_6week3month = alpha_all_age %>% 
  filter(Host_age_cat_new == "6 weeks-2.9 months")

alpha_all_age_312month = alpha_all_age %>% 
  filter(Host_age_cat_new == "3-11.9 months")

alpha_all_age_13year = alpha_all_age %>% 
  filter(Host_age_cat_new == "1-2.9 years")

alpha_all_age_312year = alpha_all_age %>% 
  filter(Host_age_cat_new == "3-11.9 years")

alpha_all_age_filtered2 = distinct(alpha_all_age_filtered, Host_age_cat_new)
alpha_all_age_filtered2$Host_age_cat_new = factor(alpha_all_age_filtered2$Host_age_cat_new, 
                                    levels = c("1-5.9 weeks", "6 weeks-2.9 months", "3-11.9 months", "1-2.9 years", "3-11.9 years"))
alpha_all_age_filtered2$yloc = max(alpha_all_age_filtered$shannon_entropy + 0.5)
alpha_all_age_filtered2$label = c("C", "C", "A", "A", "B")

alpha_all_age_graph = ggplot(alpha_all_age_filtered,
                      aes(x = Host_age_cat_new, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#8dd3c7", "#bebada", "#fb8072", 
                               "#80b1d3", "#fdb462"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("Shannon Diversity") +
  ylim(0, max(alpha_all_age_filtered$shannon_entropy + 1.25)) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(2.4)),
        axis.ticks.x=element_blank(),
        plot.title=element_text(size=rel(2), face = "bold")) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = alpha_all_age_filtered2, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Age: p < 0.001</b></i>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")


alpha_16week = ggplot(alpha_all_age_16week,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#8dd3c7",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#8dd3c7",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylab("All Age Categories\nShannon Diversity") + 
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month_asvs = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#bebada",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#bebada",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#bebada",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#bebada",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")
  
alpha_312month_asvs = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#fb8072",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#fb8072",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8,
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_312month = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#fb8072",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#fb8072",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8,
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_13year_asvs = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#80b1d3",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#80b1d3",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_13year = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#80b1d3",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#80b1d3",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_312year_asvs = ggplot(alpha_all_age_312year,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#fdb462",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#fdb462",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_312year = ggplot(alpha_all_age_312year,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = "#fdb462",
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = "#fdb462",
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Race),
                  color = "black") + 
  scale_shape_manual(values = c(16, 17, 15)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month_race = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("6 weeks - 3 months\nShannon Diversity") +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month_race_asvs = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("6 weeks - 3 months\nObserved ASVs") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_6week3month_race_faith = ggplot(alpha_all_age_6week3month,
                      aes(x = Race, y = faith_pd)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("6 weeks - 3 months\nFaith's PD") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")
  
alpha_312month_race = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black") +
  ylab("3 - 12 months\nShannon Diversity") +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_312month_race_asvs = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black") +
  ylab("3 - 12 months\nObserved ASVs") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_312month_race_faith = ggplot(alpha_all_age_312month,
                      aes(x = Race, y = faith_pd)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black") +
  ylab("3 - 12 months\nFaith's PD") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")

alpha_13year_race = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("1 - 3 years\nShannon Diversity") +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_13year_race_asvs = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = observed_features)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("1 - 3 years\nObserved ASVs") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_13year_race_faith = ggplot(alpha_all_age_13year,
                      aes(x = Race, y = faith_pd)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
                  color = "black") +
  ylab("1 - 3 years\nFaith's PD") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(2)),
        axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

alpha_6week3month_eth = ggplot(alpha_all_age_6week3month,
                      aes(x = Ethnicity, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("light gray", "dark gray"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("light gray", "dark gray"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Ethnicity),
                  color = "black", fill = "black") + 
  scale_shape_manual(values = c(23, 25)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none")
  
alpha_312month_eth = ggplot(alpha_all_age_312month,
                      aes(x = Ethnicity, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("light gray", "dark gray"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("light gray", "dark gray"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Ethnicity),
                  color = "black", fill = "black") + 
  scale_shape_manual(values = c(23, 25)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") 

alpha_13year_eth = ggplot(alpha_all_age_13year,
                      aes(x = Ethnicity, y = shannon_entropy)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("light gray", "dark gray"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("light gray", "dark gray"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              aes(shape = Ethnicity),
                  color = "black", fill = "black") + 
  scale_shape_manual(values = c(23, 25)) +
  ylim(0, 7.8) +
  theme_classic() + 
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 1, size = 8, 
                     comparisons = list(c("Hispanic", "Non Hispanic")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.01, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

tiff(file="alpha_combined_age_race.tif", 
     res=300, width=15, height=6, units="in")
alpha_age_race_plot = ggarrange(alpha_16week, alpha_6week3month, alpha_312month,
          alpha_13year, alpha_312year, nrow = 1, ncol = 5, 
          align = "v")
alpha_age_race_plot
dev.off()
```

#####Dot-whisker plots
```{r alpha_dwp}
#0-1 weeks
f01r_df2 = f01r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f01e_df2 = f01e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f01r_dw = dwplot(f01r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        axis.title.y=element_text(size=rel(2),
                                  hjust = 0),
        legend.position = "none") + 
  ylab("0-1 weeks") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black",
                              "Asian/PI\n- White",
                              "Black\n- White"))
f01e_dw = dwplot(f01e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic"))

o01r_df2 = o01r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o01e_df2 = o01e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o01r_dw = dwplot(o01r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))
o01e_dw = dwplot(o01e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))

s01r_df2 = s01r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s01e_df2 = s01e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s01r_dw = dwplot(s01r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity"))
s01e_dw = dwplot(s01e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity"))

p01r_df2 = p01r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p01e_df2 = p01e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p01r_dw = dwplot(p01r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness"))
p01e_dw = dwplot(p01e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness"))

c01r_df2 = c01r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c01e_df2 = c01e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c01r_dw = dwplot(c01r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1"))
c01e_dw = dwplot(c01e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1"))

legend_models = rbind(f01r_df2, s01r_df2, p01r_df2, o01r_df2, c01r_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  xlab("Estimate") + scale_color_manual(values = c("#418204", "#834600",
                                                   "#b05b00", "#4a228c",
                                                   "#873eff"),
                                        labels = c("Faith's PD", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness", 
                                                   "Observed ASVs", "Chao 1")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_01weeks.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f01r_dw, s01r_dw, p01r_dw, o01r_dw, c01r_dw, 
          f01e_dw, s01e_dw, p01e_dw, o01e_dw, c01e_dw,
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

#1-6 weeks
f16r_df2 = f16r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f16e_df2 = f16e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f16r_dw = dwplot(f16r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        axis.title.y=element_text(size=rel(2.5),
                                  hjust = 0),
        legend.position = "none") + 
  ylab("1-6 weeks") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White"))
f16e_dw = dwplot(f16e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic"))

o16r_df2 = o16r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o16e_df2 = o16e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o16r_dw = dwplot(o16r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))
o16e_dw = dwplot(o16e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))

s16r_df2 = s16r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s16e_df2 = s16e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s16r_dw = dwplot(s16r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity"))
s16e_dw = dwplot(s16e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity"))

p16r_df2 = p16r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p16e_df2 = p16e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p16r_dw = dwplot(p16r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness"))
p16e_dw = dwplot(p16e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness"))

c16r_df2 = c16r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c16e_df2 = c16e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c16r_dw = dwplot(c16r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1"))
c16e_dw = dwplot(c16e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1"))

legend_models = rbind(f16r_df2, s16r_df2, p16r_df2, o16r_df2, c16r_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  xlab("Estimate") + scale_color_manual(values = c("#418204", "#834600",
                                                   "#b05b00", "#4a228c",
                                                   "#873eff"),
                                        labels = c("Faith's PD", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness", 
                                                   "Observed ASVs", "Chao 1")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_16weeks.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f16r_dw, s16r_dw, p16r_dw, o16r_dw, c16r_dw, 
          f16e_dw, s16e_dw, p16e_dw, o16e_dw, c16e_dw,
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

#6 weeks - 3 months
f63r_df2 = f63r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f63e_df2 = f63e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f63r_dw = dwplot(f63r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  ggtitle("6 weeks-3 months") +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.374", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

f63e_dw = dwplot(f63e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.758", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

o63r_df2 = o63r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o63e_df2 = o63e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o63r_dw = dwplot(o63r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))+
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.209", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

o63e_dw = dwplot(o63e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.973", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

s63r_df2 = s63r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s63e_df2 = s63e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s63r_dw = dwplot(s63r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.682", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

s63e_dw = dwplot(s63e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.839", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

p63r_df2 = p63r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p63e_df2 = p63e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p63r_dw = dwplot(p63r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.746", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

p63e_dw = dwplot(p63e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.533", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

c63r_df2 = c63r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c63e_df2 = c63e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c63r_dw = dwplot(c63r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.186", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

c63e_dw = dwplot(c63e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.977", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

legend_models = rbind(f63r_df2, s63r_df2, p63r_df2, o63r_df2, c63r_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  xlab("Estimate") + scale_color_manual(values = c("#418204", "#834600",
                                                   "#b05b00", "#4a228c",
                                                   "#873eff"),
                                        labels = c("Faith's PD", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness", 
                                                   "Observed ASVs", "Chao 1")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_6weeks3months.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f63r_dw, s63r_dw, p63r_dw, o63r_dw, c63r_dw, 
          f63e_dw, s63e_dw, p63e_dw, o63e_dw, c63e_dw,
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

#0-3 months
f1m3mr_df2 = f1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f1m3me_df2 = f1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f1m3mr_dw = dwplot(f1m3mr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  ggtitle("6 weeks-3 months") +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.718", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

f1m3me_dw = dwplot(f1m3me_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.384", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

o1m3mr_df2 = o1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o1m3me_df2 = o1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o1m3mr_dw = dwplot(o1m3mr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))+
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.448", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

o1m3me_dw = dwplot(o1m3me_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.670", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

s1m3mr_df2 = s1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s1m3me_df2 = s1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s1m3mr_dw = dwplot(s1m3mr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.642", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

s1m3me_dw = dwplot(s1m3me_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.406", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

p1m3mr_df2 = p1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p1m3me_df2 = p1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p1m3mr_dw = dwplot(p1m3mr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.747", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

p1m3me_dw = dwplot(p1m3me_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.548", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

c1m3mr_df2 = c1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c1m3me_df2 = c1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c1m3mr_dw = dwplot(c1m3mr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.311", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

c1m3me_dw = dwplot(c1m3me_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.668", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

legend_models = rbind(f1m3mr_df2, s1m3mr_df2, p1m3mr_df2, o1m3mr_df2, c1m3mr_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  xlab("Estimate") + scale_color_manual(values = c("#418204", "#834600",
                                                   "#b05b00", "#4a228c",
                                                   "#873eff"),
                                        labels = c("Faith's PD", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness", 
                                                   "Observed ASVs", "Chao 1")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_6weeks3months.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f1m3mr_dw, s1m3mr_dw, p1m3mr_dw, o1m3mr_dw, c1m3mr_dw, 
          f1m3me_dw, s1m3me_dw, p1m3me_dw, o1m3me_dw, c1m3me_dw,
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

#3-12 months
f312r_df2 = f312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f312e_df2 = f312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f312r_dw = dwplot(f312r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  ggtitle("3-12 months") + 
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.233", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

f312e_dw = dwplot(f312e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.880", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

o312r_df2 = o312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o312e_df2 = o312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o312r_dw = dwplot(o312r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.005</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

o312e_dw = dwplot(o312e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.914", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

s312r_df2 = s312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s312e_df2 = s312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s312r_dw = dwplot(s312r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.006</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

s312e_dw = dwplot(s312e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.857", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

p312r_df2 = p312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p312e_df2 = p312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p312r_dw = dwplot(p312r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.026</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

p312e_dw = dwplot(p312e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.815", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

c312r_df2 = c312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c312e_df2 = c312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c312r_dw = dwplot(c312r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.001</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

c312e_dw = dwplot(c312e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.945", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

legend_models = rbind(f312r_df2, s312r_df2, p312r_df2, o312r_df2, c312r_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  xlab("Estimate") + scale_color_manual(values = c("#418204", "#834600",
                                                   "#b05b00", "#4a228c",
                                                   "#873eff"),
                                        labels = c("Faith's PD", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness", 
                                                   "Observed ASVs", "Chao 1")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_312months.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f312r_dw, s312r_dw, p312r_dw, o312r_dw, c312r_dw, 
          f312e_dw, s312e_dw, p312e_dw, o312e_dw, c312e_dw,
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

#1-3 years
f13r_df2 = f13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f13e_df2 = f13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f13r_dw = dwplot(f13r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  ggtitle("1-3 years") + 
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.006</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

f13e_dw = dwplot(f13e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Ethnicity: p = 0.015</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

o13r_df2 = o13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o13e_df2 = o13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o13r_dw = dwplot(o13r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.001</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

o13e_dw = dwplot(o13e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Ethnicity: p = 0.023</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

s13r_df2 = s13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s13e_df2 = s13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s13r_dw = dwplot(s13r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.020</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")
s13e_dw = dwplot(s13e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Ethnicity: p = 0.042</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

p13r_df2 = p13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p13e_df2 = p13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p13r_dw = dwplot(p13r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.142", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

p13e_dw = dwplot(p13e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Ethnicity: p = 0.174", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1)

c13r_df2 = c13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c13e_df2 = c13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c13r_dw = dwplot(c13r_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Race: p = 0.002</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

c13e_dw = dwplot(c13e_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>Ethnicity: p = 0.018</b>", 
           x = -Inf, y = Inf, size = 6,
           hjust = 0, vjust = 1, color = "blue1")

legend_models = rbind(f13r_df2, s13r_df2, p13r_df2, o13r_df2, c13r_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  scale_color_manual(values = c("#418204", "#834600",
                                                   "#b05b00", "#4a228c",
                                                   "#873eff"),
                                        labels = c("Faith's PD", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness", 
                                                   "Observed ASVs", "Chao 1")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_13years.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f13r_dw, s13r_dw, p13r_dw, o13r_dw, c13r_dw, 
          f13e_dw, s13e_dw, p13e_dw, o13e_dw, c13e_dw,
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

#3-12 years
f312yr_df2 = f312yr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f312ye_df2 = f312ye_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Faith's PD")
f312yr_dw = dwplot(f312yr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        axis.title.y=element_text(size=rel(2.5),
                                  hjust = 0),
        legend.position = "none") + 
  ylab("3-12 years") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White"))
f312ye_dw = dwplot(f312ye_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic"))

o312yr_df2 = o312yr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o312ye_df2 = o312ye_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Observed ASVs")
o312yr_dw = dwplot(o312yr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))
o312ye_dw = dwplot(o312ye_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#4a228c"),
                                        labels = c("Observed ASVs"))

s312yr_df2 = s312yr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s312ye_df2 = s312ye_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Shannon Diversity")
s312yr_dw = dwplot(s312yr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
 scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity"))
s312ye_dw = dwplot(s312ye_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", values = c("#834600"),
                                        labels = c("Shannon Diversity"))

p312yr_df2 = p312yr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p312ye_df2 = p312ye_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Pielou's Evenness")
p312yr_dw = dwplot(p312yr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness"))
p312ye_dw = dwplot(p312ye_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#b05b00"),
                                        labels = c("Pielou's Evenness"))

c312yr_df2 = c312yr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c312ye_df2 = c312ye_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "Chao 1") 
c312yr_dw = dwplot(c312yr_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  scale_color_manual(name = "Metric", values = c("#873eff"),
                     labels = c("Chao 1"))
c312ye_dw = dwplot(c312ye_df2, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(color = "model", size=3)),
       whisker_args = list(aes(color = "model", size=2))) +
  theme_classic() + 
  theme(axis.title.x=element_text(size=rel(1.5)),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        legend.position = "none") + 
  xlab("Estimate") + scale_color_manual(name = "Metric", 
                                        values = c("#873eff"),
                                        labels = c("Chao 1"))

legend_models = rbind(f312yr_df2, o312yr_df2, c312yr_df2, s312yr_df2, p312yr_df2)
legend_plot = dwplot(legend_models,
       dot_args = list(aes(color = model)),
       whisker_args = list(aes(color = model))) + 
  xlab("Estimate") + scale_color_manual(values = c("#418204", "#4a228c",
                                                   "#873eff", "#834600",
                                                   "#b05b00"),
                                        labels = c("Faith's PD", 
                                                   "Observed ASVs", "Chao 1", 
                                                   "Shannon Diversity", 
                                                   "Pielou's Evenness")) +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(colour = guide_legend(override.aes = list(size=7)))
combo_legend = get_legend(legend_plot)

tiff(file="alpha_effect_race_eth_312years.tif", 
     res=300, width=15, height=8, units="in")
ggarrange(f312yr_dw, o312yr_dw, c312yr_dw, s312yr_dw, p312yr_dw,  
          f312ye_dw, o312ye_dw, c312ye_dw, s312ye_dw, p312ye_dw, 
          nrow = 2, ncol = 5, heights = c(2,1), align = "v",
          legend.grob = combo_legend, legend = "bottom")
dev.off()

###Combo
f1m3mr_df3 = f1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
o1m3mr_df3 = o1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
s1m3mr_df3 = s1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
p1m3mr_df3 = p1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
c1m3mr_df3 = c1m3mr_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")

f312r_df3 = f312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
o312r_df3 = o312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
s312r_df3 = s312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
p312r_df3 = p312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
c312r_df3 = c312r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")

f13r_df3 = f13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
o13r_df3 = o13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
s13r_df3 = s13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
p13r_df3 = p13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
c13r_df3 = c13r_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")

faithr_df = rbind(f1m3mr_df3, f312r_df3, f13r_df3)
faithr_df$model = factor(faithr_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
shannonr_df = rbind(s1m3mr_df3, s312r_df3, s13r_df3)
shannonr_df$model = factor(shannonr_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
pielour_df = rbind(p1m3mr_df3, p312r_df3, p13r_df3)
pielour_df$model = factor(pielour_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
observedr_df = rbind(o1m3mr_df3, o312r_df3, o13r_df3)
observedr_df$model = factor(observedr_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
chaor_df = rbind(c1m3mr_df3, c312r_df3, c13r_df3)
chaor_df$model = factor(chaor_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 

faithr_dw = dwplot(faithr_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  ggtitle("Faith's PD") +
  scale_color_manual(name = "Metric", 
                                        values = c("#418204","#418204","#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.718<br>
           3-11.9 months: p = 0.233<br>
           <span style='color: blue1;'><b><i>1-2.9 years: p = 0.006</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

observedr_dw = dwplot(observedr_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  ggtitle("Observed ASVs") +
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c","#4a228c","#4a228c"),
                                        labels = c("Observed ASVs")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.448<br>
           <span style='color: blue1;'><b><i>3-11.9 months: p = 0.005<br>
           1-2.9 years: p = 0.001</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

chaor_dw = dwplot(chaor_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  ggtitle("Chao 1") +
  scale_color_manual(name = "Metric", 
                                        values = c("#873eff","#873eff","#873eff"),
                                        labels = c("Chao 1")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.311<br>
           <span style='color: blue1;'><b><i>3-11.9 months: p = 0.001<br>
           1-2.9 years: p = 0.002</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

shannonr_dw = dwplot(shannonr_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  ggtitle("Shannon Diversity") +
  scale_color_manual(name = "Metric", 
                                        values = c("#834600","#834600","#834600"),
                                        labels = c("Shannon Diversity")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.642<br>
           <span style='color: blue1;'><b><i>3-11.9 months: p = 0.006<br>
           1-2.9 years: p = 0.020</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

pielour_dw = dwplot(pielour_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_text(size=rel(2), face = "bold"),
        legend.position = "none") + 
  ggtitle("Pielou's Evenness") +
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00","#b05b00","#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  scale_y_discrete(labels = c("Asian/PI\n- Black", 
                              "Asian/PI\n- White", 
                              "Black\n- White")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.747<br>
           <span style='color: blue1;'><b><i>3-11.9 months: p = 0.026</b></i><br></span>
           1-2.9 years: p = 0.142", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

f1m3me_df3 = f1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
o1m3me_df3 = o1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
s1m3me_df3 = s1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
p1m3me_df3 = p1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")
c1m3me_df3 = c1m3me_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "0-2.9 months")

f312e_df3 = f312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
o312e_df3 = o312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
s312e_df3 = s312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
p312e_df3 = p312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")
c312e_df3 = c312e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "3-11.9 months")

f13e_df3 = f13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
o13e_df3 = o13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
s13e_df3 = s13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
p13e_df3 = p13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")
c13e_df3 = c13e_df %>% dplyr::select(contrast:adj.p.value) %>% 
  rename(term = contrast) %>% rename(p.value = adj.p.value) %>% 
  mutate(model = "1-2.9 years")

faithe_df = rbind(f1m3me_df3, f312e_df3, f13e_df3)
faithe_df$model = factor(faithe_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
shannone_df = rbind(s1m3me_df3, s312e_df3, s13e_df3)
shannone_df$model = factor(shannone_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
pieloue_df = rbind(p1m3me_df3, p312e_df3, p13e_df3)
pieloue_df$model = factor(pieloue_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
observede_df = rbind(o1m3me_df3, o312e_df3, o13e_df3)
observede_df$model = factor(observede_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 
chaoe_df = rbind(c1m3me_df3, c312e_df3, c13e_df3) 
chaoe_df$model = factor(chaoe_df$model, levels = c("0-2.9 months", "3-11.9 months", "1-2.9 years")) 

faithe_dw = dwplot(faithe_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.text.y=element_text(size=rel(2),
                                 angle = 90,
                                 hjust = 0.5),
        plot.title=element_blank(),
        legend.position = "none")+
  scale_color_manual(name = "Metric", 
                                        values = c("#418204","#418204","#418204"),
                                        labels = c("Faith's PD")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.384<br>
           3-11.9 months: p = 0.880<br>
           <span style='color: blue1;'><b><i>1-2.9 years: p = 0.015</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

observede_dw = dwplot(observede_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_blank(),
        legend.position = "none") +
  scale_color_manual(name = "Metric", 
                                        values = c("#4a228c","#4a228c","#4a228c"),
                                        labels = c("Observed ASVs")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.670<br>
           3-11.9 months: p = 0.914<br>
           <span style='color: blue1;'><b><i>1-2.9 years: p = 0.023</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

chaoe_dw = dwplot(chaoe_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_blank(),
        legend.position = "none") +
  scale_color_manual(name = "Metric", 
                                        values = c("#873eff","#873eff","#873eff"),
                                        labels = c("Chao 1")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.977<br>
           3-11.9 months: p = 0.668<br>
           <span style='color: blue1;'><b><i>1-2.9 years: p = 0.018</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

shannone_dw = dwplot(shannone_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_blank(),
        legend.position = "none") +
  scale_color_manual(name = "Metric", 
                                        values = c("#834600","#834600","#834600"),
                                        labels = c("Shannon Diversity")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.839<br>
           3-11.9 months: p = 0.406<br>
           <span style='color: blue1;'><b><i>1-2.9 years: p = 0.042</b></i></span>", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

pieloue_dw = dwplot(pieloue_df, vline = geom_vline(xintercept = 0, color = "gray40",
                                    linetype = 2),
       dot_args = list(aes(shape = model, size = 3)),
       whisker_args = list(aes(linetype = model, size = 2))) +
  theme_classic() + 
  theme(axis.title.x=element_blank(),
        axis.text.x=element_text(size=rel(1.5)),
        axis.title.y=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        plot.title=element_blank(),
        legend.position = "none")+
  scale_color_manual(name = "Metric", 
                                        values = c("#b05b00","#b05b00","#b05b00"),
                                        labels = c("Pielou's Evenness")) +
  scale_y_discrete(labels = c("Non Hispanic\n- Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "0-2.9 months: p = 0.548<br>
           3-11.9 months: p = 0.815<br>
           1-2.9 years: p = 0.174", 
           x = Inf, y = -Inf, size = 6,
           hjust = 1, vjust = 0)

pielour_df$model = factor(pielour_df$model, levels = c("0-2.9 months",
                                                       "3-11.9 months",
                                                       "1-2.9 years"))
legend_plot = dwplot(pielour_df,
                     dot_args = list(aes(shape = model, size = 3)),
                     whisker_args = list(aes(linetype = model, 
                                             size = 2))) + 
  xlab("Estimate") + scale_color_manual(values = c("black", 
                                                   "black",
                                                   "black"),
                                        guide = "none") +
  theme(legend.title = element_blank(),
        legend.key = element_rect(fill = "white", colour = "white"),
        legend.text = element_text(size = rel(2)),
        legend.position = "bottom") + 
  guides(shape = guide_legend(override.aes = list(size=7), reverse = T),
         color = "none",
         size = "none")
combo_legend = get_legend(legend_plot) 
```


###By study
```{r bystudy}
alpha_kim = alpha_all %>% 
  filter(Study == "Kim")
f_kim = lm(faith_pd ~ Race + Ethnicity + host_sex, data=alpha_kim, na.action = na.omit)
summary(f_kim)
Anova(f_kim)

o_kim = lm(observed_features ~ Race + Ethnicity + host_sex, data=alpha_kim, na.action = na.omit)
summary(o_kim)
Anova(o_kim)

s_kim = lm(shannon_entropy ~ Race + Ethnicity + host_sex, data=alpha_kim, na.action = na.omit)
summary(s_kim)
Anova(s_kim)

p_kim = lm(pielou_evenness ~ Race + Ethnicity + host_sex, data=alpha_kim, na.action = na.omit)
summary(p_kim)
Anova(p_kim)

c_kim = lm(chao1 ~ Race + Ethnicity + host_sex, data=alpha_kim, na.action = na.omit)
summary(c_kim)
Anova(c_kim)

alpha_herman = alpha_all %>% 
  filter(Study == "Herman")
f_herman = lmer(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_herman, na.action = na.omit)
summary(f_herman)
Anova(f_herman)
summary(glht(f_herman,linfct=mcp(Race="Tukey")))
summary(glht(f_herman,linfct=mcp(Ethnicity="Tukey")))

o_herman = lmer(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_herman, na.action = na.omit)
summary(o_herman)
Anova(o_herman)
summary(glht(o_herman,linfct=mcp(Race="Tukey")))
summary(glht(o_herman,linfct=mcp(Ethnicity="Tukey")))

s_herman = lmer(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_herman, na.action = na.omit)
summary(s_herman)
Anova(s_herman)

p_herman = lmer(pielou_evenness ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_herman, na.action = na.omit)
summary(p_herman)
Anova(p_herman)

c_herman = lmer(chao1 ~ Race + Ethnicity + Host_age_cat + host_sex + Breastfeeding + (1|host_subject_id), data=alpha_herman, na.action = na.omit)
summary(c_herman)
Anova(c_herman)
summary(glht(c_herman,linfct=mcp(Race="Tukey")))
summary(glht(c_herman,linfct=mcp(Ethnicity="Tukey")))

alpha_levin = alpha_all %>% 
  filter(Study == "Chu")
f_levin = lm(faith_pd ~ Race + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(f_levin)
Anova(f_levin)

o_levin = lm(observed_features ~ Race + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(o_levin)
Anova(o_levin)

s_levin = lm(shannon_entropy ~ Race + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(s_levin)
Anova(s_levin)

p_levin = lm(pielou_evenness ~ Race + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(p_levin)
Anova(p_levin)

c_levin = lm(chao1 ~ Race + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(c_levin)
Anova(c_levin)

alpha_robinson = alpha_all %>% 
  filter(Study == "Robinson")
f_robinson = lm(faith_pd ~ Race + Ethnicity + host_sex, data=alpha_robinson, na.action = na.omit)
summary(f_robinson)
Anova(f_robinson)

o_robinson = lm(observed_features ~ Race + Ethnicity + host_sex, data=alpha_robinson, na.action = na.omit)
summary(o_robinson)
Anova(o_robinson)

s_robinson = lm(shannon_entropy ~ Race + Ethnicity + host_sex, data=alpha_robinson, na.action = na.omit)
summary(s_robinson)
Anova(s_robinson)
summary(glht(s_robinson,linfct=mcp(Race="Tukey")))

p_robinson = lm(pielou_evenness ~ Race + Ethnicity + host_sex, data=alpha_robinson, na.action = na.omit)
summary(p_robinson)
Anova(p_robinson)
summary(glht(p_robinson,linfct=mcp(Race="Tukey")))

c_robinson = lm(chao1 ~ Race + Ethnicity + host_sex, data=alpha_robinson, na.action = na.omit)
summary(c_robinson)
Anova(c_robinson)
summary(glht(c_robinson,linfct=mcp(Race="Tukey")))

alpha_planer = alpha_all %>% 
  filter(Study == "Planer")
f_planer = lmer(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_planer, na.action = na.omit)
summary(f_planer)
Anova(f_planer)
summary(glht(f_planer,linfct=mcp(Race="Tukey")))
summary(glht(f_planer,linfct=mcp(Ethnicity="Tukey")))

o_planer = lmer(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_planer, na.action = na.omit)
summary(o_planer)
Anova(o_planer)
summary(glht(o_planer,linfct=mcp(Race="Tukey")))

s_planer = lmer(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_planer, na.action = na.omit)
summary(s_planer)
Anova(s_planer)
summary(glht(s_planer,linfct=mcp(Race="Tukey")))

p_planer = lmer(pielou_evenness ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_planer, na.action = na.omit)
summary(p_planer)
Anova(p_planer)
summary(glht(p_planer,linfct=mcp(Race="Tukey")))

c_planer = lmer(chao1 ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_planer, na.action = na.omit)
summary(c_planer)
Anova(c_planer)
summary(glht(c_planer,linfct=mcp(Race="Tukey")))

alpha_levin = alpha_all %>% 
  filter(Study == "Levin")
f_levin= lm(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(f_levin)
Anova(f_levin)
summary(glht(f_levin,linfct=mcp(Race="Tukey")))
summary(glht(f_levin,linfct=mcp(Ethnicity="Tukey")))

o_levin= lm(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(o_levin)
Anova(o_levin)
summary(glht(o_levin,linfct=mcp(Race="Tukey")))
summary(glht(o_levin,linfct=mcp(Ethnicity="Tukey")))

s_levin= lm(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(s_levin)
Anova(s_levin)
summary(glht(s_levin,linfct=mcp(Race="Tukey")))

p_levin= lm(pielou_evenness ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(p_levin)
Anova(p_levin)
summary(glht(p_levin,linfct=mcp(Race="Tukey")))

c_levin= lm(chao1 ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_levin, na.action = na.omit)
summary(c_levin)
Anova(c_levin)
summary(glht(c_levin,linfct=mcp(Race="Tukey")))
summary(glht(c_levin,linfct=mcp(Ethnicity="Tukey")))

alpha_db = alpha_all %>% 
  filter(Study == "Dominguez-Bello")
f_db = lmer(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_db, na.action = na.omit)
summary(f_db)
Anova(f_db)

o_db = lmer(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_db, na.action = na.omit)
summary(o_db)
Anova(o_db)

s_db = lmer(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_db, na.action = na.omit)
summary(s_db)
Anova(s_db)

p_db = lmer(pielou_evenness ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_db, na.action = na.omit)
summary(p_db)
Anova(p_db)

c_db = lmer(chao1 ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding + (1|host_subject_id), data=alpha_db, na.action = na.omit)
summary(c_db)
Anova(c_db)

alpha_cioffi = alpha_all %>% 
  filter(Study == "Cioffi")
f_cioffi = lm(faith_pd ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_cioffi, na.action = na.omit)
summary(f_cioffi)
Anova(f_cioffi)

o_cioffi = lm(observed_features ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_cioffi, na.action = na.omit)
summary(o_cioffi)
Anova(o_cioffi)

s_cioffi = lm(shannon_entropy ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_cioffi, na.action = na.omit)
summary(s_cioffi)
Anova(s_cioffi)

p_cioffi = lm(pielou_evenness ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_cioffi, na.action = na.omit)
summary(p_cioffi)
Anova(p_cioffi)

c_cioffi = lm(chao1 ~ Race + Ethnicity + Host_age_cat + host_sex + host_del_route + Breastfeeding, data=alpha_cioffi, na.action = na.omit)
summary(c_cioffi)
Anova(c_cioffi)
```
####Graphs
```{r alpha_div_race_study_graphs}
plot1_faith_all = ggboxplot(alpha_all, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "All Studies") 
plot1_faith_all = ggpar(plot1_faith_all, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.082", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_all = ggboxplot(alpha_all, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_all = ggpar(plot2_asvs_all, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("***", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.002", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_all = ggboxplot(alpha_all, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_all = ggpar(plot3_shannon_all, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.008", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_all = ggboxplot(alpha_all, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_all = ggpar(plot4_pielou_all, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.117", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_all = ggboxplot(alpha_all, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_all = ggpar(plot5_chao1_all, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_kim = ggboxplot(alpha_kim, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Kim") 
plot1_faith_kim = ggpar(plot1_faith_kim, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.116", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_kim = ggboxplot(alpha_kim, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_kim = ggpar(plot2_asvs_kim, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.125", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_kim = ggboxplot(alpha_kim, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_kim = ggpar(plot3_shannon_kim, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.124", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_kim = ggboxplot(alpha_kim, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_kim = ggpar(plot4_pielou_kim, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.110", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_kim = ggboxplot(alpha_kim, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_kim = ggpar(plot5_chao1_kim, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.219", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_herman = ggboxplot(alpha_herman, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Herman") 
plot1_faith_herman = ggpar(plot1_faith_herman, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("***", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.032", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_herman = ggboxplot(alpha_herman, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_herman = ggpar(plot2_asvs_herman, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.099", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_herman = ggboxplot(alpha_herman, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_herman = ggpar(plot3_shannon_herman, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.635", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_herman = ggboxplot(alpha_herman, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_herman = ggpar(plot4_pielou_herman, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.589", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_herman = ggboxplot(alpha_herman, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_herman = ggpar(plot5_chao1_herman, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.066", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_chu = ggboxplot(alpha_chu, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Chu") 
plot1_faith_chu = ggpar(plot1_faith_chu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.910", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_chu = ggboxplot(alpha_chu, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_chu = ggpar(plot2_asvs_chu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.560", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_chu = ggboxplot(alpha_chu, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_chu = ggpar(plot3_shannon_chu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.362", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_chu = ggboxplot(alpha_chu, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_chu = ggpar(plot4_pielou_chu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.211", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_chu = ggboxplot(alpha_chu, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_chu = ggpar(plot5_chao1_chu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.572", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_robinson = ggboxplot(alpha_robinson, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Robinson") 
plot1_faith_robinson = ggpar(plot1_faith_robinson, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.123", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_robinson = ggboxplot(alpha_robinson, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_robinson = ggpar(plot2_asvs_robinson, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.109", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_robinson = ggboxplot(alpha_robinson, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_robinson = ggpar(plot3_shannon_robinson, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Asian/Pacific Islander")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.027", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_robinson = ggboxplot(alpha_robinson, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_robinson = ggpar(plot4_pielou_robinson, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Asian/Pacific Islander")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.044", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_robinson = ggboxplot(alpha_robinson, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_robinson = ggpar(plot5_chao1_robinson, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.085", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_planer = ggboxplot(alpha_planer, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Planer") 
plot1_faith_planer = ggpar(plot1_faith_planer, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.053", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_planer = ggboxplot(alpha_planer, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_planer = ggpar(plot2_asvs_planer, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("***", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.011", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_planer = ggboxplot(alpha_planer, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_planer = ggpar(plot3_shannon_planer, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.024", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_planer = ggboxplot(alpha_planer, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_planer = ggpar(plot4_pielou_planer, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.048", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_planer = ggboxplot(alpha_planer, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_planer = ggpar(plot5_chao1_planer, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.014", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_levin = ggboxplot(alpha_levin, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Levin") 
plot1_faith_levin = ggpar(plot1_faith_levin, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p < 0.001", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_levin = ggboxplot(alpha_levin, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_levin = ggpar(plot2_asvs_levin, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("***", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_levin = ggboxplot(alpha_levin, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_levin = ggpar(plot3_shannon_levin, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.006", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_levin = ggboxplot(alpha_levin, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_levin = ggpar(plot4_pielou_levin, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.111", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_levin = ggboxplot(alpha_levin, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_levin = ggpar(plot5_chao1_levin, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.5, 1),
                                        symbols = c("**", "ns"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_db = ggboxplot(alpha_db, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Dominguez-Bello") 
plot1_faith_db = ggpar(plot1_faith_db, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.650", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_db = ggboxplot(alpha_db, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_db = ggpar(plot2_asvs_db, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.476", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_db = ggboxplot(alpha_db, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_db = ggpar(plot3_shannon_db, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.809", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_db = ggboxplot(alpha_db, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_db = ggpar(plot4_pielou_db, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.849", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_db = ggboxplot(alpha_db, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_db = ggpar(plot5_chao1_db, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.464", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot1_faith_cioffi = ggboxplot(alpha_cioffi, x = "Race", 
                  y = "faith_pd", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Faith's Phylogenetic Diversity",
                  title = "Cioffi") 
plot1_faith_cioffi = ggpar(plot1_faith_cioffi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.749", 
           x = Inf, y = Inf, size = 3,
           hjust = 1, vjust = 1)

plot2_asvs_cioffi = ggboxplot(alpha_cioffi, x = "Race", 
                  y = "observed_features", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter",  
                  add.params = list(fill = "white"), 
                  ylab = "Observed ASVs") 
plot2_asvs_cioffi = ggpar(plot2_asvs_cioffi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.582", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot3_shannon_cioffi = ggboxplot(alpha_cioffi, x = "Race", 
                  y = "shannon_entropy", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") 
plot3_shannon_cioffi = ggpar(plot3_shannon_cioffi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.247", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot4_pielou_cioffi = ggboxplot(alpha_cioffi, x = "Race", 
                  y = "pielou_evenness", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Pielou's Evenness") 
plot4_pielou_cioffi = ggpar(plot4_pielou_cioffi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.209", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

plot5_chao1_cioffi = ggboxplot(alpha_cioffi, x = "Race", 
                  y = "chao1", color = "Race", 
                  palette = c("#377eb8", 
                              "#ffd92f", "#e41a1c"), 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Chao1") 
plot5_chao1_cioffi = ggpar(plot5_chao1_cioffi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Race: p = 0.636", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="alpha_comp_studies.tif", 
     res=100, width=30, height=15, units="in")
ggarrange(plot1_faith_all, plot1_faith_kim, plot1_faith_herman, plot1_faith_chu,
          plot1_faith_robinson, plot1_faith_planer, plot1_faith_levin,
          plot1_faith_db, plot1_faith_cioffi,
          plot2_asvs_all, plot2_asvs_kim, plot2_asvs_herman, plot2_asvs_chu,
          plot2_asvs_robinson, plot2_asvs_planer, plot2_asvs_levin,
          plot2_asvs_db, plot2_asvs_cioffi,
          plot3_shannon_all, plot3_shannon_kim, plot3_shannon_herman, 
          plot3_shannon_chu,
          plot3_shannon_robinson, plot3_shannon_planer, plot3_shannon_levin,
          plot3_shannon_db, plot3_shannon_cioffi,
          plot4_pielou_all, plot4_pielou_kim, plot4_pielou_herman, 
          plot4_pielou_chu,
          plot4_pielou_robinson, plot4_pielou_planer, plot4_pielou_levin,
          plot4_pielou_db, plot4_pielou_cioffi,
          plot5_chao1_all, plot5_chao1_kim, plot5_chao1_herman, plot5_chao1_chu,
          plot5_chao1_robinson, plot5_chao1_planer, plot5_chao1_levin,
          plot5_chao1_db, plot5_chao1_cioffi,
          nrow = 5, ncol = 9, 
          common.legend = T, align = "hv", legend = "right")
dev.off()
```

###By study and age category


##Different abundance of taxa
ANCOM-BC (a bias correcting version of analysis of microbiome compositions) was used to identify differentially abundant taxa at the level of phyla, family, and ASV.

###Phyla
```{r phyla_abund}
phyla = read_tsv("feature-table-phyla.tsv")
taxonomy = phyla %>% dplyr::select(Phyla)

phyla_matrix = phyla %>% column_to_rownames("Phyla") %>% as.matrix()
phyla_phylo = otu_table(phyla_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

phyla_p = phyloseq(phyla_phylo, meta_phylo)

#Race
race = ancombc(phyla_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Phyla") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_phyla.csv")

#Ethnicity
ethnicity = ancombc(phyla_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e = ethnicity$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Phyla") %>% 
  inner_join(taxonomy)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_phyla.csv")

#Age
age = ancombc(phyla_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Phyla") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_phyla.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
phyla_p_ab = phyloseq(phyla_phylo, meta_ab)
race_ab = ancombc(phyla_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_phyla_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
phyla_p_aw = phyloseq(phyla_phylo, meta_aw)
race_aw = ancombc(phyla_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_phyla_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
phyla_p_bw = phyloseq(phyla_phylo, meta_bw)
race_bw = ancombc(phyla_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Phyla"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_phyla_bw.csv")

```
####Phyla graphs
```{r phyla_graphs_race}
phyla_ra = phyla %>% column_to_rownames("Phyla") %>% 
  decostand(method = "total", MARGIN = 2)
phyla_meta = phyla_ra %>% 
  rownames_to_column("Phyla") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Phyla", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

bact_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes", color = "Race",
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Bacteroidetes") 
bact_race = ggpar(bact_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "White"), 
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.05, 1),
                                        symbols = c("***", "**"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

prot_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria", color = "Race", 
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Proteobacteria") 
prot_race = ggpar(prot_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "White"), 
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.05, 
                                                      1),
                                        symbols = c("***", "***"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tene_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Tenericutes", color = "Race", 
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Tenericutes") 
tene_race = ggpar(tene_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "Black")),
                     symnum.args = list(cutpoints = c(0, 0.05, 1),
                                        symbols = c("***", "***"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.003", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

verr_race = ggboxplot(phyla_meta, x = "Race", 
                  y = "k__Bacteria;p__Verrucomicrobia", color = "Race", 
                  palette = c("#377eb8", "#ffd92f", "#e41a1c"), 
                 add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Verrucomicrobia") 
verr_race = ggpar(verr_race, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "Black")),
                     symnum.args = list(cutpoints = c(0, 0.05,1),
                                        symbols = c("*", "*"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.002", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="phyla_diffs_race.tif", 
     res=300, width=12, height=9, units="in")
ggarrange(bact, prot, tene, verr, nrow = 2, ncol = 2, 
          common.legend = T, align = "hv", legend = "right",
          labels = c("A", "B", "C", "D"))
dev.off()
```

```{r phyla_graphs_ethnicity}
phyla_ra = phyla %>% column_to_rownames("Phyla") %>% 
  decostand(method = "total", MARGIN = 2)
phyla_meta = phyla_ra %>% 
  rownames_to_column("Phyla") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Phyla", values_from = "rel_abund") %>% 
  inner_join(metadata_re)


prot_eth = ggboxplot(phyla_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Proteobacteria", 
                 color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Proteobacteria") 
prot_eth = ggpar(prot_eth, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", 
                                          "Hispanic")),
                     symnum.args = list(cutpoints = c(0, 
                                                      0.5, 1),
                                        symbols = c("***", "***"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q < 0.001", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tene_eth = ggboxplot(phyla_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Tenericutes", 
                 color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Tenericutes") 
tene_eth = ggpar(tene_eth, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.5, 1),
                                        symbols = c("*", "*"))) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.024", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

acti_eth = ggboxplot(phyla_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Actinobacteria", 
                 color = "Ethnicity", 
                  palette = c("light gray", "dark gray"), add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Actinobacteria") 
acti_eth = ggpar(acti_eth, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic",
                                          "Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.5, 1),
                                        symbols = c("**", "**")))  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "q = 0.004", 
           x = Inf, y = Inf, size = 4,
           hjust = 1, vjust = 1)

tiff(file="phyla_diffs_ethnicity.tif", 
     res=300, width=15, height=5, units="in")
ggarrange(acti, prot, tene, nrow = 1, ncol = 3, 
          common.legend = T, align = "v", legend = "right",
          labels = c("A", "B", "C"))
dev.off()
```

###Families
```{r families_abund}
family = read_tsv("feature-table-family.tsv")
taxonomy = family %>% dplyr::select(Family)

family_matrix = family %>% column_to_rownames("Family") %>% 
  as.matrix()
family_phylo = otu_table(family_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

family_p = phyloseq(family_phylo, meta_phylo)

#Race
race = ancombc(family_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Family") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_family.csv")

#Ethnicity
ethnicity = ancombc(family_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_family.csv")

#Age
age = ancombc(family_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Family") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_family.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
family_p_ab = phyloseq(family_phylo, meta_ab)
race_ab = ancombc(family_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_family_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
family_p_aw = phyloseq(family_phylo, meta_aw)
race_aw = ancombc(family_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_family_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
family_p_bw = phyloseq(family_phylo, meta_bw)
race_bw = ancombc(family_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Family"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_family_bw.csv")
```

```{r family_graphs_race}
family_ra = family %>% column_to_rownames("Family") %>% 
  decostand(method = "total", MARGIN = 2)
family_meta = family_ra %>% 
  rownames_to_column("Family") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Family", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

prev = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Prevotellaceae") 
prev = ggpar(prev, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", 
                                          "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

porp = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Porphyromonadaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Porphyromonadaceae") 
porp = ggpar(porp, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 
                                                      0.001, 0.05, 
                                                      0.5, 1),
                                        symbols = c("*", "*", "*", 
                                                    "*", "*")))

lact = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Lactobacillaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Lactobacillaceae") 
lact = ggpar(lact, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

camp = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Epsilonproteobacteria;o__Campylobacterales;f__Campylobacteraceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Campylobacteraceae") 
camp = ggpar(camp, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Black", "Other"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

veil = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Veillonellaceae") 
veil = ggpar(veil, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

pept = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Peptostreptococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Peptostreptococcaceae") 
pept = ggpar(pept, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

tiss = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__[Tissierellaceae]", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Tissierellaceae") 
tiss = ggpar(tiss, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

ente = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Enterobacteriaceae") 
ente = ggpar(ente, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

clos = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Clostridiaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Clostridiaceae") 
clos = ggpar(clos, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Other"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

fuso = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Fusobacteria;c__Fusobacteriia;o__Fusobacteriales;f__Fusobacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Fusobacteriaceae") 
fuso = ggpar(fuso, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

stap = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Bacillales;f__Staphylococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Staphylococcaceae") 
stap = ggpar(stap, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

geme = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Gemellales;f__Gemellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Gemellaceae") 
geme = ggpar(geme, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

mogi = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__[Mogibacteriaceae]", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Mogibacteriaceae") 
mogi = ggpar(mogi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Other"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

mora = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pseudomonadales;f__Moraxellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Moraxellaceae") 
mora = ggpar(mora, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Black", "Other"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

cory = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Actinobacteria;c__Actinobacteria;o__Actinomycetales;f__Corynebacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Corynebacteriaceae") 
cory = ggpar(cory, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

entec = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Enterococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Enterococcaceae") 
entec = ggpar(entec, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

past = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Pasteurellaceae") 
past = ggpar(past, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

para = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__[Paraprevotellaceae]", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Paraprevotellaceae") 
para = ggpar(para, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "Black"),
                                        c("Asian/Pacific Islander", "Other"),
                                        c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

micr = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Actinobacteria;c__Actinobacteria;o__Actinomycetales;f__Micrococcaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Micrococcaceae") 
micr = ggpar(micr, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander",
                                          "White"),
                                        c("Black", "Other"),
                                        c("Black", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

euba = ggboxplot(family_meta, x = "Race", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Eubacteriaceae", 
                 color = "Race", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Eubacteriaceae") 
euba = ggpar(euba, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Asian/Pacific Islander", "White"),
                                        c("Black", "Other"),
                                        c("Other", "White")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

tiff(file="family_diffs_race.tif", 
     res=300, width=22, height=15, units="in")
ggarrange(prev, porp, lact, camp, veil, pept, tiss, ente, clos, fuso, stap, 
          geme, mogi, mora, cory, entec, past, para, micr, euba, 
          nrow = 4, ncol = 5, 
          common.legend = T, align = "hv", legend = "right")
dev.off()
```

```{r family_graphs_ethn}
family_ra = family %>% column_to_rownames("Family") %>% 
  decostand(method = "total", MARGIN = 2)
family_meta = family_ra %>% 
  rownames_to_column("Family") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Family", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

lach = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Lachnospiraceae") 
lach = ggpar(lach, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

bact = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Bacteroidaceae") 
bact = ggpar(bact, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic",
                                          "Other"), 
                                        c("Hispanic", "Other")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 
                                                      0.001, 0.05, 
                                                      0.5, 1),
                                        symbols = c("*", "*", "*", 
                                                    "*", "*")))

prev = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Prevotellaceae") 
prev = ggpar(prev, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", "Other"),
                                        c("Hispanic", "Other"),
                                        c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

barn = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__[Barnesiellaceae]", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Barnesiellaceae") 
barn = ggpar(barn, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

ente = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Enterobacteriaceae") 
ente = ggpar(ente, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") 

rike = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Rikenellaceae") 
rike = ggpar(rike, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", "Other"),
                                        c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

stre = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Streptococcaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Streptococcaceae") 
stre = ggpar(stre, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

geme = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Gemellales;f__Gemellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Gemellaceae") 
geme = ggpar(geme, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

chri = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Christensenellaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Christensenellaceae") 
chri = ggpar(chri, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") 

pseu = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pseudomonadales;f__Pseudomonadaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Pseudomonadaceae") 
pseu = ggpar(pseu, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

verr = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Verrucomicrobia;c__Verrucomicrobiae;o__Verrucomicrobiales;f__Verrucomicrobiaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Verrucomicrobiaceae") 
verr = ggpar(verr, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

turi = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Bacilli;o__Turicibacterales;f__Turicibacteraceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Turicibacteraceae") 
turi = ggpar(turi, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))

euba = ggboxplot(family_meta, x = "Ethnicity", 
                  y = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Eubacteriaceae", 
                 color = "Ethnicity", 
                  palette = "Set1", add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative Abundance", 
                 title = "Eubacteriaceae") 
euba = ggpar(euba, legend = "right") + rremove("xlab") + 
  rremove("x.text") + rremove("x.ticks") + 
  rremove("legend.title") + 
  stat_compare_means(label = "p.signif", 
                     comparisons = list(c("Non Hispanic", "Other"),
                                        c("Hispanic", 
                                          "Non Hispanic")),
                     symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.05, 0.5, 1),
                                        symbols = c("*", "*", "*", "*", "*")))


tiff(file="family_diffs_ethn.tif", 
     res=300, width=22, height=12, units="in")
ggarrange(lach, bact, prev, barn, ente, rike, stre, geme, 
          chri, pseu, verr, turi, euba, 
          nrow = 3, ncol = 5, 
          common.legend = T, align = "hv", legend = "right")
dev.off()
```

```{r family_graphs_race_heritable}
tiff(file="family_diffs_race_heritable.tif", 
     res=300, width=12, height=4, units="in")
ggarrange(clos, past, pept, 
          nrow = 1, ncol = 3, 
          common.legend = T, align = "v", legend = "right")
dev.off()
```

```{r family_graphs_ethn_heritable}
tiff(file="family_diffs_ethn_heritable.tif", 
     res=300, width=9, height=4, units="in")
ggarrange(lach, chri, 
          nrow = 1, ncol = 2, 
          common.legend = T, align = "v", legend = "right")
dev.off()
```

###Genera
```{r genus_abund}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus.csv")

#Age
age = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_genus.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw.csv")
```

###Species
```{r species_abund}
species = read_tsv("feature-table-species.tsv")
taxonomy = species %>% dplyr::select(Species)

species_matrix = species %>% column_to_rownames("Species") %>% as.matrix()
species_phylo = otu_table(species_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

species_p = phyloseq(species_phylo, meta_phylo)

#Race
race = ancombc(species_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Species") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_species.csv")

#Ethnicity
ethnicity = ancombc(species_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_species.csv")

#Age
age = ancombc(species_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Species") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_species.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
species_p_ab = phyloseq(species_phylo, meta_ab)
race_ab = ancombc(species_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_species_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
species_p_aw = phyloseq(species_phylo, meta_aw)
race_aw = ancombc(species_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_species_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
species_p_bw = phyloseq(species_phylo, meta_bw)
race_bw = ancombc(species_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Species"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_species_bw.csv")
```

###ASVs
```{r ASV_abund}
taxonomy = asvs %>% dplyr::select(ASV, taxonomy)

ASV_matrix = asvs %>% 
  dplyr::select(-taxonomy) %>% 
  column_to_rownames("ASV") %>% 
  as.matrix()
ASV_phylo = otu_table(ASV_matrix, taxa_are_rows = T)
meta_phylo = metadata_re %>% column_to_rownames("SampleID") %>% 
  sample_data()

ASV_p = phyloseq(ASV_phylo, meta_phylo)

#Race
race = ancombc(ASV_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr",
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "taxonomy") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_ASV.csv")

#Ethnicity
ethnicity = ancombc(ASV_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr",
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_ASV.csv")

#Age
age = ancombc(ASV_p, 
               "Race + Ethnicity + Host_age_cat + Study",
                 p_adj_method = "fdr",
                 group = "Host_age_cat", global = T)

res_a = age$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "ASV") %>% 
  inner_join(taxonomy)

write_csv(res_a, "Diff_abund_global_age_ASV.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
ASV_p_ab = phyloseq(ASV_phylo, meta_ab)
race_ab = ancombc(ASV_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_ASV_ab.csv")

##Asian vs. White
meta_aw = metadata_re %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
ASV_p_aw = phyloseq(ASV_phylo, meta_aw)
race_aw = ancombc(ASV_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_ASV_aw.csv")

##Black vs. White
meta_bw = metadata_re %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
ASV_p_bw = phyloseq(ASV_phylo, meta_bw)
race_bw = ancombc(ASV_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "ASV"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_ASV_bw.csv")
```

###Genera by study
```{r study_genera_kim}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)
meta_phylo_kim = metadata_re_kim %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_kim)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_kim.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_kim %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_kim.csv")

##Asian vs. White
meta_aw = metadata_re_kim %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_kim.csv")

##Black vs. White
meta_bw = metadata_re_kim %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_kim.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_kim.csv")
```

```{r study_genera_herman}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_herman = metadata_re_herman %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_herman)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_herman.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_herman %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_herman.csv")

##Asian vs. White
meta_aw = metadata_re_herman %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_herman.csv")

##Black vs. White
meta_bw = metadata_re_herman %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_herman.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_herman.csv")
```

```{r study_genera_levin}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_levin = metadata_re_levin %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()


genus_p = phyloseq(genus_phylo, meta_phylo_levin)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_levin.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_levin %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_levin.csv")

##Asian vs. White
meta_aw = metadata_re_levin %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_levin.csv")

##Black vs. White
meta_bw = metadata_re_levin %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_levin.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_levin.csv")
```

```{r study_genera_robinson}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_robinson = metadata_re_robinson %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_robinson)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_robinson.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_robinson %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_robinson.csv")

##Asian vs. White
meta_aw = metadata_re_robinson %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_robinson.csv")

##Black vs. White
meta_bw = metadata_re_robinson %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_robinson.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_robinson.csv")
```

```{r study_genera_planer}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_planer = metadata_re_planer %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_planer)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r_df = data.frame(
  Species = row.names(race$res$beta),
  beta = unlist(race$res$beta),
  se = unlist(race$res$se),
  W = unlist(race$res$W),
  p_val = unlist(race$res$p_val),
  q_val = unlist(race$res$q_val),
  diff_abn = unlist(race$res$diff_abn))

fdr_ancom_r <- res_r_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r, "Diff_abund_global_race_genus_planer.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_planer.csv")
```

```{r study_genera_levin}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_levin = metadata_re_levin %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()


genus_p = phyloseq(genus_phylo, meta_phylo_levin)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r_df = data.frame(
  Species = row.names(race$res$beta),
  beta = unlist(race$res$beta),
  se = unlist(race$res$se),
  W = unlist(race$res$W),
  p_val = unlist(race$res$p_val),
  q_val = unlist(race$res$q_val),
  diff_abn = unlist(race$res$diff_abn))

fdr_ancom_r <- res_r_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r, "Diff_abund_global_race_genus_levin.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_levin.csv")
```

```{r study_genera_db}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_db = metadata_re_db %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()


genus_p = phyloseq(genus_phylo, meta_phylo_db)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r = race$res_global %>% 
  filter(diff_abn == "TRUE") %>% 
  rownames_to_column(var = "Gensu") %>% 
  inner_join(taxonomy)

write_csv(res_r, "Diff_abund_global_race_genus_db.csv")

#Group pairs race

##Asian vs. Black 
meta_ab = metadata_re_db %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "Black") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_ab = phyloseq(genus_phylo, meta_ab)
race_ab = ancombc(genus_p_ab, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_ab = data.frame(
  Species = row.names(race_ab$res$beta),
  beta = unlist(race_ab$res$beta),
  se = unlist(race_ab$res$se),
  W = unlist(race_ab$res$W),
  p_val = unlist(race_ab$res$p_val),
  q_val = unlist(race_ab$res$q_val),
  diff_abn = unlist(race_ab$res$diff_abn))

fdr_ancom_r_ab <- res_r_df_ab %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_ab, "Diff_abund_race_genus_ab_db.csv")

##Asian vs. White
meta_aw = metadata_re_db %>% 
  filter(Race == "Asian/Pacific Islander" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_aw = phyloseq(genus_phylo, meta_aw)
race_aw = ancombc(genus_p_aw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_aw = data.frame(
  Species = row.names(race_aw$res$beta),
  beta = unlist(race_aw$res$beta),
  se = unlist(race_aw$res$se),
  W = unlist(race_aw$res$W),
  p_val = unlist(race_aw$res$p_val),
  q_val = unlist(race_aw$res$q_val),
  diff_abn = unlist(race_aw$res$diff_abn))

fdr_ancom_r_aw <- res_r_df_aw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_aw, "Diff_abund_race_genus_aw_db.csv")

##Black vs. White
meta_bw = metadata_re_db %>% 
  filter(Race == "Black" | Race == "White") %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()
genus_p_bw = phyloseq(genus_phylo, meta_bw)
race_bw = ancombc(genus_p_bw, "Race",
                       p_adj_method = "fdr", lib_cut = 10000,
                       group = "Race")

res_r_df_bw = data.frame(
  Species = row.names(race_bw$res$beta),
  beta = unlist(race_bw$res$beta),
  se = unlist(race_bw$res$se),
  W = unlist(race_bw$res$W),
  p_val = unlist(race_bw$res$p_val),
  q_val = unlist(race_bw$res$q_val),
  diff_abn = unlist(race_bw$res$diff_abn))

fdr_ancom_r_bw <- res_r_df_bw %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r_bw, "Diff_abund_race_genus_bw_db.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_db.csv")
```

```{r study_genera_cioffi}
genus = read_tsv("feature-table-genus.tsv")
taxonomy = genus %>% dplyr::select(Gensu)

genus_matrix = genus %>% column_to_rownames("Gensu") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)

meta_phylo_cioffi = metadata_re_cioffi %>% 
  column_to_rownames("SampleID") %>% 
  sample_data()

genus_p = phyloseq(genus_phylo, meta_phylo_cioffi)

#Race
race = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Race", global = T)

res_r_df = data.frame(
  Species = row.names(race$res$beta),
  beta = unlist(race$res$beta),
  se = unlist(race$res$se),
  W = unlist(race$res$W),
  p_val = unlist(race$res$p_val),
  q_val = unlist(race$res$q_val),
  diff_abn = unlist(race$res$diff_abn))

fdr_ancom_r <- res_r_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Race")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_r, "Diff_abund_global_race_genus_cioffi.csv")

#Ethnicity
ethnicity = ancombc(genus_p, 
               "Race + Ethnicity + Host_age_cat",
                 p_adj_method = "fdr", lib_cut = 10000,
                 group = "Ethnicity", global = T)

res_e_df = data.frame(
  Species = row.names(ethnicity$res$beta),
  beta = unlist(ethnicity$res$beta),
  se = unlist(ethnicity$res$se),
  W = unlist(ethnicity$res$W),
  p_val = unlist(ethnicity$res$p_val),
  q_val = unlist(ethnicity$res$q_val),
  diff_abn = unlist(ethnicity$res$diff_abn))

fdr_ancom_e <- res_e_df %>%
  dplyr::filter(q_val < 0.05) %>% 
  rownames_to_column(var = "Comparison") %>% 
  dplyr::filter(str_detect(Comparison, "^Ethnicity")) %>% 
  inner_join(taxonomy, by = c("Species" = "Gensu"))

write_csv(fdr_ancom_e, "Diff_abund_global_ethnicity_genus_cioffi.csv")
```

###Graphs of DA taxa
Box plots of taxa that were differentially abundant based on ANCOM results and that were identified as important features in the Random Forest model.
```{r da_graphs}
phyla = read_tsv("feature-table-phyla.tsv")
phyla_ra = phyla %>% column_to_rownames("Phyla") %>% 
  decostand(method = "total", MARGIN = 2)
phyla_meta = phyla_ra %>% 
  rownames_to_column("Phyla") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Phyla", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

family = read_tsv("feature-table-family.tsv")
family_ra = family %>% column_to_rownames("Family") %>% 
  decostand(method = "total", MARGIN = 2)
family_meta = family_ra %>% 
  rownames_to_column("Family") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Family", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

genus = read_tsv("feature-table-genus.tsv")
genus_ra = genus %>% column_to_rownames("Gensu") %>% 
  decostand(method = "total", MARGIN = 2)
genus_meta = genus_ra %>% 
  rownames_to_column("Gensu") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Gensu", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

species = read_tsv("feature-table-species.tsv")
species_ra = species %>% column_to_rownames("Species") %>% 
  decostand(method = "total", MARGIN = 2)
species_meta = species_ra %>% 
  rownames_to_column("Species") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Species", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

phyla_meta_OD1 = distinct(phyla_meta, Race)
phyla_meta_OD1$Race = factor(phyla_meta_OD1$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
phyla_meta_OD1$yloc = max(phyla_meta$`k__Bacteria;p__OD1` + 0.05)
phyla_meta_OD1$label = c("A", "C", "B")

plot_OD1 = ggplot(phyla_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__OD1`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("OD1\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = phyla_meta_OD1, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

family_meta_entero = distinct(family_meta, Race)
family_meta_entero$Race = factor(family_meta_entero$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
family_meta_entero$yloc = max(family_meta$`k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae` + 0.05)
family_meta_entero$label = c("A", "B", "B")
plot_Enterobacteriaceae = ggplot(family_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Enterobacteriaceae\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = family_meta_entero, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

genus_meta_bifido = distinct(genus_meta, Race)
genus_meta_bifido$Race = factor(genus_meta_bifido$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
genus_meta_bifido$yloc = max(genus_meta$`k__Bacteria;p__Actinobacteria;c__Actinobacteria;o__Bifidobacteriales;f__Bifidobacteriaceae;g__Bifidobacterium` + 0.05)
genus_meta_bifido$label = c("A", "A", "B")
plot_Bifidobacterium = ggplot(genus_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Actinobacteria;c__Actinobacteria;o__Bifidobacteriales;f__Bifidobacteriaceae;g__Bifidobacterium`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Bifidobacterium\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = genus_meta_bifido, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

genus_meta_blaut = distinct(genus_meta, Race)
genus_meta_blaut$Race = factor(genus_meta_blaut$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
genus_meta_blaut$yloc = max(genus_meta$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae;g__Blautia` + 0.05)
genus_meta_blaut$label = c("A", "C", "B")
plot_Blautia = ggplot(genus_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae;g__Blautia`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Blautia\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = genus_meta_blaut, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

species_meta_bfrag = distinct(species_meta, Race)
species_meta_bfrag$Race = factor(species_meta_bfrag$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_bfrag$yloc = max(species_meta$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__fragilis` + 0.05)
species_meta_bfrag$label = c("A", "B", "A")
plot_B.fragilis = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__fragilis`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Bacteroides fragilis\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = species_meta_bfrag, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

species_meta_cram = distinct(species_meta, Race)
species_meta_cram$Race = factor(species_meta_cram$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_cram$yloc = max(species_meta$`k__Bacteria;p__Firmicutes;c__Erysipelotrichi;o__Erysipelotrichales;f__Erysipelotrichaceae;g__Clostridium;s__ramosum` + 0.05)
species_meta_cram$label = c("A", "A", "B")
plot_C.ramosum = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Erysipelotrichi;o__Erysipelotrichales;f__Erysipelotrichaceae;g__Clostridium;s__ramosum`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Clostridium ramosum\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = species_meta_cram, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

species_meta_caer = distinct(species_meta, Race)
species_meta_caer$Race = factor(species_meta_caer$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_caer$yloc = max(species_meta$`k__Bacteria;p__Actinobacteria;c__Coriobacteriia;o__Coriobacteriales;f__Coriobacteriaceae;g__Collinsella;s__aerofaciens` + 0.05)
species_meta_caer$label = c("A", "B", "B")
plot_C.aerofaciens = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Actinobacteria;c__Coriobacteriia;o__Coriobacteriales;f__Coriobacteriaceae;g__Collinsella;s__aerofaciens`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Collinsella aerofaciens\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = species_meta_caer, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

species_meta_pcop = distinct(species_meta, Race)
species_meta_pcop$Race = factor(species_meta_pcop$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_pcop$yloc = max(species_meta$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri` + 0.05)
species_meta_pcop$label = c("A", "B", "B")
plot_P.copri = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Prevotella copri\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = species_meta_pcop, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

species_meta_rgna = distinct(species_meta, Race)
species_meta_rgna$Race = factor(species_meta_rgna$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_rgna$yloc = max(species_meta$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae;g__[Ruminococcus];s__gnavus` + 0.05)
species_meta_rgna$label = c("A", "C", "B")
plot_R.gnavus = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae;g__[Ruminococcus];s__gnavus`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Ruminococcus gnavus\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = species_meta_rgna, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

species_meta_vdis = distinct(species_meta, Race)
species_meta_vdis$Race = factor(species_meta_vdis$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_vdis$yloc = max(species_meta$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar` + 0.05)
species_meta_vdis$label = c("A", "B", "B")
plot_V.dispar = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Veillonella dispar\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") + 
  geom_text(data = species_meta_vdis, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8) 

plot_Haemo = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Haemophilus\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 0.6, size = 6, 
                     comparisons = list(c("Black", "White")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

plot_Faecali = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Faecalibacterium\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 0.6, size = 6, 
                     comparisons = list(c("Black", "White"),
                                        c("Black", "Asian/Pacific Islander")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))

plot_V.parvula = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Veillonella parvula\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 0.6, size = 6, 
                     comparisons = list(c("Black", "White"),
                                        c("Black", "Asian/Pacific Islander")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "*", "*",
                                                    "ns")))

plot_Dialister = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Dialister;s__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  ylab("Dialister\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(0.8)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(0.8)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  stat_compare_means(label = "p.signif", 
                     bracket.size = 0.6, size = 6, 
                     comparisons = list(c("Black", "White"),
                                        c("White", "Asian/Pacific Islander")),
                     symnum.args = list(cutpoints = 
                                          c(0, 0.0001, 
                                            0.4, 0.5, 1),
                                        symbols = c("***", "**", "*",
                                                    "ns")))
```

###Graphs of highlighted DA taxa
Box plots of taxa that were differentially abundant based on ANCOM results and that highlighted for their relevance to health in the text.
```{r da_highlight_graphs}
phyla = read_tsv("feature-table-phyla.tsv")
phyla_ra = phyla %>% column_to_rownames("Phyla") %>% 
  decostand(method = "total", MARGIN = 2)
phyla_meta = phyla_ra %>% 
  rownames_to_column("Phyla") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Phyla", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

family = read_tsv("feature-table-family.tsv")
family_ra = family %>% column_to_rownames("Family") %>% 
  decostand(method = "total", MARGIN = 2)
family_meta = family_ra %>% 
  rownames_to_column("Family") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Family", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

genus = read_tsv("feature-table-genus.tsv")
genus_ra = genus %>% column_to_rownames("Gensu") %>% 
  decostand(method = "total", MARGIN = 2)
genus_meta = genus_ra %>% 
  rownames_to_column("Gensu") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Gensu", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

species = read_tsv("feature-table-species.tsv")
species_ra = species %>% column_to_rownames("Species") %>% 
  decostand(method = "total", MARGIN = 2)
species_meta = species_ra %>% 
  rownames_to_column("Species") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Species", values_from = "rel_abund") %>% 
  inner_join(metadata_re)

species_meta_pcop = distinct(species_meta, Race)
species_meta_pcop$Race = factor(species_meta_pcop$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_pcop$yloc = 0.8
species_meta_pcop$label = c("A", "B", "B")

species_meta_new = species_meta %>% 
  mutate(P_copri = log10(`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri` + 0.001))
plot_P.copri = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt", limits = c(NA, 0.8)) + 
  ylab("Prevotella copri\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  geom_text(data = species_meta_pcop, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 5) 

species_meta_vdis = distinct(species_meta, Race)
species_meta_vdis$Race = factor(species_meta_vdis$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_vdis$yloc = 0.42
species_meta_vdis$label = c("A", "B", "B")
species_meta_new = species_meta %>% 
  mutate(V_dispar = log10(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar` + 0.001))
plot_V.dispar = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") +
  ylab("Veillonella dispar\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  geom_text(data = species_meta_vdis, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 5) 

species_meta_haem = distinct(species_meta, Race)
species_meta_haem$Race = factor(species_meta_haem$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_haem$yloc = 0.6
species_meta_haem$label = c("A", "A", "B")
species_meta_new = species_meta %>% 
  mutate(Haemo = log10(`k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__` + 0.00001))
plot_Haemo = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") +
  ylab("Haemophilus\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  geom_text(data = species_meta_haem, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 5)

species_meta_faec = distinct(species_meta, Race)
species_meta_faec$Race = factor(species_meta_faec$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_faec$yloc = 0.13
species_meta_faec$label = c("A", "A", "B")
species_meta_new = species_meta %>% 
  mutate(Faecali = log10(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__` + 0.001))
plot_Faecali = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") +
  ylab("Faecalibacterium\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  geom_text(data = species_meta_faec, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 5)

species_meta_vpar = distinct(species_meta, Race)
species_meta_vpar$Race = factor(species_meta_vpar$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_vpar$yloc = 0.8
species_meta_vpar$label = c("A", "A", "B")
species_meta_new = species_meta %>% 
  mutate(V_parv = log10(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula` + 0.0001))
plot_V.parvula = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") +
  ylab("Veillonella parvula\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(1.2)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  geom_text(data = species_meta_vpar, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 5)

species_meta_dial = distinct(species_meta, Race)
species_meta_dial$Race = factor(species_meta_dial$Race, 
                                levels = c("White", 
                                           "Black", 
                                           "Asian/Pacific Islander"))
species_meta_dial$yloc = 1.5
species_meta_dial$label = c("A", "B", "B")
plot_Dialister = ggplot(species_meta,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Dialister;s__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = 'log10') +
  ylab("Dialister\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_text(size=rel(1.1)),
        axis.ticks.x=element_blank()) +
  guides(fill = "none", shape = "none") +
  geom_text(data = species_meta_dial, aes(y = yloc, label = label), 
            position = position_dodge(width = 0.75), size = 8)
```

### Histograms of highlighted taxa
Regular
```{r histogram}
histo_P.copri = gghistogram(species_meta, 
                            x = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri", 
                            y = "..density..",
                            xlab = "Relative Abundance of Prevotella copri ",
                            ylab = "Density",
                            rug = T,
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#f6d128", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge",
                            facet.by = "Race")
histo_P.copri = ggpar(histo_P.copri, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") +
  coord_flip()

histo_V.dispar = gghistogram(species_meta, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar", 
                            y = "..density..",
                            xlab = "Relative Abundance of Veillonella dispar",
                            ylab = "Density",
                            rug = T,
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#f6d128", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge",
                            facet.by = "Race")
histo_V.dispar = ggpar(histo_V.dispar, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_Faecali = gghistogram(species_meta, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__", 
                            y = "..density..",
                            xlab = "Relative Abundance of Faecalibacterium",
                            ylab = "Density",
                            rug = T,
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#f6d128", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge",
                            facet.by = "Race")
histo_Faecali = ggpar(histo_Faecali, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_Haemo = gghistogram(species_meta, 
                            x = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__", 
                            y = "..density..",
                            xlab = "Relative Abundance of Haemophilus",
                            ylab = "Density",
                            rug = T,
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#f6d128", "#e41a1c"),
                          alpha = 0.4,
                            bins = 25,
                            position = "dodge",
                            facet.by = "Race")
histo_Haemo = ggpar(histo_Haemo, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_V.parvula = gghistogram(species_meta, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula", 
                            y = "..density..",
                            xlab = "Relative Abundance of Veillonella parvula",
                            ylab = "Density",
                            rug = T,
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#f6d128", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge",
                            facet.by = "Race")
histo_V.parvula = ggpar(histo_V.parvula, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") +
  rremove("y.title") + rremove("y.ticks") + coord_flip()
```

Back to back
Regular
```{r histogram}
species_meta_a= species_meta %>% filter(Race == "Asian/Pacific Islander")
species_meta_b = species_meta %>% filter(Race == "Black")
species_meta_w = species_meta %>% filter(Race == "White")
species_meta_ab = species_meta %>% filter(Race == "Black" | Race == "Asian/Pacific Islander")
species_meta_aw = species_meta %>% filter(Race == "White" | Race == "Asian/Pacific Islander")
species_meta_bw = species_meta %>% filter(Race == "Black" | Race == "White")

histo_P.copri2 = gghistogram(species_meta_ab, 
                            x = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri", 
                            y = "..density..",
                            xlab = "Relative Abundance of Prevotella copri ",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#f6d128", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_P.copri2 = ggpar(histo_P.copri2, xscale = "sqrt", 
                       legend = "none", orientation = "reverse") + 
  rremove("y.text") + rremove("y.axis") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") +
  coord_flip()

histo_P.copri3 = gghistogram(species_meta_w, 
                            x = "k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri", 
                            y = "..density..",
                            xlab = "Relative Abundance of Prevotella copri ",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_P.copri3 = ggpar(histo_P.copri3, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") +
  rremove("y.axis") + coord_flip()

histo_V.dispar2 = gghistogram(species_meta_ab, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar", 
                            y = "..density..",
                            xlab = "Relative Abundance of Veillonella dispar",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#f6d128", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_V.dispar2 = ggpar(histo_V.dispar2, xscale = "sqrt", 
                       legend = "none", orientation = "reverse") + 
  rremove("y.text") + rremove("y.axis") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_V.dispar3 = gghistogram(species_meta_w, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar", 
                            y = "..density..",
                            xlab = "Relative Abundance of Veillonella dispar",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_V.dispar3 = ggpar(histo_V.dispar3, xscale = "sqrt", 
                       legend = "none") + 
  rremove("y.text") + rremove("y.axis") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_Faecali2 = gghistogram(species_meta_aw, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__", 
                            y = "..density..",
                            xlab = "Relative Abundance of Faecalibacterium",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_Faecali2 = ggpar(histo_Faecali2, xscale = "sqrt", 
                       legend = "none", orientation = "reverse") + rremove("y.text") + rremove("y.axis") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_Faecali3 = gghistogram(species_meta_b, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__", 
                            y = "..density..",
                            xlab = "Relative Abundance of Faecalibacterium",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#f6d128"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_Faecali3 = ggpar(histo_Faecali3, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") + rremove("y.axis") +
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_Haemo2 = gghistogram(species_meta_aw, 
                            x = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__", 
                            y = "..density..",
                            xlab = "Relative Abundance of Haemophilus",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#e41a1c"),
                          alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_Haemo2 = ggpar(histo_Haemo2, xscale = "sqrt", 
                       legend = "none", orientation = "reverse") + rremove("y.text") + rremove("y.axis") + 
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_Haemo3 = gghistogram(species_meta_b, 
                            x = "k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__", 
                            y = "..density..",
                            xlab = "Relative Abundance of Haemophilus",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#f6d128"),
                          alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_Haemo3 = ggpar(histo_Haemo3, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") + rremove("y.axis") + 
  rremove("y.title") + rremove("y.ticks") + rremove("x.title") + coord_flip()

histo_V.parvula2 = gghistogram(species_meta_aw, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula", 
                            y = "..density..",
                            xlab = "Relative Abundance of Veillonella parvula",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#377eb8", "#e41a1c"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_V.parvula2 = ggpar(histo_V.parvula2, xscale = "sqrt", 
                       legend = "none", orientation = "reverse") + rremove("y.text") + rremove("y.axis") + 
  rremove("y.title") + rremove("y.ticks") + coord_flip()

histo_V.parvula3 = gghistogram(species_meta_b, 
                            x = "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula", 
                            y = "..density..",
                            xlab = "Relative Abundance of Veillonella parvula",
                            ylab = "Density",
                            color = "Race",
                            fill = "Race",
                            palette = c("#f6d128"),
                            alpha = 0.4,
                            bins = 25,
                            position = "dodge")
histo_V.parvula3 = ggpar(histo_V.parvula3, xscale = "sqrt", 
                       legend = "none") + rremove("y.text") + rremove("y.axis") + 
  rremove("y.title") + rremove("y.ticks") + coord_flip()
```

###K-S tests
We used a Kolmogorox-Smirnov test to examine if the distribution differed between racial categories for relative abundances of taxa that were differentially abundant in children and adults in the same direction.
```{r ks_test}
species_meta_black = species_meta %>% filter(Race == "Black") 
species_meta_white = species_meta %>% filter(Race == "White")
species_meta_asian = species_meta %>% filter(Race == "Asian/Pacific Islander")

V.dispar.B = species_meta_black$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar`
V.dispar.W = species_meta_white$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar`
V.dispar.A = species_meta_asian$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar`
ks.test(V.dispar.B, V.dispar.W)
ks.test(V.dispar.B, V.dispar.A)
ks.test(V.dispar.W, V.dispar.A)

P.copri.B = species_meta_black$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri`
P.copri.W = species_meta_white$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri`
P.copri.A = species_meta_asian$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri`
ks.test(P.copri.B, P.copri.W)
ks.test(P.copri.B, P.copri.A)
ks.test(P.copri.W, P.copri.A)

Bacter.B = species_meta_black$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;__`
Bacter.W = species_meta_white$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;__`
Bacter.A = species_meta_asian$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;__`
ks.test(Bacter.B, Bacter.W)
ks.test(Bacter.B, Bacter.A)
ks.test(Bacter.W, Bacter.A)

R.bromii.B = species_meta_black$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus;s__bromii`
R.bromii.W = species_meta_white$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus;s__bromii`
R.bromii.A = species_meta_asian$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus;s__bromii`
ks.test(R.bromii.B, R.bromii.W)
ks.test(R.bromii.B, R.bromii.A)
ks.test(R.bromii.W, R.bromii.A)

Faecali.B = species_meta_black$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__`
Faecali.W = species_meta_white$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__`
Faecali.A = species_meta_asian$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__`
ks.test(Faecali.B, Faecali.W)
ks.test(Faecali.B, Faecali.A)
ks.test(Faecali.W, Faecali.A)

A.putre.B = species_meta_black$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__putredinis`
A.putre.W = species_meta_white$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__putredinis`
A.putre.A = species_meta_asian$`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__putredinis`
ks.test(A.putre.B, A.putre.W)
ks.test(A.putre.B, A.putre.A)
ks.test(A.putre.W, A.putre.A)

V.parv.B = species_meta_black$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula`
V.parv.W = species_meta_white$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula`
V.parv.A = species_meta_asian$`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula`
ks.test(V.parv.B, V.parv.W)
ks.test(V.parv.B, V.parv.A)
ks.test(V.parv.W, V.parv.A)

Haemo.B = species_meta_black$`k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__`
Haemo.W = species_meta_white$`k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__`
Haemo.A = species_meta_asian$`k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__`
ks.test(Haemo.B, Haemo.W)
ks.test(Haemo.B, Haemo.A)
ks.test(Haemo.W, Haemo.A)
```

## Presence-absence models
We used a generalized linear mixed effects model with a binomial distribution to examine if race and/or ethnicity was associated with the presence or absence of specific taxa.

### Import data
Presence-absence features tables at the level of phyla, family, genus, species, and ASV were exported from QIIME and imported here (after modifying the heads to remove # characters).
```{r import_pa}
phyla_pa = read_tsv("feature-table-phyla-pa.tsv")
phyla_pa_meta = phyla_pa %>% 
  pivot_longer(-1, names_to = "SampleID", 
               values_to = "presence_absence") %>% 
  pivot_wider(names_from = "Phyla", 
              values_from = "presence_absence") %>% 
  inner_join(metadata_re)

family_pa = read_tsv("feature-table-family-pa.tsv")
family_pa_meta = family_pa %>% 
  pivot_longer(-1, names_to = "SampleID", 
               values_to = "presence_absence") %>% 
  pivot_wider(names_from = "Family", 
              values_from = "presence_absence") %>% 
  inner_join(metadata_re)

genus_pa = read_tsv("feature-table-genus-pa.tsv")
genus_pa_meta = genus_pa %>% 
  pivot_longer(-1, names_to = "SampleID", 
               values_to = "presence_absence") %>% 
  pivot_wider(names_from = "Genus", 
              values_from = "presence_absence") %>% 
  inner_join(metadata_re)

species_pa = read_tsv("feature-table-species-pa.tsv")
species_pa_meta = species_pa %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "presence_absence") %>% 
  pivot_wider(names_from = "Species", values_from = "presence_absence") %>% 
  inner_join(metadata_re)
```
###GLMMs
Run GLMMs using a binomial distribution.
#### Phyla
```{r pa_glmms_phyla}
test_glmm = Anova(glmer(`k__Bacteria;p__OD1` ~ Race + Ethnicity + Host_age_cat + (1|Study), data = phyla_pa_meta, family = binomial))
test_glmm

race_matrix = mat.or.vec(66,3)
eth_matrix = mat.or.vec(66,3)

for(i in 66:67) {
  variable = as.data.frame(phyla_pa_meta)[,i]
  b<-glmer(variable ~ Race + Ethnicity + Host_age_cat + (1|Study), 
                 data = phyla_pa_meta, family = binomial)
  anova = Anova(b)
  race_matrix[i-1,2] = anova[1,1]
  race_matrix[i-1,3] = anova[1,3]
  eth_matrix[i-1,2] = anova[2,1]
  eth_matrix[i-1,3] = anova[2,3]
  race_matrix[i-1,1] = names(phyla_pa_meta)[i]
  eth_matrix[i-1,1] = names(phyla_pa_meta)[i]
}

race_phyla = as.data.frame(race_matrix) %>% rename(Taxa = V1) %>% 
  rename(`Chi-square` = V2) %>% rename(`p-value` = V3)
write.csv(race_phyla, "phyla_race_pa.csv")
race_phyla_nona = filter(race_phyla, race_phyla[,3] != 0)
race_phyla_nona[,3] = as.numeric(as.character(race_phyla_nona[,3]))
race_phyla_corrected = bind_cols(race_phyla_nona, as.data.frame(fdrtool(race_phyla_nona[,3], statistic = "pvalue", plot = F)))
write.csv(race_phyla_corrected, "race_phyla_pa_corrected.csv")

ethnicity_phyla = as.data.frame(ethnicity_matrix)
write.csv(ethnicity_phyla, "phyla_ethnicity_pa.csv")
ethnicity_phyla_nona = filter(ethnicity_phyla, ethnicity_phyla[,3] != 0)
ethnicity_phyla_nona[,3] = as.numeric(as.character(ethnicity_phyla_nona[,3]))
ethnicity_phyla_corrected = bind_cols(ethnicity_phyla_nona, as.data.frame(fdrtool(ethnicity_phyla_nona[,3], statistic = "pvalue", plot = F)))
write.csv(ethnicity_phyla_corrected, "ethnicity_phyla_pa_corrected.csv")
```
#### Species
```{r pa_glmms_species}
test_glmm = Anova(glmer(`k__Bacteria;p__OD1;c__;o__;f__;g__;s__` ~ Race + Ethnicity + Host_age_cat + (1|Study), data = species_pa_meta, family = binomial))
test_glmm

race_matrix = mat.or.vec(1489,3)
eth_matrix = mat.or.vec(1489,3)

for(i in 2:1489) {
  variable = as.data.frame(species_pa_meta)[,i]
  try(b<-glmer(variable ~ Race + Ethnicity + Host_age_cat + (1|Study), 
                 data = species_pa_meta, family = binomial))
  anova = Anova(b)
  race_matrix[i-1,2] = anova[1,1]
  race_matrix[i-1,3] = anova[1,3]
  eth_matrix[i-1,2] = anova[2,1]
  eth_matrix[i-1,3] = anova[2,3]
  race_matrix[i-1,1] = names(species_pa_meta)[i]
  eth_matrix[i-1,1] = names(species_pa_meta)[i]
}

race_species = as.data.frame(race_matrix) %>% rename(Taxa = V1) %>% 
  rename(`Chi-square` = V2) %>% rename(`p-value` = V3)
write.csv(race_species, "species_race_pa.csv")
race_species_nona = filter(race_species, race_species[,3] != 0)
race_species_nona[,3] = as.numeric(as.character(race_species_nona[,3]))
race_species_corrected = bind_cols(race_species_nona, as.data.frame(fdrtool(race_species_nona[,3], statistic = "pvalue", plot = F)))
write.csv(race_species_corrected, "race_species_pa_corrected.csv")

ethnicity_species = as.data.frame(eth_matrix)
write.csv(ethnicity_species, "species_ethnicity_pa.csv")
ethnicity_species_nona = filter(ethnicity_species, ethnicity_species[,3] != 0)
ethnicity_species_nona[,3] = as.numeric(as.character(ethnicity_species_nona[,3]))
ethnicity_species_corrected = bind_cols(ethnicity_species_nona, as.data.frame(fdrtool(ethnicity_species_nona[,3], statistic = "pvalue", plot = F)))
write.csv(ethnicity_species_corrected, "ethnicity_species_pa_corrected.csv")
```
##Missing taxa comparison
To further probe the underlying data structure of taxa that were differentially abundant in children and adults in the same direction, we did a quick comparison of the number of zero abundance values across different categories, we used a Chi-square test to examine whether the proportion of zeros within a category differed from chance.

```{r prop_zero}
V.dispar_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__dispar")
chisq.test(as.data.frame(V.dispar_race))
chisq.test(as.data.frame(V.dispar_race))$observed
chisq.test(as.data.frame(V.dispar_race))$expected
chisq.test(as.data.frame(V.dispar_race))$residuals

P.copri_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__copri")
chisq.test(as.data.frame(P.copri_race))
chisq.test(as.data.frame(P.copri_race))$observed
chisq.test(as.data.frame(P.copri_race))$expected
chisq.test(as.data.frame(P.copri_race))$residuals

Bacter_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;__`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;__")
chisq.test(as.data.frame(Bacter_race))
chisq.test(as.data.frame(Bacter_race))$observed
chisq.test(as.data.frame(Bacter_race))$expected
chisq.test(as.data.frame(Bacter_race))$residuals

R.bromii_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus;s__bromii`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus;s__bromii")
chisq.test(as.data.frame(R.bromii_race))
chisq.test(as.data.frame(R.bromii_race))$observed
chisq.test(as.data.frame(R.bromii_race))$expected
chisq.test(as.data.frame(R.bromii_race))$residuals

Faecali_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;__")
chisq.test(as.data.frame(Faecali_race))
chisq.test(as.data.frame(Faecali_race))$observed
chisq.test(as.data.frame(Faecali_race))$expected
chisq.test(as.data.frame(Faecali_race))$residuals

A.putre_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__putredinis`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__putredinis")
chisq.test(as.data.frame(A.putre_race))
chisq.test(as.data.frame(A.putre_race))$observed
chisq.test(as.data.frame(A.putre_race))$expected
chisq.test(as.data.frame(A.putre_race))$residuals

V.parv_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Veillonellaceae;g__Veillonella;s__parvula")
chisq.test(as.data.frame(V.parv_race))
chisq.test(as.data.frame(V.parv_race))$observed
chisq.test(as.data.frame(V.parv_race))$expected
chisq.test(as.data.frame(V.parv_race))$residuals

Haemo_race = species_pa_meta %>% group_by(Race) %>% 
  count(`k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__`) %>% 
  pivot_wider(names_from = "Race", values_from = "n") %>% 
  column_to_rownames("k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pasteurellales;f__Pasteurellaceae;g__Haemophilus;__")
chisq.test(as.data.frame(Haemo_race))
chisq.test(as.data.frame(Haemo_race))$observed
chisq.test(as.data.frame(Haemo_race))$expected
chisq.test(as.data.frame(Haemo_race))$residuals
```

##Venn diagram
```{r venn}
venn = read_csv("venn_input.csv")
venn_c = venn %>% filter(!is.na(Children))
venn_a = venn %>% filter(!is.na(Adults))
c = venn_c$Children
a = venn_a$Adults

venn.diagram(x = list(c, a), 
             category.names = c("Children (57)", "Adults (144)"), 
             filename = "Venn_BW.png", output = T, imagetype = "png", 
             height = 1000, width = 1000, resolution = 300, 
             col = c("#4b265c", "#25a165"), lwd = 1,
             fill = c(alpha("#4b265c", 0.3), alpha("#25a165", 0.3)), 
             fontfamily = "sans", cat.fontfamily = "sans",
             cat.default.pos = "outer", cat.dist = c(0.1, 0.1),
             margin = 0.2)
```

##ROC curves
Random Forest classification was performed in QIIME and the predictions/probabilities were exported. (pROC)
```{r roc_curve_race}
pred = read_tsv("merged-table-filtered-race-classifier/predictions.tsv",
                col_types = "cf")
prob = read_tsv("merged-table-filtered-race-classifier/probabilities.tsv")

rf = metadata %>% inner_join(pred) %>% 
  inner_join(prob)

response_black = rf %>% dplyr::select(Race, Black) %>% 
  mutate(Race_binary = if_else(Race == "Black", 1, 0))
response_white = rf %>% dplyr::select(Race, White) %>% 
  mutate(Race_binary = if_else(Race == "White", 1, 0))
response_asian = rf %>% dplyr::select(Race, `Asian/Pacific Islander`) %>% 
  mutate(Race_binary = if_else(Race == "Asian/Pacific Islander", 1, 0))

White = roc(response_white$Race_binary ~ response_white$White)
Black = roc(response_black$Race_binary ~ response_black$Black)
Asian = roc(response_asian$Race_binary ~ response_asian$`Asian/Pacific Islander`)
roc_plot = ggroc(list(White, Black, Asian), legacy.axes = TRUE, lwd = 2) + xlab("False Positive Rate") + ylab("True Positive Rate") + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) + 
  theme(axis.title.x=element_text(size=rel(1.2)), 
        axis.title.y=element_text(size=rel(1.2)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1))) +
  annotate(geom = "richtext", 
                                          fill = NA, label.color = NA,
           label = "White, AUC = 0.931<br>
           Black, AUC = 0.965<br>
           Asian/Pacific Islander, AUC = 0.930", 
           x = Inf, y = -Inf, size = 3,
           hjust = 1, vjust = 0) + 
  geom_segment(aes(x = 0, y = 0, xend = 1, yend = 1),
               color = "grey40", linetype = "dashed", size = 1.5) 

plot(White, col="#377eb8", lwd = 4, main = "Random Forest ROC", 
     legacy.axes = TRUE, xlab = "False Positive Rate",
     ylab = "True Positive Rate", grid = T, xlim = c(1,0))
plot(Black, col = "#e41a1c", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
plot(Asian, col = "#ffd92f", lwd = 4, add = TRUE, legacy.axes = TRUE, 
     xlab = "False Positive Rate",
     ylab = "True Positive Rate")
legend("bottomright", 
       legend = c("White, AUC = 0.931", "Black, AUC = 0.965", "Asian/Pacific Islander, AUC = 0.930"), 
       col = c("#377eb8", "#e41a1c", "#ffd92f"), lwd = 4)
```

```{r roc_curve_eth}
pred = read_tsv("merged-table-filtered-ethnicity-classifier/predictions.tsv",
                col_types = "cf")
prob = read_tsv("merged-table-filtered-ethnicity-classifier/probabilities.tsv")

rf = metadata %>% inner_join(pred) %>% 
  inner_join(prob)

response_hispanic = rf %>% dplyr::select(Ethnicity, Hispanic) %>% 
  mutate(Ethnicity_binary = if_else(Ethnicity == "Hispanic", 1, 0))
response_nonhispanic = rf %>% dplyr::select(Ethnicity, `Non Hispanic`) %>% 
  mutate(Ethnicity_binary = if_else(Ethnicity == "Non Hispanic", 1, 0))

Hispanic = roc(response_hispanic$Ethnicity_binary ~ response_hispanic$Hispanic)
NonHispanic = roc(response_nonhispanic$Ethnicity_binary ~ response_nonhispanic$`Non Hispanic`)

roc_plot_eth = ggroc(list(Hispanic, NonHispanic), legacy.axes = TRUE, lwd = 2) + xlab("False Positive Rate") + ylab("True Positive Rate") + 
  scale_color_manual(values = c("darkslategray4", "darkorange4"), 
                     labels = c("Hispanic", "Non Hispanic")) + 
  theme(axis.title.x=element_text(size=rel(1.2)), 
        axis.title.y=element_text(size=rel(1.2)),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1))) +
  annotate(geom = "richtext", 
                                          fill = NA, label.color = NA,
           label = "Hispanic, AUC = 0.948<br>
           Non Hispanic, AUC = 0.948", 
           x = Inf, y = -Inf, size = 3,
           hjust = 1, vjust = 0) + 
  geom_segment(aes(x = 0, y = 0, xend = 1, yend = 1),
               color = "grey40", linetype = "dashed", size = 1.5) 
```



## Machine learning models - children only

In order to maximize the ability to predict categories in adults from models created in children, we used more rigorous machine learning models at the genus level.

First, we developed a model using children to predict children.

### Pre-process the data

```{r child_rf_preprocess}
genus_child = read_tsv("feature-table-genus.tsv") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Gensu", values_from = "rel_abund")
metadata_re = read_tsv("metadata_re.tsv")

genus_child_meta = metadata_re %>% 
  inner_join(genus_child) %>% 
  mutate(outcome = 
           case_when(Race == "Asian/Pacific Islander" ~ "AAPI", 
            Race == "Black" ~ "Black", 
            Race == "White" ~ "White")) %>% 
  dplyr::select(outcome, 41:1198)

genus_child_meta_preprocess = genus_child_meta %>% preprocess_data(outcome_colname = "outcome")
dat = genus_child_meta_preprocess$dat_transformed
```
###Hyperparameter tuning
```{r child_ml_hyptune}
initial_run = run_ml(dataset = dat, 
         method = "rf", 
         outcome_colname = "outcome", 
         cv_times = 100, 
         seed = 1018)

new_hp = list(mtry = c(6, 9, 11, 12, 14, 17))
test_hp = function(seed) {
  run_ml(dataset = dat, 
         method = "rf", 
         outcome_colname = "outcome", 
         cv_times = 100, 
         hyperparameters = new_hp,
         seed = seed)
}

plan("multisession")

iterative_run_ml_results = future_map(1:100, test_hp,
                                      .options = furrr_options(seed = T))

performance = iterative_run_ml_results %>%
  map(pluck, "trained_model") %>% 
  combine_hp_performance()
tiff("hyperparameter_tuning.tif", width = 6, height = 4, 
     units = "in", res = 300)
plot_hp_performance(performance$dat, mtry, logLoss)
dev.off()

performance$dat %>% group_by(mtry) %>% 
  summarize(mean_logLoss = mean(logLoss), 
            lquartile = quantile(logLoss, prob=0.25),
            uquartile = quantile(logLoss, prob=0.75),
                                 .groups = "drop") %>% 
  slice_min(n=3, mean_logLoss)

iterative_run_ml_results %>%
  map(pluck, "trained_model") %>% 
  combine_hp_performance() %>% 
  pluck("dat") %>% 
  write_tsv("child_genus_ml_pooled_testhp.tsv")
  
lapply(iterative_run_ml_results, function(result) {
  result[['performance']] %>% 
    select(cv_metric_logLoss, Accuracy, method)
}) %>% 
  bind_rows() %>% 
  write_tsv("child_genus_ml_pooled_testperformance.tsv")

plan("sequential")
```

###Run final model
```{r child_ml_finalmodel}
final_hp = list(mtry = c(11,12,13))
genus_race_children_ml = function(seed) {
  run_ml(dataset = dat, 
         method = "rf", 
         outcome_colname = "outcome", 
         cv_times = 100, 
         find_feature_importance = T,
         hyperparameters = final_hp,
         seed = seed)
}

plan("multisession")

child_ml_results = future_map(1:100, genus_race_children_ml,
                              .options = furrr_options(seed = T))

child_ml_results %>%
  map(pluck, "trained_model") %>% 
  combine_hp_performance() %>% 
  pluck("dat") %>% 
  write_tsv("child_genus_ml_pooled_hp.tsv")

saveRDS(child_ml_results, file = "child_ml_results.rds")

plan("sequential")

lapply(child_ml_results, function(result) {
  result[['performance']] %>% 
    select(cv_metric_logLoss, Accuracy, method)
}) %>% 
  bind_rows() %>%  
  write_tsv("child_genus_ml_pooled_performance.tsv")

get_importance = function(model_num) {
  model = child_ml_results[[model_num]]
  
  model$feature_importance %>% 
    as_tibble() %>% 
    mutate(seed = model_num)
}

importance = map_dfr(1:100, get_importance)
write_tsv(importance, "feature_importance_children.tsv")

feature_importance_plot = ggplot(as.data.frame(importance), aes(x = perf_metric, y = names)) + 
  stat_summary(fun.data = "median_hilow",
               fun.args = list(conf.int = 0.5))

tiff("feature_importance_children.tiff", width = 10, height = 5, units = "in", res = 300)
feature_importance_plot
dev.off()

importance_summary = importance %>% 
  select(-perf_metric_diff, -pvalue, 
         -method, -perf_metric_name) %>% 
  pivot_wider(id_cols = "names", 
              names_from = "seed", 
              names_prefix = "seed",
              values_from = "perf_metric") %>% 
  mutate(mean_importance = rowMeans(select(., starts_with("seed"))))

write_csv(importance_summary, "feature_importance_children_mean.csv")
```

###Roc curves
```{r roc_curves_child}
get_sensitivity = function(x, data) {
  data %>% 
    filter(specificity - x >= 0) %>% 
    top_n(sensitivity, n = 1) %>% 
    mutate(specificity = x, fpr = 1-x) %>% 
    distinct()
  
}

get_pooled_sens_spec_black = function(model_num, specificity) {
  model = child_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_black = bind_cols(prob, observed = observed) %>% dplyr::select(Black, observed)
  total_black = prob_obs_black %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_black = prob_obs_black %>% arrange(desc(Black)) %>% 
    mutate(is_black = (observed == "Black"),
           tp = cumsum(is_black),
           fp = cumsum(!is_black),
           sensitivity = tp / total_black$Black,
           fpr = fp / (total_black$AAPI + total_black$White),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_black) %>% 
    mutate(Race = "Black")
  
}

get_pooled_sens_spec_white = function(model_num, specificity) {
  model = child_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_white = bind_cols(prob, observed = observed) %>% dplyr::select(White, observed)
  total_white = prob_obs_white %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_white = prob_obs_white %>% arrange(desc(White)) %>% 
    mutate(is_white = (observed == "White"),
           tp = cumsum(is_white),
           fp = cumsum(!is_white),
           sensitivity = tp / total_white$White,
           fpr = fp / (total_white$AAPI + total_white$Black),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_white) %>% 
    mutate(Race = "White")
  
}

get_pooled_sens_spec_aapi = function(model_num, specificity) {
  model = child_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_aapi = bind_cols(prob, observed = observed) %>% dplyr::select(AAPI, observed)
  total_aapi = prob_obs_aapi %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_aapi = prob_obs_aapi %>% arrange(desc(AAPI)) %>% 
    mutate(is_aapi = (observed == "AAPI"),
           tp = cumsum(is_aapi),
           fp = cumsum(!is_aapi),
           sensitivity = tp / total_aapi$AAPI,
           fpr = fp / (total_aapi$White + total_aapi$Black),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_aapi) %>% 
    mutate(Race = "Asian/Pacific Islander")
  
}

child_ml_results = readRDS("AGP_comp/child_ml_results.rds")

performance = lapply(child_ml_results, function(result) {
  result[['performance']] %>% 
    dplyr::select(AUC, prAUC, Accuracy, Mean_Sensitivity, Mean_Specificity)
}) %>% 
  bind_rows() %>% 
  summarize(AUC = median(AUC), prAUC = median(prAUC),
            Accuracy = median(Accuracy), 
            Sensitivity = median(Mean_Sensitivity),
            Specificity = median(Mean_Specificity))

specificity = seq(0, 1, 0.01)

pooled_sens_spec_black = map_dfr(1:100, get_pooled_sens_spec_black, specificity)
pooled_sens_spec_white = map_dfr(1:100, get_pooled_sens_spec_white, specificity)
pooled_sens_spec_aapi = map_dfr(1:100, get_pooled_sens_spec_aapi, specificity)

pooled_sens_spec = bind_rows(pooled_sens_spec_black, pooled_sens_spec_white, pooled_sens_spec_aapi)
pooled_sens_spec$Race = factor(pooled_sens_spec$Race, levels = c("White", "Black", "Asian/Pacific Islander"))

saveRDS(pooled_sens_spec, "pooled_sens_spec_r.rds")

roc_curve_child_r = pooled_sens_spec %>% group_by(Race, specificity) %>% 
  summarize(lquartile = quantile(sensitivity, prob = 0.25),
            uquartile = quantile(sensitivity, prob = 0.75),
            sensitivity = median(sensitivity),
            .groups = "drop") %>% 
  ggplot(aes(x=1-specificity, y=sensitivity, 
             ymin=lquartile, ymax=uquartile)) + 
  geom_ribbon(alpha=0.25, aes(fill = Race)) + 
  geom_step(aes(color = Race), size = 0.8) + 
  xlab("False Positive Rate") + ylab("True Positive Rate") + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) + 
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "AUC = 0.914<br>
           prAUC = 0.629<br>
           Accuracy = 0.879<br>
           Sensitivity = 0.489<br>
           Specificity = 0.795", 
           x = Inf, y = -Inf, size = 3,
           hjust = 1, vjust = 0) + 
  geom_segment(aes(x = 0, y = 0, xend = 1, yend = 1),
               color = "grey40", linetype = "dashed", size = 0.6) 

roc_curve_child_r

ggsave("roc_curve_child.tiff", width = 7, height = 5)

```

```{r roc_curves_child_eth}
get_sensitivity = function(x, data) {
  data %>% 
    filter(specificity - x >= 0) %>% 
    top_n(sensitivity, n = 1) %>% 
    mutate(specificity = x, fpr = 1-x) %>% 
    distinct()
  
}

get_pooled_sens_spec_his = function(model_num, specificity) {
  model = child_eth_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_his = bind_cols(prob, observed = observed) %>% dplyr::select(Hispanic, observed)
  total_his = prob_obs_his %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_his = prob_obs_his %>% arrange(desc(Hispanic)) %>% 
    mutate(is_his = (observed == "Hispanic"),
           tp = cumsum(is_his),
           fp = cumsum(!is_his),
           sensitivity = tp / total_his$Hispanic,
           fpr = fp / (total_his$NonHispanic),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_his) %>% 
    mutate(Ethnicity = "Hispanic")
  
}

get_pooled_sens_spec_nhis = function(model_num, specificity) {
  model = child_eth_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_nhis = bind_cols(prob, observed = observed) %>% dplyr::select(NonHispanic, observed)
  total_nhis = prob_obs_nhis %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_nhis = prob_obs_nhis %>% arrange(desc(NonHispanic)) %>% 
    mutate(is_nhis = (observed == "NonHispanic"),
           tp = cumsum(is_nhis),
           fp = cumsum(!is_nhis),
           sensitivity = tp / total_nhis$NonHispanic,
           fpr = fp / (total_nhis$Hispanic),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_nhis) %>% 
    mutate(Ethnicity = "Non Hispanic")
  
}

child_eth_ml_results = readRDS("AGP_comp/child_ml_eth_results.rds")

performance = lapply(child_eth_ml_results, function(result) {
  result[['performance']] %>% 
    dplyr::select(AUC, prAUC, Accuracy, Sensitivity, Specificity)
}) %>% 
  bind_rows() %>% 
  summarize(AUC = median(AUC), prAUC = median(prAUC),
            Accuracy = median(Accuracy), 
            Sensitivity = median(Sensitivity),
            Specificity = median(Specificity))

specificity = seq(0, 1, 0.01)

pooled_sens_spec_his = map_dfr(1:100, get_pooled_sens_spec_his, specificity)
pooled_sens_spec_nhis = map_dfr(1:100, get_pooled_sens_spec_nhis, specificity)

pooled_sens_spec = bind_rows(pooled_sens_spec_his, pooled_sens_spec_nhis)
pooled_sens_spec$Ethnicity = factor(pooled_sens_spec$Ethnicity, levels = c("Hispanic", "Non Hispanic"))
saveRDS(pooled_sens_spec, "pooled_sens_spec_e.rds")
pooled_sens_spec = readRDS("pooled_sens_spec_e.rds")

roc_curve_child_e = pooled_sens_spec %>% group_by(Ethnicity, specificity) %>% 
  summarize(lquartile = quantile(sensitivity, prob = 0.25),
            uquartile = quantile(sensitivity, prob = 0.75),
            sensitivity = median(sensitivity),
            .groups = "drop") %>% 
  ggplot(aes(x=1-specificity, y=sensitivity, 
             ymin=lquartile, ymax=uquartile)) + 
  geom_ribbon(alpha=0.25, aes(fill = Ethnicity)) + 
  geom_step(aes(color = Ethnicity), size = 0.8) + 
  xlab("False Positive Rate") + ylab("True Positive Rate") + 
  scale_color_manual(values = c("darkslategray4", "darkorange4"), 
                     labels = c("Hispanic", "Non Hispanic")) + 
  scale_fill_manual(values = c("darkslategray4", "darkorange4"), 
                     labels = c("Hispanic", "Non Hispanic")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "AUC = 0.886<br>
           prAUC = 0.797<br>
           Accuracy = 0.892<br>
           Sensitivity = 0.983<br>
           Specificity = 0.354", 
           x = Inf, y = -Inf, size = 3,
           hjust = 1, vjust = 0) + 
  geom_segment(aes(x = 0, y = 0, xend = 1, yend = 1),
               color = "grey40", linetype = "dashed", size = 0.6) 

roc_curve_child_e

ggsave("roc_curve_child_eth.tiff", width = 7, height = 5)

```

###AUC calculations
Area under the curve was calculated for individual ROC curves using the pROC package.
```{r oldroc_child}
child_ml_results = readRDS("AGP_comp/child_ml_results.rds")

get_roc_auc_black = function(model_num) {
  rf = child_ml_results[[model_num]]$trained_model$pred 
  response = rf %>% 
    dplyr::select(obs, pred) %>% 
    mutate(obs_binary = if_else(obs == "Black", 1, 0)) %>% 
    mutate(pred_binary = if_else(pred == "Black", 1, 0))
  roc = roc(response$obs_binary ~ response$pred_binary)
  roc$auc[1]

}  

get_roc_auc_white = function(model_num) {
  rf = child_ml_results[[model_num]]$trained_model$pred 
  response = rf %>% 
    dplyr::select(obs, pred) %>% 
    mutate(obs_binary = if_else(obs == "White", 1, 0)) %>% 
    mutate(pred_binary = if_else(pred == "White", 1, 0))
  roc = roc(response$obs_binary ~ response$pred_binary)
  roc$auc[1]

}

get_roc_auc_aapi = function(model_num) {
  rf = child_ml_results[[model_num]]$trained_model$pred 
  response = rf %>% 
    dplyr::select(obs, pred) %>% 
    mutate(obs_binary = if_else(obs == "AAPI", 1, 0)) %>% 
    mutate(pred_binary = if_else(pred == "AAPI", 1, 0))
  roc = roc(response$obs_binary ~ response$pred_binary)
  roc$auc[1]

}

auc_black = map(1:100, get_roc_auc_black) %>% unlist() %>% median()
auc_white = map(1:100, get_roc_auc_white) %>% unlist() %>% median()
auc_aapi = map(1:100, get_roc_auc_aapi) %>% unlist() %>% median()

rf = child_ml_results[[1]]$trained_model$pred

response_black = rf %>% dplyr::select(obs, pred) %>% 
  mutate(obs_binary = if_else(obs == "Black", 1, 0)) %>% 
  mutate(pred_binary = if_else(pred == "Black", 1, 0))
response_white = rf %>% dplyr::select(obs, White) %>% 
  mutate(obs_binary = if_else(obs == "White", 1, 0))
response_asian = rf %>% dplyr::select(obs, AAPI) %>% 
  mutate(obs_binary = if_else(obs == "AAPI", 1, 0))

White = roc(response_white$obs_binary ~ response_white$White)
Black = roc(response_black$obs_binary ~ response_black$pred_binary)
Asian = roc(response_asian$obs_binary ~ response_asian$AAPI)
```

## Machine learning models - children and adults

Second, we used data from children to predict adult metadata.

Genus-level feature tables from studies included in the analysis of children and from the American Gut Project data were merged and imported for this analysis. Data is preprocessed and hyperparameters are tuned below.

###Preprocess the data

```{r merge_ml_preprocess}
Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 2)

genus_merged = read_tsv("AGP_comp/merged-adult-child-feature-table-genus.tsv") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Genus", values_from = "rel_abund")
metadata_merged = read_tsv("AGP_comp/merged-adult-child-metadata.txt")

genus_merged_meta = metadata_merged %>% 
  inner_join(genus_merged) %>% 
  mutate(outcome = 
           case_when(Race == "Asian/Pacific Islander" ~ "AAPI", 
            Race == "Black" ~ "Black", 
            Race == "White" ~ "White")) %>% 
  dplyr::select(outcome, 5:1504)

genus_merged_meta_groups = metadata_merged %>% 
  inner_join(genus_merged) %>% 
  dplyr::select(Group)

genus_merged_meta_preprocess = genus_merged_meta %>% preprocess_data(outcome_colname = "outcome")
dat = genus_merged_meta_preprocess$dat_transformed
grps = genus_merged_meta_groups$Group
```

###Tune hyperparameters
```{r merge_ml_hyptune}
initial_run = run_ml(dataset = dat, 
         method = "rf", 
         outcome_colname = "outcome", 
         cv_times = 100, 
         seed = 1018,
         groups = grps,
         group_partitions = list(train = c('C'),
                                 test = c('A')))

new_hp = list(mtry = c(8, 9, 10, 11, 12))
test_hp = function(seed) {
  run_ml(dataset = dat, 
         method = "rf", 
         outcome_colname = "outcome", 
         cv_times = 100, 
         hyperparameters = new_hp,
         seed = seed,
         groups = grps,
         group_partitions = list(train = c('C'),
                                 test = c('A')))
}



plan("multisession")

iterative_run_ml_results = future_map(1:100, test_hp,
                                      .options = furrr_options(seed = T))

performance = iterative_run_ml_results %>%
  map(pluck, "trained_model") %>% 
  combine_hp_performance()
tiff("hyperparameter_tuning_merged.tif", width = 6, height = 4, 
     units = "in", res = 300)
plot_hp_performance(performance$dat, mtry, logLoss)
dev.off()
```

```{r merge_ml_finalmodel}
final_hp = list(mtry = c(8,9,10))

genus_race_merged_ml = function(seed) {
  run_ml(dataset = dat, 
         method = "rf", 
         outcome_colname = "outcome", 
         cv_times = 100, 
         find_feature_importance = T,
         hyperparameters = final_hp,
         seed = seed,
         groups = grps,
         group_partitions = list(train = c('C'),
                                 test = c('A')))
}

plan("multisession")

merged_ml_results = future_map(1:100, genus_race_merged_ml,
                              .options = furrr_options(seed = T))

saveRDS(merged_ml_results, "merged_ml_results.rds")

plan("sequential")

merged_ml_results %>%
  map(pluck, "trained_model") %>% 
  combine_hp_performance() %>% 
  pluck("dat") %>% 
  write_tsv("merged_genus_ml_pooled_hp.tsv")

lapply(merged_ml_results, function(result) {
  result[['performance']] %>% 
    select(cv_metric_logLoss, Accuracy, method)
}) %>% 
  bind_rows() %>%  
  write_tsv("merged_genus_ml_pooled_performance.tsv")

get_importance = function(model_num) {
  model = merged_ml_results[[model_num]]
  
  model$feature_importance %>% 
    as_tibble() %>% 
    mutate(seed = model_num)
}

importance = map_dfr(1:100, get_importance)
write_tsv(importance, "feature_importance_merged.tsv")

feature_importance_plot = ggplot(as.data.frame(importance), aes(x = perf_metric, y = names)) + 
  stat_summary(fun.data = "median_hilow",
               fun.args = list(conf.int = 0.5))

tiff("feature_importance_merged.tiff", width = 10, height = 5, units = "in", res = 300)
feature_importance_plot
dev.off()

importance_summary = importance %>% 
  select(-perf_metric_diff, -pvalue, 
         -method, -perf_metric_name) %>% 
  pivot_wider(id_cols = "names", 
              names_from = "seed", 
              names_prefix = "seed",
              values_from = "perf_metric") %>% 
  mutate(mean_importance = rowMeans(select(., starts_with("seed"))))

write_csv(importance_summary, "feature_importance_merged_mean.csv")
```

###Roc curves
```{r merge_ml_roc}
merged_ml_results = readRDS("merged_ml_results.rds")

get_sensitivity = function(x, data) {
  data %>% 
    filter(specificity - x >= 0) %>% 
    top_n(sensitivity, n = 1) %>% 
    mutate(specificity = x, fpr = 1-x) %>% 
    distinct()
  
}

get_pooled_sens_spec_black = function(model_num, specificity) {
  model = merged_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_black = bind_cols(prob, observed = observed) %>% dplyr::select(Black, observed)
  total_black = prob_obs_black %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_black = prob_obs_black %>% arrange(desc(Black)) %>% 
    mutate(is_black = (observed == "Black"),
           tp = cumsum(is_black),
           fp = cumsum(!is_black),
           sensitivity = tp / total_black$Black,
           fpr = fp / (total_black$AAPI + total_black$White),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_black) %>% 
    mutate(Race = "Black")
  
}

get_pooled_sens_spec_white = function(model_num, specificity) {
  model = merged_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_white = bind_cols(prob, observed = observed) %>% dplyr::select(White, observed)
  total_white = prob_obs_white %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_white = prob_obs_white %>% arrange(desc(White)) %>% 
    mutate(is_white = (observed == "White"),
           tp = cumsum(is_white),
           fp = cumsum(!is_white),
           sensitivity = tp / total_white$White,
           fpr = fp / (total_white$AAPI + total_white$Black),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_white) %>% 
    mutate(Race = "White")
  
}

get_pooled_sens_spec_aapi = function(model_num, specificity) {
  model = merged_ml_results[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_aapi = bind_cols(prob, observed = observed) %>% dplyr::select(AAPI, observed)
  total_aapi = prob_obs_aapi %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_aapi = prob_obs_aapi %>% arrange(desc(AAPI)) %>% 
    mutate(is_aapi = (observed == "AAPI"),
           tp = cumsum(is_aapi),
           fp = cumsum(!is_aapi),
           sensitivity = tp / total_aapi$AAPI,
           fpr = fp / (total_aapi$White + total_aapi$Black),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_aapi) %>% 
    mutate(Race = "Asian/Pacific Islander")
  
}

performance = lapply(merged_ml_results, function(result) {
  result[['performance']] %>% 
    dplyr::select(AUC, prAUC, Accuracy, Mean_Sensitivity, Mean_Specificity)
}) %>% 
  bind_rows() %>% 
  summarize(AUC = median(AUC), prAUC = median(prAUC),
            Accuracy = median(Accuracy), 
            Sensitivity = median(Mean_Sensitivity),
            Specificity = median(Mean_Specificity))

specificity = seq(0, 1, 0.01)

pooled_sens_spec_black = map_dfr(1:100, get_pooled_sens_spec_black, specificity)
pooled_sens_spec_white = map_dfr(1:100, get_pooled_sens_spec_white, specificity)
pooled_sens_spec_aapi = map_dfr(1:100, get_pooled_sens_spec_aapi, specificity)

pooled_sens_spec = bind_rows(pooled_sens_spec_black, pooled_sens_spec_white, pooled_sens_spec_aapi)
pooled_sens_spec$Race = factor(pooled_sens_spec$Race, levels = c("White", "Black", "Asian/Pacific Islander"))

pooled_sens_spec %>% group_by(Race, specificity) %>% 
  summarize(lquartile = quantile(sensitivity, prob = 0.25),
            uquartile = quantile(sensitivity, prob = 0.75),
            sensitivity = median(sensitivity),
            .groups = "drop") %>% 
  ggplot(aes(x=1-specificity, y=sensitivity, 
             ymin=lquartile, ymax=uquartile)) + 
  geom_ribbon(alpha=0.25, aes(fill = Race)) + 
  geom_step(aes(color = Race), size = 0.8) + 
  xlab("False Positive Rate") + ylab("True Positive Rate") + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) + 
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "AUC = 0.570<br>
           prAUC = 0.344<br>
           Accuracy = 0.922<br>
           Sensitivity = 0.333<br>
           Specificity = 0.668", 
           x = Inf, y = -Inf, size = 3,
           hjust = 1, vjust = 0) + 
  geom_segment(aes(x = 0, y = 0, xend = 1, yend = 1),
               color = "grey40", linetype = "dashed", size = 0.6) 

ggsave("roc_curve_merge.tiff", width = 7, height = 5)
```

###AUC calculations
```{r merge_ml_auc}
merged_ml_results = readRDS("merged_ml_results.rds")

get_roc_auc_black = function(model_num) {
  rf = merged_ml_results[[model_num]]$trained_model$pred 
  response = rf %>% 
    dplyr::select(obs, pred) %>% 
    mutate(obs_binary = if_else(obs == "Black", 1, 0)) %>% 
    mutate(pred_binary = if_else(pred == "Black", 1, 0))
  roc = roc(response$obs_binary ~ response$pred_binary)
  roc$auc[1]

}  

get_roc_auc_white = function(model_num) {
  rf = merged_ml_results[[model_num]]$trained_model$pred 
  response = rf %>% 
    dplyr::select(obs, pred) %>% 
    mutate(obs_binary = if_else(obs == "White", 1, 0)) %>% 
    mutate(pred_binary = if_else(pred == "White", 1, 0))
  roc = roc(response$obs_binary ~ response$pred_binary)
  roc$auc[1]

}

get_roc_auc_aapi = function(model_num) {
  rf = merged_ml_results[[model_num]]$trained_model$pred 
  response = rf %>% 
    dplyr::select(obs, pred) %>% 
    mutate(obs_binary = if_else(obs == "AAPI", 1, 0)) %>% 
    mutate(pred_binary = if_else(pred == "AAPI", 1, 0))
  roc = roc(response$obs_binary ~ response$pred_binary)
  roc$auc[1]

}

auc_black = map(1:100, get_roc_auc_black) %>% unlist() %>% median()
auc_white = map(1:100, get_roc_auc_white) %>% unlist() %>% median()
auc_aapi = map(1:100, get_roc_auc_aapi) %>% unlist() %>% median()

rf = merged_ml_results[[3]]$trained_model$pred

response_black = rf %>% dplyr::select(obs, pred) %>% 
  mutate(obs_binary = if_else(obs == "Black", 1, 0)) %>% 
  mutate(pred_binary = if_else(pred == "Black", 1, 0))
response_white = rf %>% dplyr::select(obs, White) %>% 
  mutate(obs_binary = if_else(obs == "White", 1, 0))
response_asian = rf %>% dplyr::select(obs, AAPI) %>% 
  mutate(obs_binary = if_else(obs == "AAPI", 1, 0))

White = roc(response_white$obs_binary ~ response_white$White)
Black = roc(response_black$obs_binary ~ response_black$pred_binary)
Asian = roc(response_asian$obs_binary ~ response_asian$AAPI)
```
##Machine learning child-adult downsampled
We then downsampled the feature tables to evenly sample across groups. 50 Black children, 50 White children, 50 Asian American children, 50 Black adults, 50 White adults, and 50 Asian American adults were randomly selected from the tables. This was repeated 9 more times for a total of 10 randomly downsampled tables. All preprocessing, hyperparameter tuning, and random forest modeling steps were then performed as above for each of the 10 tables.

###Preprocess
```{r preprocess_ac_down}
plan("multisession")

genus_merged = read_tsv("~/merged-adult-child-feature-table-genus.tsv") %>%
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>%
  pivot_wider(names_from = "Genus", values_from = "rel_abund")
metadata_merged = read_tsv("~/merged-adult-child-metadata.txt")

genus_merged_meta = metadata_merged %>%
  inner_join(genus_merged) %>%
  mutate(outcome =
           case_when(Race == "Asian/Pacific Islander" ~ "AAPI",
                     Race == "Black" ~ "Black",
                     Race == "White" ~ "White"))
genus_merged_down = replicate(n=10, genus_merged_meta %>%
                                group_by(Race, Group) %>%
                                slice_sample(n=50) %>%
                                ungroup(), simplify = F)
genus_merged_down_go = lapply(genus_merged_down, function(down) {
  down %>% dplyr::select(outcome, 5:1504) %>%
    preprocess_data(outcome_colname = "outcome")
})
dat = lapply(genus_merged_down_go, function(processed) {
  processed$dat_transformed
})
grps = lapply(genus_merged_down, function(down) {
  down$Group
})

saveRDS(dat, "dat_down.rds")
saveRDS(grps, "groups_down.rds")

dat_down = readRDS("AGP_comp/dat_down.rds")
```

###Hyperparameter tuning
```{r hyper_ac_down}
new_hp = list(mtry = c(15, 17, 18, 19, 21, 24))
test_hp = function(seed) {
  run_ml(dataset = dat[[1]],
         method = "rf",
         outcome_colname = "outcome",
         cv_times = 100,
         hyperparameters = new_hp,
         seed = seed,
         groups = grps[[1]],
         group_partitions = list(train = c('C'),
                                 test = c('A')))
}

iterative_run_ml_results = future_map2(1:50, test_hp,
                                      .options = furrr_options(seed = T))

saveRDS(iterative_run_ml_results, "~/iterative_merged_down_ml.rds")

iterative_run_ml_results_1 = readRDS("AGP_comp/iterative_merged_down_ml_1.rds")

performance = iterative_run_ml_results_1 %>%
  map(pluck, "trained_model") %>% 
  combine_hp_performance()
tiff("hyperparameter_tuning_merged_down2.tif", width = 6, height = 4, 
     units = "in", res = 300)
plot_hp_performance(performance$dat, mtry, logLoss)
dev.off()

performance = lapply(iterative_run_ml_results_1, function(result) {
  result[['performance']] %>% 
    dplyr::select(AUC, prAUC, Accuracy, Mean_Sensitivity, Mean_Specificity)
}) %>% 
  bind_rows() %>% 
  summarize(AUC = median(AUC), prAUC = median(prAUC),
            Accuracy = median(Accuracy), 
            Sensitivity = median(Mean_Sensitivity),
            Specificity = median(Mean_Specificity))

get_sensitivity = function(x, data) {
  data %>% 
    filter(specificity - x >= 0) %>% 
    top_n(sensitivity, n = 1) %>% 
    mutate(specificity = x, fpr = 1-x) %>% 
    distinct()
  
}

get_pooled_sens_spec_black = function(model_num, specificity) {
  model = iterative_run_ml_results_1[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_black = bind_cols(prob, observed = observed) %>% dplyr::select(Black, observed)
  total_black = prob_obs_black %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_black = prob_obs_black %>% arrange(desc(Black)) %>% 
    mutate(is_black = (observed == "Black"),
           tp = cumsum(is_black),
           fp = cumsum(!is_black),
           sensitivity = tp / total_black$Black,
           fpr = fp / (total_black$AAPI + total_black$White),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_black) %>% 
    mutate(Race = "Black")
  
}

get_pooled_sens_spec_white = function(model_num, specificity) {
  model = iterative_run_ml_results_1[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_white = bind_cols(prob, observed = observed) %>% dplyr::select(White, observed)
  total_white = prob_obs_white %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_white = prob_obs_white %>% arrange(desc(White)) %>% 
    mutate(is_white = (observed == "White"),
           tp = cumsum(is_white),
           fp = cumsum(!is_white),
           sensitivity = tp / total_white$White,
           fpr = fp / (total_white$AAPI + total_white$Black),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_white) %>% 
    mutate(Race = "White")
  
}

get_pooled_sens_spec_aapi = function(model_num, specificity) {
  model = iterative_run_ml_results_1[[model_num]]
  prob = predict(model$trained_model, model$test_data, type = "prob")
  observed = model$test_data$outcome
  prob_obs_aapi = bind_cols(prob, observed = observed) %>% dplyr::select(AAPI, observed)
  total_aapi = prob_obs_aapi %>% count(observed) %>% 
    pivot_wider(names_from = observed, values_from = n)
  sens_spec_aapi = prob_obs_aapi %>% arrange(desc(AAPI)) %>% 
    mutate(is_aapi = (observed == "AAPI"),
           tp = cumsum(is_aapi),
           fp = cumsum(!is_aapi),
           sensitivity = tp / total_aapi$AAPI,
           fpr = fp / (total_aapi$White + total_aapi$Black),
           specificity = 1 - fpr) %>% 
    dplyr::select(sensitivity, specificity, fpr)
  map_dfr(specificity, get_sensitivity, sens_spec_aapi) %>% 
    mutate(Race = "Asian/Pacific Islander")
  
}

specificity = seq(0, 1, 0.01)

pooled_sens_spec_black = map_dfr(1:50, get_pooled_sens_spec_black, specificity)
pooled_sens_spec_white = map_dfr(1:50, get_pooled_sens_spec_white, specificity)
pooled_sens_spec_aapi = map_dfr(1:50, get_pooled_sens_spec_aapi, specificity)

pooled_sens_spec = bind_rows(pooled_sens_spec_black, pooled_sens_spec_white, pooled_sens_spec_aapi)
pooled_sens_spec$Race = factor(pooled_sens_spec$Race, levels = c("White", "Black", "Asian/Pacific Islander"))

pooled_sens_spec %>% group_by(Race, specificity) %>% 
  summarize(lquartile = quantile(sensitivity, prob = 0.25),
            uquartile = quantile(sensitivity, prob = 0.75),
            sensitivity = median(sensitivity),
            .groups = "drop") %>% 
  ggplot(aes(x=1-specificity, y=sensitivity, 
             ymin=lquartile, ymax=uquartile)) + 
  geom_ribbon(alpha=0.25, aes(fill = Race)) + 
  geom_step(aes(color = Race), size = 0.8) + 
  xlab("False Positive Rate") + ylab("True Positive Rate") + 
  scale_color_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) + 
  scale_fill_manual(values = c("#377eb8", "#ffd92f", "#e41a1c"), 
                     labels = c("White", "Black", 
                                "Asian/Pacific Islander")) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "AUC = 0.632<br>
           prAUC = 0.475<br>
           Accuracy = 0.483<br>
           Sensitivity = 0.483<br>
           Specificity = 0.742", 
           x = Inf, y = -Inf, size = 3,
           hjust = 1, vjust = 0) + 
  geom_segment(aes(x = 0, y = 0, xend = 1, yend = 1),
               color = "grey40", linetype = "dashed", size = 0.6)

ggsave("roc_curve_merge_down_1.tiff", width = 7, height = 5)
```

###Final model
```{r final_model_merged_down}

```

##Important feature graphs
```{r imp_feat_graphs}
genus_merged = read_tsv("AGP_comp/merged-adult-child-feature-table-genus.tsv") %>% 
  pivot_longer(-1, names_to = "SampleID", values_to = "rel_abund") %>% 
  pivot_wider(names_from = "Genus", values_from = "rel_abund")
metadata_merged = read_tsv("AGP_comp/merged-adult-child-metadata.txt")

genus_merged_graphs = metadata_merged %>% 
  inner_join(genus_merged) %>% 
  mutate(Age = case_when(Group == "C" ~ "Children",
                         Group == "A" ~ "Adults")) 
genus_merged_graphs_relab = decostand(genus_merged_graphs[,5:1507], method = "total")
genus_merged_graphs_relab2 = cbind(genus_merged_graphs[,1:4], genus_merged_graphs[,1508], genus_merged_graphs_relab)

genus_merged_graphs_relab2$Race = factor(genus_merged_graphs_relab2$Race, levels = c("White", "Black", "Asian/Pacific Islander"))

plot_ac_Entero = ggplot(genus_merged_graphs_relab2,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacteriales;f__Enterobacteriaceae;__`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c", 
                        "#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c", 
                         "#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") + 
  ylab("Enterobacteriaceae\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  facet_wrap(~ Age)

plot_ac_Prevot = ggplot(genus_merged_graphs_relab2,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c", 
                        "#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c", 
                         "#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") + 
  ylab("Prevotella\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  facet_wrap(~ Age)

plot_ac_Rumin = ggplot(genus_merged_graphs_relab2,
                      aes(x = Race, 
                          y = `k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus`)) + 
  stat_boxplot(geom = 'errorbar', width = 0, 
               color = "black") +
  geom_boxplot(coef = 0, outlier.shape = NA,
               color = "white",
               fill = c("#377eb8", "#f6d128", "#e41a1c", 
                        "#377eb8", "#f6d128", "#e41a1c"),
               fatten = 4) +
  geom_boxplot(coef = 0, outlier.shape = NA, 
               color = c("#377eb8", "#f6d128", "#e41a1c", 
                         "#377eb8", "#f6d128", "#e41a1c"),
               fill = NA, 
               fatten = NULL) + 
  geom_jitter(width = 0.2,
              color = "black", size = 0.5) +
  scale_y_continuous(trans = "sqrt") + 
  ylab("Ruminococcus\nRelative Abundance") +
  theme_classic() + 
  theme(axis.title.y=element_text(size=rel(1.2)),
        axis.text.y=element_text(size=rel(1)),
        axis.title.x=element_blank(), 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  facet_wrap(~ Age)

legend_plot_ac = ggboxplot(genus_merged_graphs_relab2, "Race", "k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Ruminococcus", fill = "Race", palette = c("#377eb8", "#f6d128", "#e41a1c"), color = "white") +
  theme(legend.position = "right")
```

##Graphs for publication
```{r pub_graphs_fig1v5}
age_legend = get_legend(weight_age)

tiff(file="fig1v5.tif", res=300, width=24, height=8, units="in")
row1 = plot_grid(alpha_all_age_graph, weight_age + theme(legend.position = "none"),
                 nrow = 1, ncol = 2,
                 rel_widths = c(2, 1.1), labels = c("A", "B"),
                 label_size = 24)
fig1 = plot_grid(row1, age_legend, nrow = 2, ncol = 1, 
                rel_heights = c(1, 0.2))

fig1
dev.off()

setEPS()
postscript(file="fig1v5.eps", width=24, height=8, paper = "special")
fig1
dev.off()
```

```{r pub_graphs_fig2v2}
nmds_legend = get_legend(weight_age_6week3month)

tiff(file="fig2v2.tif", res=300, width=22, height=20, units="in")


row1 = plot_grid(faithr_dw + theme(legend.position = "none"), 
                 observedr_dw + theme(legend.position = "none"), 
                 chaor_dw + theme(legend.position = "none"), 
                 shannonr_dw + theme(legend.position = "none"), 
                 pielour_dw + theme(legend.position = "none"),
                 nrow = 1, ncol = 5, 
                 align = "h", axis = "tb")
row2 = plot_grid(faithe_dw + theme(legend.position = "none"), 
                 observede_dw + theme(legend.position = "none"), 
                 chaoe_dw + theme(legend.position = "none"), 
                 shannone_dw + theme(legend.position = "none"), 
                 pieloue_dw + theme(legend.position = "none"),
                 nrow = 1, ncol = 5, 
                 align = "h", axis = "tb")

top = plot_grid(row1, row2, combo_legend,
                nrow = 3, ncol = 1,
                rel_heights = c(2,1,.2))

row4 = plot_grid(weight_age_03month + 
                     theme(legend.position = "none"),
                    weight_age_312month + 
                     theme(legend.position = "none"),
                   weight_age_13year + 
                     theme(legend.position = "none"),
                 axis = "tb", align = "v", nrow = 1, ncol = 3)

bottom = plot_grid(row4, nmds_legend,
                 nrow = 2, ncol = 1,
                 rel_heights = c(2,.2))

plot = plot_grid(top, bottom,
                 nrow = 2, ncol = 1,
                 rel_heights = c(3,2),
                 labels = c("A", "B"), label_size = 36)
plot
dev.off()

setEPS()
postscript(file="fig2v2.eps", width=22, height=20, paper = "special")
plot
dev.off()
```

```{r pub_graphs_alpha_comp}
tiff(file="alpha_comp.tif", res=90, width=15, height=15, units="in")
plot_grid(alpha_6week3month_race, alpha_312month_race,alpha_13year_race,
          alpha_6week3month_race_asvs, alpha_312month_race_asvs, 
          alpha_13year_race_asvs, alpha_6week3month_race_faith, 
          alpha_312month_race_faith, alpha_13year_race_faith, 
          nrow = 3, ncol = 3, align = "hv")
dev.off()
```

```{r pub_graphs_fig3b}
venn.diagram(x = list(c, a), 
             category.names = c("Children (57)", "Adults (144)"), 
             filename = "fig3Bv2.tif", output = T, imagetype = "tiff", 
             height = 5, width = 5, units = "in", resolution = 300, 
             col = c("#4b265c", "#25a165"), lwd = 1,
             fill = c(alpha("#4b265c", 0.3), alpha("#25a165", 0.3)), 
             fontfamily = "sans", cat.fontfamily = "sans",
             cat.default.pos = "outer", cat.dist = c(0.1, 0.1),
             margin = 0.2)

venn.diagram(x = list(c, a), 
             category.names = c("Children (57)", "Adults (144)"), 
             filename = "fig3Bv2.svg", output = T, imagetype = "svg", 
             height = 5, width = 5, units = "in", resolution = 300, 
             col = c("#4b265c", "#25a165"), lwd = 1,
             fill = c(alpha("#4b265c", 0.3), alpha("#25a165", 0.3)), 
             fontfamily = "sans", cat.fontfamily = "sans",
             cat.default.pos = "outer", cat.dist = c(0.1, 0.1),
             margin = 0.2)
```

```{r fig3C}
tiff(file="fig3Cv20.tif", res=300, width=7, height=10, units="in")
plot_grid(roc_curve_child_r, roc_curve_child_e, nrow = 2, ncol = 1,
          rel_widths = c(1, 1), align = "hv", axis = "tb")
dev.off()

setEPS()
postscript(file="fig3Cv20.eps", width=7, height=10, paper = "special")
plot_grid(roc_curve_child_r, roc_curve_child_e, nrow = 2, ncol = 1,
          rel_widths = c(1, 1), align = "hv", axis = "tb")
dev.off()
```

```{r fig3c}
tiff(file="fig3Av2.tif", 
     res=300, width=10, height=18, units="in")
plot3c = plot_grid(plot_Haemo, histo_Haemo, 
                   plot_P.copri, histo_P.copri, 
                   plot_Faecali, histo_Faecali,
                   plot_V.dispar, histo_V.dispar,
                   plot_V.parvula, histo_V.parvula,
          nrow = 5, ncol = 2, rel_widths = c(0.8,1.2),
          axis = "tbl", align = "hv")
plot3c
dev.off()

setEPS()
postscript(file="fig3Av2.eps", width=10, height=18, paper = "special")
plot3c
dev.off()
```

```{r fig3c3}
tiff(file="fig3Av3.tif", 
     res=300, width=10, height=18, units="in")
plot3c = plot_grid(plot_Haemo, histo_Haemo2, histo_Haemo3,
                   plot_P.copri, histo_P.copri2, histo_P.copri3,
                   plot_Faecali, histo_Faecali2, histo_Faecali3,
                   plot_V.dispar, histo_V.dispar2, histo_V.dispar3,
                   plot_V.parvula, histo_V.parvula2, histo_V.parvula3,
          nrow = 5, ncol = 3, rel_widths = c(0.8, 0.6, 0.6),
          axis = "tbl", align = "hv")
plot3c
dev.off()

setEPS()
postscript(file="fig3Av3.eps", width=10, height=18, paper = "special")
plot3c
dev.off()
```

###Supplement graphs
```{r figs1}
tiff(file = "figs1.tif", res = 300, width = 12, height = 8, units = "in")
unweight_age
dev.off()

setEPS()
postscript(file="figs1.eps", width=12, height=8, paper = "special")
unweight_age
dev.off()
```

```{r figs2}
tiff(file = "figs2.tif", res = 300, width = 12, height = 8, units = "in")
unweight_race
dev.off()

setEPS()
postscript(file="figs2.eps", width=12, height=8, paper = "special")
unweight_race
dev.off()
```

```{r figs3}
nmds_legend = get_legend(weight_age_6week3month)

tiff(file="figs3.tif", res=300, width=12, height=24, units="in")


top = plot_grid(weight_age_01week + theme(legend.position = "none"), 
                weight_age_16week + theme(legend.position = "none"),
                weight_age_6week3month + theme(legend.position = "none"), 
                weight_age_36month + theme(legend.position = "none"), 
                weight_age_69month + theme(legend.position = "none"),
                weight_age_912month + theme(legend.position = "none"),
                weight_age_012month + theme(legend.position = "none"),
                weight_age_312year + theme(legend.position = "none"),
                nrow = 4, ncol = 2, 
                align = "hv", axis = "lb")

plot = plot_grid(top, nmds_legend,
                 nrow = 2, ncol = 1,
                 rel_heights = c(4, 0.2))
plot
dev.off()

setEPS()
postscript(file="figs3.eps", width=12, height=24, paper = "special")
plot
dev.off()
```

```{r figs4}
nmds_legend = get_legend(unweight_age_6week3month)

tiff(file="figs4.tif", res=300, width=18, height=24, units="in")
plot = plot_grid(unweight_age_01week + theme(legend.position = "none"), 
                unweight_age_16week + theme(legend.position = "none"),
                unweight_age_6week3month + theme(legend.position = "none"),
                unweight_age_03month + theme(legend.position = "none"), 
                unweight_age_312month + theme(legend.position = "none"),
                unweight_age_36month + theme(legend.position = "none"), 
                unweight_age_69month + theme(legend.position = "none"),
                unweight_age_912month + theme(legend.position = "none"),
                unweight_age_012month + theme(legend.position = "none"),
                unweight_age_13year + theme(legend.position = "none"),
                unweight_age_312year + theme(legend.position = "none"),
                nmds_legend,
                nrow = 4, ncol = 3, 
                align = "hv", axis = "lb")
plot
dev.off()

setEPS()
postscript(file="figs4.eps", width=18, height=24, paper = "special")
plot
dev.off()
```

```{r figs16}
plot_legend = get_legend(legend_plot_ac)

tiff(file="figs16.tif", res=300, width=20, height=6, units="in")
plot = plot_grid(plot_ac_Entero + theme(legend.position = "none"), 
                plot_ac_Prevot + theme(legend.position = "none"),
                plot_ac_Rumin + theme(legend.position = "none"),
                plot_legend,
                nrow = 1, ncol = 4, 
                rel_widths = c(1,1,1,0.3),
                align = "h", axis = "b", 
                labels = c("A", "B", "C", ""))
plot
dev.off()

setEPS()
postscript(file="figs16.eps", width=20, height=6, paper = "special")
plot
dev.off()
```

##Graphs for presentation

```{r taxa_graphs}
tiff(file="taxa_graphs_present_1.tif", 
     res=300, width=16, height=5, units="in")
plot1p = plot_grid(plot_Bifidobacterium, plot_Haemo, 
          plot_P.copri, plot_Faecali, 
          nrow = 1, ncol = 4,
          axis = "tb", align = "hv")
plot1p
dev.off()

setEPS()
postscript(file="taxa_graphs_present_1.eps", width=16, height=5, paper = "special")
plot1p
dev.off()

tiff(file="taxa_graphs_present_2.tif", 
     res=300, width=12, height=5, units="in")
plot2p = plot_grid(plot_V.dispar, plot_V.parvula, 
          plot_Dialister,
          nrow = 1, ncol = 3,
          axis = "tb", align = "hv")
plot2p
dev.off()

setEPS()
postscript(file="taxa_graphs_present_2.eps", width=12, height=5, paper = "special")
plot2p
dev.off()
```
